From 5fcf2d94d7c56f0cf7bdb05e25a0503bb1c2a6cd Mon Sep 17 00:00:00 2001
From: lwfinger <Larry.Finger@lwfinger.net>
Date: Fri, 19 Apr 2013 14:08:09 -0700
Subject: [PATCH 041/210] Initial commit

---
 README.md  |  4 ++++
 1 file changed, 4 insertions(+)
 create mode 100644 README.md

--- /dev/null
+++ b/drivers/staging/rtl8192du/README.md
@@ -0,0 +1,4 @@
+rtl8192du
+=========
+
+Source code for RTL8192DU device
\ No newline at end of file
