From b6998bf57ab4aa416f4aaec6b63578fa4c679884 Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Sat, 27 Apr 2013 11:39:21 -0500
Subject: [PATCH 070/390] Remove "typedef enum" from rtw_eeprom.h

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 include/rtw_eeprom.h | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

--- a/drivers/staging/rtl8192du/include/rtw_eeprom.h
+++ b/drivers/staging/rtl8192du/include/rtw_eeprom.h
@@ -62,8 +62,7 @@
 // Besides, CustomerID of registry has precedence of that of EEPROM.
 // defined below. 060703, by rcnjko.
 //
-typedef enum _RT_CUSTOMER_ID
-{
+enum RT_CUSTOMER_ID {
 	RT_CID_DEFAULT = 0,
 	RT_CID_8187_ALPHA0 = 1,
 	RT_CID_8187_SERCOMM_PS = 2,
@@ -101,7 +100,7 @@ typedef enum _RT_CUSTOMER_ID
 	RT_CID_819x_PRONETS = 34,
 	RT_CID_819x_Edimax_ASUS = 35,
 	RT_CID_819x_CAMEO_NETGEAR = 36,
-}RT_CUSTOMER_ID, *PRT_CUSTOMER_ID;
+};
 
 struct eeprom_priv
 {
