From 7d4f4331dcb69a2a597fb70f52f6b390d85a096e Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Sun, 28 Apr 2013 17:25:40 -0500
Subject: [PATCH 085/390] Fix compile error in previous commit

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 include/osdep_service.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/staging/rtl8192du/include/osdep_service.h
+++ b/drivers/staging/rtl8192du/include/osdep_service.h
@@ -90,9 +90,9 @@
 #endif
 
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,37))
-	typedef struct mutex		_mutex;
+	#define _mutex	struct mutex
 #else
-	typedef struct semaphore	_mutex;
+	#define _mutex	struct semaphore
 #endif
 	struct	__queue	{
 		struct	list_head	queue;
