From 77890254186daa4127175f2b50c0a783caeec967 Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Sun, 28 Apr 2013 18:41:10 -0500
Subject: [PATCH 089/210] Clean up typedef statements in ieee80211.h

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 include/ieee80211.h | 15 ++++-----------
 1 file changed, 4 insertions(+), 11 deletions(-)

--- a/drivers/staging/rtl8192du/include/ieee80211.h
+++ b/drivers/staging/rtl8192du/include/ieee80211.h
@@ -199,7 +199,7 @@ enum NETWORK_TYPE
 #define IsSupportedTxMCS(NetType) ((NetType) & (WIRELESS_11_24N|WIRELESS_11_5N) ? true : false)
 
 
-typedef struct ieee_param {
+struct ieee_param {
 	u32 cmd;
 	u8 sta_addr[ETH_ALEN];
 	union {
@@ -240,14 +240,14 @@ typedef struct ieee_param {
 #endif
 
 	} u;
-}ieee_param;
+};
 
 #ifdef CONFIG_AP_MODE
-typedef struct ieee_param_ex {
+struct ieee_param_ex {
 	u32 cmd;
 	u8 sta_addr[ETH_ALEN];
 	u8 data[0];
-}ieee_param_ex;
+};
 
 struct sta_data{
 	u16 aid;
@@ -961,13 +961,6 @@ extern __inline int is_broadcast_mac_add
 #define CFG_IEEE80211_RESERVE_FCS (1<<0)
 #define CFG_IEEE80211_COMPUTE_FCS (1<<1)
 
-typedef struct tx_pending_t{
-	int frag;
-	struct ieee80211_txb *txb;
-}tx_pending_t;
-
-
-
 #define MAXTID	16
 
 #define IEEE_A            (1<<0)
