From 10f4a07e816ba1635308ef03460d413506d23dd9 Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Sun, 28 Apr 2013 19:57:03 -0500
Subject: [PATCH 092/390] Convert typedef statements in swab.h

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 include/swab.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

--- a/drivers/staging/rtl8192du/include/swab.h
+++ b/drivers/staging/rtl8192du/include/swab.h
@@ -21,19 +21,19 @@
 #define _LINUX_BYTEORDER_SWAB_H
 
 #ifndef __u16
-typedef unsigned short __u16;
+  #define __u16 unsigned short
 #endif
 
 #ifndef __u32
-typedef unsigned int	__u32;
+  #define __u32 unsigned int
 #endif
 
 #ifndef __u8
-typedef unsigned char __u8;
+  #define __u8 unsigned char
 #endif
 
 #ifndef __u64
-typedef unsigned long long	__u64;
+  #define __u64 unsigned long long
 #endif
 
 
