From dd2816bfd972000c4d20eeed257cea926ccc7267 Mon Sep 17 00:00:00 2001
From: "David B. Adrian" <david.adrian@tum.de>
Date: Thu, 8 Aug 2013 16:18:55 +0200
Subject: [PATCH 169/390] fix in rtw_pwrctrl

---
 core/rtw_pwrctrl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/staging/rtl8192du/core/rtw_pwrctrl.c
+++ b/drivers/staging/rtl8192du/core/rtw_pwrctrl.c
@@ -1140,7 +1140,7 @@ static void rtw_late_resume(struct early
 	struct pwrctrl_priv *pwrpriv =
 	    container_of(h, struct pwrctrl_priv, early_suspend);
 	struct rtw_adapter *adapter =
-	    container_of(pwrpriv, _adapter, pwrctrlpriv);
+	    container_of(pwrpriv, struct rtw_adapter, pwrctrlpriv);
 
 	DBG_8192D("%s\n", __func__);
 	if (pwrpriv->do_late_resume) {
