From e9541db57f36bff0929d4c87aaaa08139cc848f6 Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Tue, 10 Sep 2013 21:51:23 -0500
Subject: [PATCH 177/390] rtl8192du: Fix smatch warning in os_dep/usb_intf.c

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 os_dep/usb_intf.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

--- a/drivers/staging/rtl8192du/os_dep/usb_intf.c
+++ b/drivers/staging/rtl8192du/os_dep/usb_intf.c
@@ -801,7 +801,7 @@ static int rtw_resume(struct usb_interfa
 int rtw_resume_process(struct rtw_adapter *padapter)
 {
 	struct net_device *pnetdev;
-	struct pwrctrl_priv *pwrpriv;
+	struct pwrctrl_priv *pwrpriv = NULL;
 	int ret = -1;
 	u32 start_time = rtw_get_current_time();
 	_func_enter_;
@@ -879,7 +879,8 @@ exit:
 	rtw_unlock_suspend();
 	#endif /* CONFIG_RESUME_IN_WORKQUEUE */
 
-	pwrpriv->bInSuspend = false;
+	if (pwrpriv)
+		pwrpriv->bInSuspend = false;
 	DBG_8192D("<===  %s return %d.............. in %dms\n", __func__
 		, ret, rtw_get_passing_time_ms(start_time));
 
