From 87bdb21bea012b592487adb68dcd2d2184944226 Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Tue, 10 Sep 2013 21:56:37 -0500
Subject: [PATCH 179/390] rtl8192du: Fix smatch warning in core/rtw_recv.c

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 core/rtw_recv.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/drivers/staging/rtl8192du/core/rtw_recv.c
+++ b/drivers/staging/rtl8192du/core/rtw_recv.c
@@ -2277,8 +2277,7 @@ static int amsdu_to_msdu(struct rtw_adap
 		}
 
 		/* Indicat the packets to upper layer */
-		if (sub_skb) {
-			/* memset(sub_skb->cb, 0, sizeof(sub_skb->cb)); */
+		{
 
 #ifdef CONFIG_BR_EXT
 			/*  Insert NAT2.5 RX here! */
