From 0b7c7440f2f0c8577d791f32d67b75241519d45f Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Thu, 17 Oct 2013 22:21:07 -0500
Subject: [PATCH 187/210] rtl8192du: Set device type to wlan

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 os_dep/os_intfs.c | 5 +++++
 1 file changed, 5 insertions(+)

--- a/drivers/staging/rtl8192du/os_dep/os_intfs.c
+++ b/drivers/staging/rtl8192du/os_dep/os_intfs.c
@@ -951,6 +951,10 @@ int rtw_init_netdev_name(struct net_devi
 	return 0;
 }
 
+static const struct device_type wlan_type = {
+	.name = "wlan",
+};
+
 struct net_device *rtw_init_netdev(struct rtw_adapter *old_padapter)
 {
 	struct rtw_adapter *padapter;
@@ -966,6 +970,7 @@ struct net_device *rtw_init_netdev(struc
 	if (!pnetdev)
 		return NULL;
 
+	pnetdev->dev.type = &wlan_type;
 	padapter = rtw_netdev_priv(pnetdev);
 	padapter->pnetdev = pnetdev;
 
