From 2ebd0a14f530fd879527aa8c7bc5a534ff4081d7 Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Wed, 19 Feb 2014 21:40:48 -0600
Subject: [PATCH 216/390] rtl8192du: Remove CONFIG_FILE_FWIMG

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 include/autoconf.h | 1 -
 os_dep/os_intfs.c  | 6 ------
 2 files changed, 7 deletions(-)

--- a/drivers/staging/rtl8192du/include/autoconf.h
+++ b/drivers/staging/rtl8192du/include/autoconf.h
@@ -35,7 +35,6 @@
 /* define CONFIG_H2CLBK	1 */
 
 #define CONFIG_EMBEDDED_FWIMG	1
-/* define CONFIG_FILE_FWIMG */
 
 #ifdef CONFIG_WAKE_ON_WLAN
 #define CONFIG_WOWLAN 1
--- a/drivers/staging/rtl8192du/os_dep/os_intfs.c
+++ b/drivers/staging/rtl8192du/os_dep/os_intfs.c
@@ -203,12 +203,6 @@ module_param(rtw_max_roaming_times, uint
 MODULE_PARM_DESC(rtw_max_roaming_times,"The max roaming times to try");
 #endif /* CONFIG_LAYER2_ROAMING */
 
-#ifdef CONFIG_FILE_FWIMG
-char *rtw_fw_file_path = "";
-module_param(rtw_fw_file_path, charp, 0644);
-MODULE_PARM_DESC(rtw_fw_file_path, "The path of fw image");
-#endif /* CONFIG_FILE_FWIMG */
-
 module_param(rtw_mc2u_disable, int, 0644);
 
 module_param(rtw_mac_phy_mode, int, 0644);
