From e0f3d605f8513f46718706f31bc570d3b1b29aef Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Thu, 20 Feb 2014 13:35:19 -0600
Subject: [PATCH 230/390] rtl8192du: Remove MP_FIRMWARE_OFFLOAD

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 core/rtw_cmd.c    |  4 ----
 include/rtw_cmd.h | 62 -------------------------------------------------------
 2 files changed, 66 deletions(-)

--- a/drivers/staging/rtl8192du/core/rtw_cmd.c
+++ b/drivers/staging/rtl8192du/core/rtw_cmd.c
@@ -541,12 +541,8 @@ u8 rtw_setdatarate_cmd(struct rtw_adapte
 	}
 
 	init_h2fwcmd_w_parm_no_rsp(ph2c, pbsetdataratepara, GEN_CMD_CODE(_SETDATARATE));
-#ifdef MP_FIRMWARE_OFFLOAD
-	pbsetdataratepara->curr_rateidx = *(u32 *)rateset;
-#else
 	pbsetdataratepara->mac_id = 5;
 	memcpy(pbsetdataratepara->datarates, rateset, NUMRATES);
-#endif
 	res = rtw_enqueue_cmd(pcmdpriv, ph2c);
 exit:
 	return res;
--- a/drivers/staging/rtl8192du/include/rtw_cmd.h
+++ b/drivers/staging/rtl8192du/include/rtw_cmd.h
@@ -372,12 +372,8 @@ Command Mode
 
 */
 struct setdatarate_parm {
-#ifdef MP_FIRMWARE_OFFLOAD
-	u32	curr_rateidx;
-#else
 	u8	mac_id;
 	u8	datarates[NUMRATES];
-#endif
 };
 
 /*
@@ -758,64 +754,6 @@ struct set_ch_parm {
 	u8 ch_offset;
 };
 
-#ifdef MP_FIRMWARE_OFFLOAD
-/*H2C Handler index: 47 */
-struct SetTXPOWER_parm
-{
-	u8 TXPOWER;
-};
-
-/*H2C Handler index: 48 */
-struct SWITCHANTENNA_parm
-{
-	u16 antenna_tx;
-	u16 antenna_rx;
-/* 	R_ANTENNA_SELECT_CCK cck_txrx; */
-	u8 cck_txrx;
-};
-
-/*H2C Handler index: 49 */
-struct setcrystalcap_parm
-{
-	u32 curr_crystalcap;
-};
-
-/*H2C Handler index: 50 */
-struct setsinglecarriertx_parm
-{
-	u8 start;
-};
-
-/*H2C Handler index: 51 */
-struct setsingletonetx_parm
-{
-	u8 start;
-	u8 curr_rfpath;
-};
-
-/*H2C Handler index: 52 */
-struct setcarriersuppressiontx_parm
-{
-	u8 start;
-	u32 curr_rateidx;
-};
-
-/*H2C Handler index: 53 */
-struct setcontinuoustx_parm
-{
-	u8 start;
-	u8 cck_flag; /*1:CCK 2:OFDM*/
-	u32 curr_rateidx;
-};
-
-/*H2C Handler index: 54 */
-struct switchbandwidth_parm
-{
-	u8 curr_bandwidth;
-};
-
-#endif	/* MP_FIRMWARE_OFFLOAD */
-
 /*H2C Handler index: 59 */
 struct setchannelplan_param
 {
