From 4e83a470b6bf5006975d5b356190e733ad5cd4af Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Wed, 26 Feb 2014 23:03:39 -0600
Subject: [PATCH 245/390] Modify code for new API for ndo_select_queue in
 kernel 3.14

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 os_dep/os_intfs.c | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

--- a/drivers/staging/rtl8192du/os_dep/os_intfs.c
+++ b/drivers/staging/rtl8192du/os_dep/os_intfs.c
@@ -774,7 +774,7 @@ static unsigned int rtw_classify8021d(st
 }
 
 static u16 rtw_select_queue(struct net_device *dev, struct sk_buff *skb,
-			    void *accel)
+			    void *accel, select_queue_fallback_t fallback)
 {
 	struct rtw_adapter	*padapter = rtw_netdev_priv(dev);
 	struct mlme_priv *pmlmepriv = &padapter->mlmepriv;
@@ -1361,14 +1361,6 @@ _adapter *rtw_drv_add_vir_if (struct rtw
 	struct dvobj_priv *pdvobjpriv;
 	u8 mac[ETH_ALEN];
 
-/*
-	if ((primary_padapter->bup == false) ||
-		(rtw_buddy_adapter_up(primary_padapter) == false))
-	{
-		goto error_rtw_drv_add_iface;
-	}
-
-*/
 	/****** init netdev ******/
 	pnetdev = rtw_init_netdev(NULL);
 	if (!pnetdev)
