From 6b3d5e4e19a9d71cd8f514eb2d5505eb4d242436 Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Sat, 15 Mar 2014 13:35:04 -0500
Subject: [PATCH 269/390] Remove FULL_UNROLL

Not used.

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 core/rtw_security.c | 20 --------------------
 1 file changed, 20 deletions(-)

--- a/drivers/staging/rtl8192du/core/rtw_security.c
+++ b/drivers/staging/rtl8192du/core/rtw_security.c
@@ -2229,9 +2229,7 @@ static void rijndaelEncrypt(u32 rk[/*44
 {
 	u32 s0, s1, s2, s3, t0, t1, t2, t3;
 	int Nr = 10;
-#ifndef FULL_UNROLL
 	int r;
-#endif /* ?FULL_UNROLL */
 
 	/*
 	 * map byte array block to cipher state
@@ -2250,22 +2248,6 @@ do {									\
 	d##3 = TE0(s##3) ^ TE1(s##0) ^ TE2(s##1) ^ TE3(s##2) ^ rk[4 * i + 3]; \
 } while (0)
 
-#ifdef FULL_UNROLL
-
-	ROUND(1, t, s);
-	ROUND(2, s, t);
-	ROUND(3, t, s);
-	ROUND(4, s, t);
-	ROUND(5, t, s);
-	ROUND(6, s, t);
-	ROUND(7, t, s);
-	ROUND(8, s, t);
-	ROUND(9, t, s);
-
-	rk += Nr << 2;
-
-#else /* !FULL_UNROLL */
-
 	/* Nr - 1 full rounds: */
 	r = Nr >> 1;
 	for (;;) {
@@ -2276,8 +2258,6 @@ do {									\
 		ROUND(0, s, t);
 	}
 
-#endif /* ?FULL_UNROLL */
-
 #undef ROUND
 
 	/*
