From a15ac98e48c222cdfa99a202678ad6a2d7021a98 Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Sat, 15 Mar 2014 15:05:12 -0500
Subject: [PATCH 287/390] Remove unused CONFIG_DRV_ISSUE_PROV_REQ

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 core/rtw_p2p.c | 8 --------
 1 file changed, 8 deletions(-)

--- a/drivers/staging/rtl8192du/core/rtw_p2p.c
+++ b/drivers/staging/rtl8192du/core/rtw_p2p.c
@@ -1836,14 +1836,6 @@ int rtw_p2p_check_frames(struct rtw_adap
 					  tx ? "Tx" : "Rx", dialogToken);
 
 				if (tx) {
-#ifdef CONFIG_DRV_ISSUE_PROV_REQ /*  IOT FOR S2 */
-					if (pwdev_priv->provdisc_req_issued == false) {
-						rtw_cfg80211_issue_p2p_provision_request(padapter, buf, len);
-						pwdev_priv->provdisc_req_issued = true;
-						rtw_msleep_os(200);
-					}
-#endif /* CONFIG_DRV_ISSUE_PROV_REQ */
-
 #ifdef CONFIG_CONCURRENT_MODE
 					if (check_buddy_fwstate(padapter, _FW_LINKED))
 						rtw_cfg80211_adjust_p2pie_channel(padapter, frame_body, len-sizeof(struct ieee80211_hdr_3addr));
