From 50c4a2dbdb02ea1f0556396ade729d19414be45e Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Sat, 15 Mar 2014 16:53:32 -0500
Subject: [PATCH 296/390] Remove unused CONFIG_FORCE_HW_LED

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 hal/usb_halinit.c | 4 ----
 1 file changed, 4 deletions(-)

--- a/drivers/staging/rtl8192du/hal/usb_halinit.c
+++ b/drivers/staging/rtl8192du/hal/usb_halinit.c
@@ -2831,10 +2831,6 @@ _ReadLEDSetting(
 			pledpriv->LedStrategy = SW_LED_MODE0;
 			break;
 	}
-
-	#ifdef CONFIG_FORCE_HW_LED
-	pledpriv->LedStrategy = HW_LED;
-	#endif
 }
 
 #ifdef CONFIG_WOWLAN
