From 3b454d98b37da2c09cd0c347e9ba65c63e80c967 Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Sat, 15 Mar 2014 17:32:32 -0500
Subject: [PATCH 302/390] Remove undefined RTW_DVOBJ_CHIP_HW_TYPE

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 os_dep/usb_intf.c | 2 --
 1 file changed, 2 deletions(-)

--- a/drivers/staging/rtl8192du/os_dep/usb_intf.c
+++ b/drivers/staging/rtl8192du/os_dep/usb_intf.c
@@ -666,11 +666,9 @@ static struct rtw_adapter *rtw_usb_if1_i
 	dvobj->padapters[dvobj->iface_nums++] = padapter;
 #endif
 
-	#ifndef RTW_DVOBJ_CHIP_HW_TYPE
 	/* step 1-1., decide the chip_type via vid/pid */
 	padapter->interface_type = RTW_USB;
 	decide_chip_type_by_usb_device_id(padapter, pdid);
-	#endif
 
 	if ((pnetdev = rtw_init_netdev(padapter)) == NULL) {
 		goto free_adapter;
