From a0f405feca1b3bfc241bdad5ae2081fdcd0a51d4 Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Sat, 15 Mar 2014 17:46:50 -0500
Subject: [PATCH 304/390] Remove undefined CONFIG_REDUCE_USB_TX_INT

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 os_dep/usb_ops_linux.c | 8 --------
 1 file changed, 8 deletions(-)

--- a/drivers/staging/rtl8192du/os_dep/usb_ops_linux.c
+++ b/drivers/staging/rtl8192du/os_dep/usb_ops_linux.c
@@ -281,14 +281,6 @@ u32 usb_write_port(struct intf_hdl *pint
 	/* translate DMA FIFO addr to pipehandle */
 	pipe = ffaddr2pipehdl(pdvobj, addr);
 
-#ifdef CONFIG_REDUCE_USB_TX_INT
-	if ((pxmitpriv->free_xmitbuf_cnt%NR_XMITBUFF == 0) ||
-	    (pxmitbuf->ext_tag == true))
-		purb->transfer_flags  &=  (~URB_NO_INTERRUPT);
-	else
-		purb->transfer_flags  |=  URB_NO_INTERRUPT;
-#endif
-
 	usb_fill_bulk_urb(purb, pusbd, pipe,
 			  pxmitframe->buf_addr, /*  pxmitbuf->pbuf */
 			  cnt,
