From 1c26b4a0fc8c8f624566e8ae414a343645afdfb6 Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Sat, 15 Mar 2014 17:54:46 -0500
Subject: [PATCH 306/390] Removed undefined CUSTOM_OOB_GPIO_NUM

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 os_dep/rtw_android.c | 4 ----
 1 file changed, 4 deletions(-)

--- a/drivers/staging/rtl8192du/os_dep/rtw_android.c
+++ b/drivers/staging/rtl8192du/os_dep/rtw_android.c
@@ -537,11 +537,7 @@ int wifi_get_irq_number(unsigned long *i
 		*irq_flags_ptr = wifi_irqres->flags & IRQF_TRIGGER_MASK;
 		return (int)wifi_irqres->start;
 	}
-#ifdef CUSTOM_OOB_GPIO_NUM
-	return CUSTOM_OOB_GPIO_NUM;
-#else
 	return -1;
-#endif
 }
 
 int wifi_set_power(int on, unsigned long msec)
