From cec9037aec486099e1d48b01fd5ae12ad394e255 Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Sat, 15 Mar 2014 17:59:22 -0500
Subject: [PATCH 308/390] Remove undefined CONFIG_EASY_REPLACEMENT

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 os_dep/os_intfs.c | 28 ----------------------------
 1 file changed, 28 deletions(-)

--- a/drivers/staging/rtl8192du/os_dep/os_intfs.c
+++ b/drivers/staging/rtl8192du/os_dep/os_intfs.c
@@ -399,36 +399,8 @@ int rtw_init_netdev_name(struct net_devi
 {
 	struct rtw_adapter *padapter = rtw_netdev_priv(pnetdev);
 
-#ifdef CONFIG_EASY_REPLACEMENT
-	struct net_device	*TargetNetdev = NULL;
-	struct rtw_adapter			*target_adapter = NULL;
-	struct net		*devnet = NULL;
-
-	if (padapter->bDongle == 1)
-	{
-		devnet = dev_net(pnetdev);
-		TargetNetdev = dev_get_by_name(devnet, "wlan0");
-		if (TargetNetdev) {
-			DBG_8192D("Force onboard module driver disappear !!!\n");
-			target_adapter = rtw_netdev_priv(TargetNetdev);
-			target_adapter->DriverState = DRIVER_DISAPPEAR;
-
-			padapter->pid[0] = target_adapter->pid[0];
-			padapter->pid[1] = target_adapter->pid[1];
-			padapter->pid[2] = target_adapter->pid[2];
-
-			dev_put(TargetNetdev);
-			unregister_netdev(TargetNetdev);
-
-			padapter->DriverState = DRIVER_REPLACE_DONGLE;
-		}
-	}
-#endif
-
 	if (dev_alloc_name(pnetdev, ifname) < 0)
-	{
 		RT_TRACE(_module_os_intfs_c_, _drv_err_, ("dev_alloc_name, fail!\n"));
-	}
 
 	netif_carrier_off(pnetdev);
 	return 0;
