From 93d92ec708764ed9a96a028d0142c872fbf1fa97 Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Sat, 15 Mar 2014 18:14:52 -0500
Subject: [PATCH 310/390] Remove undefined CONFIG_XMIT_THREAD_MODE

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 os_dep/os_intfs.c | 17 ++---------------
 1 file changed, 2 insertions(+), 15 deletions(-)

--- a/drivers/staging/rtl8192du/os_dep/os_intfs.c
+++ b/drivers/staging/rtl8192du/os_dep/os_intfs.c
@@ -449,11 +449,6 @@ u32 rtw_start_drv_threads(struct rtw_ada
 	u32 _status = _SUCCESS;
 
 	RT_TRACE(_module_os_intfs_c_, _drv_info_, ("+rtw_start_drv_threads\n"));
-#ifdef CONFIG_XMIT_THREAD_MODE
-	padapter->xmitThread = kthread_run(rtw_xmit_thread, padapter, "RTW_XMIT_THREAD");
-	if (IS_ERR(padapter->xmitThread))
-		_status = _FAIL;
-#endif
 
 #ifdef CONFIG_CONCURRENT_MODE
 	if (padapter->isprimary == true)
@@ -477,19 +472,11 @@ void rtw_stop_drv_threads (struct rtw_ad
 	if (padapter->isprimary == true)
 #endif /* CONFIG_CONCURRENT_MODE */
 	{
-		/* Below is to termindate rtw_cmd_thread & event_thread... */
+		/* Below is to terminate rtw_cmd_thread & event_thread... */
 		_rtw_up_sema(&padapter->cmdpriv.cmd_queue_sema);
-		if (padapter->cmdThread) {
+		if (padapter->cmdThread)
 			_rtw_down_sema(&padapter->cmdpriv.terminate_cmdthread_sema);
-		}
 	}
-
-#ifdef CONFIG_XMIT_THREAD_MODE
-	/*  Below is to termindate tx_thread... */
-	_rtw_up_sema(&padapter->xmitpriv.xmit_sema);
-	_rtw_down_sema(&padapter->xmitpriv.terminate_xmitthread_sema);
-	RT_TRACE(_module_os_intfs_c_, _drv_info_, ("\n drv_halt: rtw_xmit_thread can be terminated !\n"));
-#endif
 }
 
 static u8 rtw_init_default_value(struct rtw_adapter *padapter)
