From be0e4d968528847e5852fd916daccd356f32cf62 Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Sat, 15 Mar 2014 20:03:54 -0500
Subject: [PATCH 312/390] Convert signal display to dBm

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 os_dep/ioctl_linux.c | 17 ++---------------
 1 file changed, 2 insertions(+), 15 deletions(-)

--- a/drivers/staging/rtl8192du/os_dep/ioctl_linux.c
+++ b/drivers/staging/rtl8192du/os_dep/ioctl_linux.c
@@ -453,11 +453,8 @@ static char *translate_scan(struct rtw_a
 
 	/* Add quality statistics */
 	iwe.cmd = IWEVQUAL;
-	iwe.u.qual.updated = IW_QUAL_QUAL_UPDATED | IW_QUAL_LEVEL_UPDATED | IW_QUAL_NOISE_INVALID
-	#ifdef CONFIG_SIGNAL_DISPLAY_DBM
-		| IW_QUAL_DBM
-	#endif
-	;
+	iwe.u.qual.updated = IW_QUAL_QUAL_UPDATED | IW_QUAL_LEVEL_UPDATED |
+			     IW_QUAL_NOISE_INVALID | IW_QUAL_DBM;
 
 	if (check_fwstate(pmlmepriv, _FW_LINKED) == true &&
 	    is_same_network(&pmlmepriv->cur_network.network, &pnetwork->network)) {
@@ -468,11 +465,7 @@ static char *translate_scan(struct rtw_a
 		sq = pnetwork->network.PhyInfo.SignalQuality;
 	}
 
-	#ifdef CONFIG_SIGNAL_DISPLAY_DBM
 	iwe.u.qual.level = (u8) translate_percentage_to_dbm(ss);/* dbm */
-	#else
-	iwe.u.qual.level = (u8)ss;/*  */
-	#endif
 
 	iwe.u.qual.qual = (u8)sq;   /*  signal quality */
 
@@ -7264,11 +7257,7 @@ static struct iw_statistics *rtw_get_wir
 		piwstats->qual.level = 0;
 		piwstats->qual.noise = 0;
 	} else {
-		#ifdef CONFIG_SIGNAL_DISPLAY_DBM
 		tmp_level = translate_percentage_to_dbm(padapter->recvpriv.signal_strength);
-		#else
-		tmp_level = padapter->recvpriv.signal_strength;
-		#endif
 
 		tmp_qual = padapter->recvpriv.signal_qual;
 		tmp_noise = padapter->recvpriv.noise;
@@ -7279,9 +7268,7 @@ static struct iw_statistics *rtw_get_wir
 	}
 	piwstats->qual.updated = IW_QUAL_ALL_UPDATED ;/* IW_QUAL_DBM; */
 
-	#ifdef CONFIG_SIGNAL_DISPLAY_DBM
 	piwstats->qual.updated = piwstats->qual.updated | IW_QUAL_DBM;
-	#endif
 
 	return &padapter->iwstats;
 }
