From 582091d193f06816b0ce15fee0e8348cfb3c37b6 Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Sat, 15 Mar 2014 20:08:31 -0500
Subject: [PATCH 313/390] Remove CONFIG_HW_PWRP_DETECTION

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 Makefile          | 5 -----
 os_dep/os_intfs.c | 7 -------
 2 files changed, 12 deletions(-)

--- a/drivers/staging/rtl8192du/Makefile
+++ b/drivers/staging/rtl8192du/Makefile
@@ -22,7 +22,6 @@ CONFIG_AUTOCFG_CP = n
 
 CONFIG_POWER_SAVING = y
 CONFIG_USB_AUTOSUSPEND = n
-CONFIG_HW_PWRP_DETECTION = n
 CONFIG_BT_COEXISTENCE = n
 CONFIG_WAKE_ON_WLAN = n
 
@@ -83,10 +82,6 @@ ifeq ($(CONFIG_POWER_SAVING), y)
 EXTRA_CFLAGS += -DCONFIG_POWER_SAVING
 endif
 
-ifeq ($(CONFIG_HW_PWRP_DETECTION), y)
-EXTRA_CFLAGS += -DCONFIG_HW_PWRP_DETECTION
-endif
-
 ifeq ($(CONFIG_BT_COEXISTENCE), y)
 EXTRA_CFLAGS += -DCONFIG_BT_COEXISTENCE
 endif
--- a/drivers/staging/rtl8192du/os_dep/os_intfs.c
+++ b/drivers/staging/rtl8192du/os_dep/os_intfs.c
@@ -116,12 +116,6 @@ static int rtw_enusbss = 0;/* 0:disable,
 
 static int rtw_hwpdn_mode = 2;/* 0:disable, 1:enable, 2: by EFUSE config */
 
-#ifdef CONFIG_HW_PWRP_DETECTION
-int rtw_hwpwrp_detect = 1;
-#else
-static int rtw_hwpwrp_detect = 0; /* HW power  ping detect 0:disable , 1:enable */
-#endif
-
 static int rtw_hw_wps_pbc = 1;
 
 int rtw_mc2u_disable = 0;
@@ -171,7 +165,6 @@ module_param(rtw_antdiv_cfg, int, 0644);
 
 module_param(rtw_enusbss, int, 0644);
 module_param(rtw_hwpdn_mode, int, 0644);
-module_param(rtw_hwpwrp_detect, int, 0644);
 
 module_param(rtw_hw_wps_pbc, int, 0644);
 
