From 1b8956af44d92c7b620e927a0d82704562db3aba Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Sat, 15 Mar 2014 20:10:52 -0500
Subject: [PATCH 314/390] Remove undefined CONFIG_DEBUG_RTW_WX_SET_PRIV

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 os_dep/ioctl_linux.c | 34 ++--------------------------------
 1 file changed, 2 insertions(+), 32 deletions(-)

--- a/drivers/staging/rtl8192du/os_dep/ioctl_linux.c
+++ b/drivers/staging/rtl8192du/os_dep/ioctl_linux.c
@@ -6266,17 +6266,12 @@ static int rtw_wx_set_priv(struct net_de
 				union iwreq_data *awrq,
 				char *extra)
 {
-
-#ifdef CONFIG_DEBUG_RTW_WX_SET_PRIV
-	char *ext_dbg;
-#endif
-
+	struct rtw_adapter *padapter = (struct rtw_adapter *)rtw_netdev_priv(dev);
+	struct iw_point *dwrq = (struct iw_point*)awrq;
 	int ret = 0;
 	int len = 0;
 	char *ext;
 	int i;
-	struct rtw_adapter *padapter = (struct rtw_adapter *)rtw_netdev_priv(dev);
-	struct iw_point *dwrq = (struct iw_point*)awrq;
 
 	if (dwrq->length == 0)
 		return -EFAULT;
@@ -6289,15 +6284,6 @@ static int rtw_wx_set_priv(struct net_de
 		return -EFAULT;
 	}
 
-	#ifdef CONFIG_DEBUG_RTW_WX_SET_PRIV
-	if (!(ext_dbg = rtw_vmalloc(len))) {
-		vfree(ext);
-		return -ENOMEM;
-	}
-
-	memcpy(ext_dbg, ext, len);
-	#endif
-
 	/* added for wps2.0 @20110524 */
 	if (dwrq->flags == 0x8766 && len > 8) {
 		u32 cp_sz;
@@ -6389,33 +6375,17 @@ static int rtw_wx_set_priv(struct net_de
 		sprintf(ext, "OK");
 		break; }
 	default :
-		#ifdef  CONFIG_DEBUG_RTW_WX_SET_PRIV
-		DBG_8192D("%s: %s unknowned req =%s\n", __func__,
-			dev->name, ext_dbg);
-		#endif
-
 		sprintf(ext, "OK");
 	}
 
 	if (copy_to_user(dwrq->pointer, ext, min(dwrq->length, (u16)(strlen(ext)+1))))
 		ret = -EFAULT;
 
-	#ifdef CONFIG_DEBUG_RTW_WX_SET_PRIV
-	DBG_8192D("%s: %s req =%s rep =%s dwrq->length =%d, strlen(ext)+1 =%d\n", __func__,
-		dev->name, ext_dbg , ext, dwrq->length, (u16)(strlen(ext)+1));
-	#endif
 #endif /* end of CONFIG_ANDROID */
 
 FREE_EXT:
 
 	vfree(ext);
-	#ifdef CONFIG_DEBUG_RTW_WX_SET_PRIV
-	vfree(ext_dbg);
-	#endif
-
-	/* DBG_8192D("rtw_wx_set_priv: (SIOCSIWPRIV) %s ret =%d\n", */
-	/*		dev->name, ret); */
-
 	return ret;
 }
 
