From ced11bf624af9ded8becbd4bb2e6d4e2452c0f1b Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Sat, 15 Mar 2014 20:28:21 -0500
Subject: [PATCH 316/390] Remove unselected CONFIG_USB_AUTOSUSPEND

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 Makefile          | 6 ------
 os_dep/os_intfs.c | 7 -------
 2 files changed, 13 deletions(-)

--- a/drivers/staging/rtl8192du/Makefile
+++ b/drivers/staging/rtl8192du/Makefile
@@ -21,7 +21,6 @@ EXTRA_CFLAGS += -I$(src)/include
 CONFIG_AUTOCFG_CP = n
 
 CONFIG_POWER_SAVING = y
-CONFIG_USB_AUTOSUSPEND = n
 CONFIG_BT_COEXISTENCE = n
 CONFIG_WAKE_ON_WLAN = n
 
@@ -73,11 +72,6 @@ ifeq ($(CONFIG_AUTOCFG_CP), y)
 $(shell cp $(TopDIR)/autoconf_$(RTL871X)_$(HCI_NAME)_linux.h $(TopDIR)/include/autoconf.h)
 endif
 
-
-ifeq ($(CONFIG_USB_AUTOSUSPEND), y)
-EXTRA_CFLAGS += -DCONFIG_USB_AUTOSUSPEND
-endif
-
 ifeq ($(CONFIG_POWER_SAVING), y)
 EXTRA_CFLAGS += -DCONFIG_POWER_SAVING
 endif
--- a/drivers/staging/rtl8192du/os_dep/os_intfs.c
+++ b/drivers/staging/rtl8192du/os_dep/os_intfs.c
@@ -108,12 +108,6 @@ static int rtw_AcceptAddbaReq = true;/*
 
 static int  rtw_antdiv_cfg = 2; /*  0:OFF , 1:ON, 2:decide by Efuse config */
 
-#ifdef CONFIG_USB_AUTOSUSPEND
-int rtw_enusbss = 1;/* 0:disable, 1:enable */
-#else
-static int rtw_enusbss = 0;/* 0:disable, 1:enable */
-#endif
-
 static int rtw_hwpdn_mode = 2;/* 0:disable, 1:enable, 2: by EFUSE config */
 
 static int rtw_hw_wps_pbc = 1;
@@ -163,7 +157,6 @@ module_param(rtw_special_rf_path, int, 0
 
 module_param(rtw_antdiv_cfg, int, 0644);
 
-module_param(rtw_enusbss, int, 0644);
 module_param(rtw_hwpdn_mode, int, 0644);
 
 module_param(rtw_hw_wps_pbc, int, 0644);
