From 195e95332250318c820dbc611862eb6c36a0c79f Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Sat, 15 Mar 2014 20:48:47 -0500
Subject: [PATCH 321/390] Remove unselected RTL8192CU_ASIC_VERIFICATION

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 hal/usb_halinit.c  | 20 --------------------
 include/autoconf.h |  2 --
 2 files changed, 22 deletions(-)

--- a/drivers/staging/rtl8192du/hal/usb_halinit.c
+++ b/drivers/staging/rtl8192du/hal/usb_halinit.c
@@ -1994,26 +1994,6 @@ _DisableRF_AFE(
 	/* disable RF/ AFE AD/DA */
 	value8 = APSDOFF;
 	rtw_write8(adapter, REG_APSD_CTRL, value8);
-
-#if (RTL8192CU_ASIC_VERIFICATION)
-
-	do
-	{
-		if (rtw_read8(adapter, REG_APSD_CTRL) & APSDOFF_STATUS) {
-			/* RT_TRACE(COMP_INIT, DBG_LOUD, ("Disable RF, AFE, AD, DA Done!\n")); */
-			break;
-		}
-
-		if (pollingCount++ > POLLING_READY_TIMEOUT_COUNT) {
-			/* RT_TRACE(COMP_INIT, DBG_SERIOUS, ("Failed to polling APSDOFF_STATUS done!\n")); */
-			return _FAIL;
-		}
-
-	}while (true);
-
-#endif
-
-	/* RT_TRACE(COMP_INIT, DBG_LOUD, ("Disable RF, AFE, AD, DA.\n")); */
 	return rtStatus;
 }
 
--- a/drivers/staging/rtl8192du/include/autoconf.h
+++ b/drivers/staging/rtl8192du/include/autoconf.h
@@ -64,8 +64,6 @@
 
 #define CONFIG_OUT_EP_WIFI_MODE	0
 
-#define RTL8192CU_ASIC_VERIFICATION	0	/*  For ASIC verification. */
-
 #define RTL8191C_FPGA_NETWORKTYPE_ADHOC 0
 
 #define ANTENNA_SELECTION_STATIC_SETTING 0
