From 23b6c286a19324c0bf83812366d950b22456b4fe Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Sat, 15 Mar 2014 21:19:45 -0500
Subject: [PATCH 323/390] Fix the message when firmware is missing

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 hal/rtl8192d_hal_init.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

--- a/drivers/staging/rtl8192du/hal/rtl8192d_hal_init.c
+++ b/drivers/staging/rtl8192du/hal/rtl8192d_hal_init.c
@@ -281,9 +281,7 @@ static bool get_fw_from_file(struct rtw_
 	const char *fw_name = "rtlwifi/rtl8192dufw.bin";
 #endif /* CONFIG_WOWLAN */
 
-	if (request_firmware(&fw, fw_name, device))
-		return false;
-	if (!fw) {
+	if (request_firmware(&fw, fw_name, device) || !fw) {
 		pr_err("Firmware %s not available\n", fw_name);
 		return false;
 	}
