From 1a2ea77171ef084cdbda0587adf6f3c8469183ec Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Sat, 15 Mar 2014 21:28:03 -0500
Subject: [PATCH 325/390] Remove undefined CONFIG_CHECK_AC_LIFETIME

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 hal/usb_halinit.c | 6 ------
 1 file changed, 6 deletions(-)

--- a/drivers/staging/rtl8192du/hal/usb_halinit.c
+++ b/drivers/staging/rtl8192du/hal/usb_halinit.c
@@ -1623,12 +1623,6 @@ static u32 rtl8192du_hal_init(struct rtw
 	_InitBeaconMaxError(padapter, true);
 
 #if defined(CONFIG_CONCURRENT_MODE)
-
-#ifdef CONFIG_CHECK_AC_LIFETIME
-	/*  Enable lifetime check for the four ACs */
-	rtw_write8(padapter, REG_LIFETIME_EN, 0x0F);
-#endif	/*  CONFIG_CHECK_AC_LIFETIME */
-
 	rtw_write16(padapter, REG_PKT_VO_VI_LIFE_TIME, 0x0400);	/*  unit: 256us. 256ms */
 	rtw_write16(padapter, REG_PKT_BE_BK_LIFE_TIME, 0x0400);	/*  unit: 256us. 256ms */
 #endif	/*  CONFIG_CONCURRENT_MODE */
