From ad2e13d4bb41078455b809e0868051a1b127a8a3 Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Sat, 15 Mar 2014 22:12:37 -0500
Subject: [PATCH 334/390] Remove undefined DBG_XMIT_ACK

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 core/rtw_mlme_ext.c | 10 ----------
 1 file changed, 10 deletions(-)

--- a/drivers/staging/rtl8192du/core/rtw_mlme_ext.c
+++ b/drivers/staging/rtl8192du/core/rtw_mlme_ext.c
@@ -4720,9 +4720,7 @@ int issue_probereq_p2p_ex(struct rtw_ada
 
 	if (ret != _FAIL) {
 		ret = _SUCCESS;
-#ifndef DBG_XMIT_ACK
 		goto exit;
-#endif
 	}
 
 	if (try_cnt && wait_ms) {
@@ -6377,9 +6375,7 @@ int issue_probereq_ex(struct rtw_adapter
 
 	if (ret != _FAIL) {
 		ret = _SUCCESS;
-#ifndef DBG_XMIT_ACK
 		goto exit;
-#endif
 	}
 
 	if (try_cnt && wait_ms) {
@@ -7319,9 +7315,7 @@ int issue_nulldata(struct rtw_adapter *a
 
 	if (ret != _FAIL) {
 		ret = _SUCCESS;
-#ifndef DBG_XMIT_ACK
 		goto exit;
-#endif
 	}
 
 	if (try_cnt && wait_ms) {
@@ -7454,9 +7448,7 @@ int issue_qos_nulldata(struct rtw_adapte
 
 	if (ret != _FAIL) {
 		ret = _SUCCESS;
-#ifndef DBG_XMIT_ACK
 		goto exit;
-#endif
 	}
 
 	if (try_cnt && wait_ms) {
@@ -7580,9 +7572,7 @@ int issue_deauth_ex(struct rtw_adapter *
 
 	if (ret != _FAIL) {
 		ret = _SUCCESS;
-#ifndef DBG_XMIT_ACK
 		goto exit;
-#endif
 	}
 
 	if (try_cnt && wait_ms) {
