From 10d3a88b652f18771d907543a1d774aae7be2bba Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Sat, 15 Mar 2014 22:39:04 -0500
Subject: [PATCH 339/390] Remove undefined USE_WORKITEM

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 hal/rtl8192d_phycfg.c | 16 +++-------------
 1 file changed, 3 insertions(+), 13 deletions(-)

--- a/drivers/staging/rtl8192du/hal/rtl8192d_phycfg.c
+++ b/drivers/staging/rtl8192du/hal/rtl8192d_phycfg.c
@@ -1901,15 +1901,9 @@ PHY_SetBWMode8192D(
 #endif
 
 	if ((!adapter->bDriverStopped) && (!adapter->bSurpriseRemoved))
-	{
-#ifndef USE_WORKITEM
-	_PHY_SetBWMode92D(adapter);
-#endif
-	}
+		_PHY_SetBWMode92D(adapter);
 	else
-	{
 		pHalData->CurrentChannelBW = tmpBW;
-	}
 }
 
 /*******************************************************************
@@ -2719,18 +2713,14 @@ PHY_SwChnl8192D(	/*  Call after initiali
 	pHalData->CurrentChannel = channel;
 
 #ifdef CONFIG_DUALMAC_CONCURRENT
-	if ((Buddyadapter != NULL) && (pHalData->bSlaveOfDMSP))
-	{
+	if ((Buddyadapter != NULL) && (pHalData->bSlaveOfDMSP)) {
 		DBG_8192D("PHY_SwChnl8192D():slave return when slave\n");
 		return;
 	}
 #endif
 
-	if ((!adapter->bDriverStopped) && (!adapter->bSurpriseRemoved))
-	{
-#ifndef USE_WORKITEM
+	if ((!adapter->bDriverStopped) && (!adapter->bSurpriseRemoved)) {
 		_PHY_SwChnl8192D(adapter, channel);
-#endif
 		if (!bResult)
 				pHalData->CurrentChannel = tmpchannel;
 
