From 239e2b715ee956f1054b874b7861a70ae04ac953 Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Sat, 15 Mar 2014 22:53:03 -0500
Subject: [PATCH 342/390] Remove undefined CONFIG_SINGLE_XMIT_BUF

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 include/rtw_xmit.h | 8 --------
 1 file changed, 8 deletions(-)

--- a/drivers/staging/rtl8192du/include/rtw_xmit.h
+++ b/drivers/staging/rtl8192du/include/rtw_xmit.h
@@ -20,21 +20,13 @@
 #include <osdep_service.h>
 #include <drv_types.h>
 #define MAX_XMITBUF_SZ		(20480)	/*  20k */
-#ifdef CONFIG_SINGLE_XMIT_BUF
-#define NR_XMITBUFF		(1)
-#else
 #define NR_XMITBUFF		(4)
-#endif /* CONFIG_SINGLE_XMIT_BUF */
 
 #define XMITBUF_ALIGN_SZ	512
 
 /*  xmit extension buff defination */
 #define MAX_XMIT_EXTBUF_SZ	(1536)
-#ifdef CONFIG_SINGLE_XMIT_BUF
-#define NR_XMIT_EXTBUFF		(1)
-#else
 #define NR_XMIT_EXTBUFF		(32)
-#endif /* CONFIG_SINGLE_XMIT_BUF */
 
 #define MAX_NUMBLKS		(1)
 
