From 6b3f26e3e750a5feca45fae0554b1367383f45b9 Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Sat, 15 Mar 2014 22:58:12 -0500
Subject: [PATCH 344/390] Remove undefined AUTO_CONFIG

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 include/rtw_mlme.h | 4 ----
 1 file changed, 4 deletions(-)

--- a/drivers/staging/rtl8192du/include/rtw_mlme.h
+++ b/drivers/staging/rtl8192du/include/rtw_mlme.h
@@ -58,10 +58,6 @@
 #define	WIFI_WDS					0x00001000
 #define	WIFI_WDS_RX_BEACON		0x00002000		/*  already rx WDS AP beacon */
 #endif
-#ifdef AUTO_CONFIG
-#define	WIFI_AUTOCONF				0x00004000
-#define	WIFI_AUTOCONF_IND		0x00008000
-#endif
 
 #define	WIFI_MP_STATE							0x00010000
 #define	WIFI_MP_CTX_BACKGROUND				0x00020000	/*  in continous tx background */
