From 7d8fc3eb6ecbc1bb11ae825bef95ce79287c72c8 Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Sat, 15 Mar 2014 23:10:42 -0500
Subject: [PATCH 347/390] Remove undefined INTF_DATA

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 include/drv_types.h | 6 ------
 1 file changed, 6 deletions(-)

--- a/drivers/staging/rtl8192du/include/drv_types.h
+++ b/drivers/staging/rtl8192du/include/drv_types.h
@@ -200,12 +200,6 @@ struct dvobj_priv {
 	u8	DualMacMode;
 	u8	irq_alloc;
 
-/*-------- below is for SDIO INTERFACE --------*/
-
-#ifdef INTF_DATA
-	INTF_DATA intf_data;
-#endif
-
 /*-------- below is for USB INTERFACE --------*/
 
 	u8	nr_endpoint;
