From 70db0f258ab3f8be6fb961810b4f3594ca0efcf7 Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Sat, 15 Mar 2014 23:13:09 -0500
Subject: [PATCH 348/390] Remove undefined GREEN_HILL

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 include/wifi.h | 4 ----
 1 file changed, 4 deletions(-)

--- a/drivers/staging/rtl8192du/include/wifi.h
+++ b/drivers/staging/rtl8192du/include/wifi.h
@@ -46,10 +46,6 @@
 
 #define P80211CAPTURE_VERSION	0x80211001
 
-#ifdef GREEN_HILL
-#pragma pack(1)
-#endif
-
 enum WIFI_FRAME_TYPE {
 	WIFI_MGT_TYPE  =	(0),
 	WIFI_CTRL_TYPE =	(BIT(2)),
