From 1621aab3f160fa6b483a58e9806f84f1a26b4c83 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Mon, 17 Mar 2014 06:29:53 +0100
Subject: [PATCH 368/390] rtl8192du: drop unused function update_bcn_wpa_ie()

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 core/rtw_ap.c | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

--- a/drivers/staging/rtl8192du/core/rtw_ap.c
+++ b/drivers/staging/rtl8192du/core/rtw_ap.c
@@ -1481,11 +1481,6 @@ static void update_bcn_erpinfo_ie(struct
 	}
 }
 
-static void update_bcn_wpa_ie(struct rtw_adapter *padapter)
-{
-	DBG_8192D("%s\n", __func__);
-}
-
 static void update_bcn_wmm_ie(struct rtw_adapter *padapter)
 {
 	DBG_8192D("%s\n", __func__);
@@ -1553,7 +1548,7 @@ static void update_bcn_vendor_spec_ie(st
 	DBG_8192D("%s\n", __func__);
 
 	if (_rtw_memcmp(RTW_WPA_OUI, oui, 4))
-		update_bcn_wpa_ie(padapter);
+		return;
 	else if (_rtw_memcmp(WMM_OUI, oui, 4))
 		update_bcn_wmm_ie(padapter);
 	else if (_rtw_memcmp(WPS_OUI, oui, 4))
