From 3eb1517138be07de360f2a714077dfd6eaf177a5 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Mon, 17 Mar 2014 06:29:55 +0100
Subject: [PATCH 369/390] rtl8192du: drop unused function update_bcn_p2p_ie()

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 core/rtw_ap.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

--- a/drivers/staging/rtl8192du/core/rtw_ap.c
+++ b/drivers/staging/rtl8192du/core/rtw_ap.c
@@ -1539,10 +1539,6 @@ static void update_bcn_wps_ie(struct rtw
 	kfree(pbackup_remainder_ie);
 }
 
-static void update_bcn_p2p_ie(struct rtw_adapter *padapter)
-{
-}
-
 static void update_bcn_vendor_spec_ie(struct rtw_adapter *padapter, u8 *oui)
 {
 	DBG_8192D("%s\n", __func__);
@@ -1554,7 +1550,7 @@ static void update_bcn_vendor_spec_ie(st
 	else if (_rtw_memcmp(WPS_OUI, oui, 4))
 		update_bcn_wps_ie(padapter);
 	else if (_rtw_memcmp(P2P_OUI, oui, 4))
-		update_bcn_p2p_ie(padapter);
+		return;
 	else
 		DBG_8192D("unknown OUI type!\n");
 }
