From b189cb637cbba7258c661b8e28e62e8d24227e88 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Mon, 17 Mar 2014 06:29:56 +0100
Subject: [PATCH 370/390] rtl8192du: simplify update_bcn_vendor_spec_ie() after
 preceding changes

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 core/rtw_ap.c | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

--- a/drivers/staging/rtl8192du/core/rtw_ap.c
+++ b/drivers/staging/rtl8192du/core/rtw_ap.c
@@ -1543,13 +1543,11 @@ static void update_bcn_vendor_spec_ie(st
 {
 	DBG_8192D("%s\n", __func__);
 
-	if (_rtw_memcmp(RTW_WPA_OUI, oui, 4))
-		return;
-	else if (_rtw_memcmp(WMM_OUI, oui, 4))
-		update_bcn_wmm_ie(padapter);
-	else if (_rtw_memcmp(WPS_OUI, oui, 4))
+	if (_rtw_memcmp(WPS_OUI, oui, 4))
 		update_bcn_wps_ie(padapter);
-	else if (_rtw_memcmp(P2P_OUI, oui, 4))
+	else if ((_rtw_memcmp(RTW_WPA_OUI, oui, 4)) ||
+		 (_rtw_memcmp(WMM_OUI, oui, 4)) ||
+		 (_rtw_memcmp(P2P_OUI, oui, 4)))
 		return;
 	else
 		DBG_8192D("unknown OUI type!\n");
