From 2bfc0730b1e947adc20d6970a719f04867de3cea Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Mon, 17 Mar 2014 06:30:03 +0100
Subject: [PATCH 376/390] rtl8192du: drop unused function fire_write_MAC_cmd()

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 core/rtw_wlan_util.c | 4 ----
 1 file changed, 4 deletions(-)

--- a/drivers/staging/rtl8192du/core/rtw_wlan_util.c
+++ b/drivers/staging/rtl8192du/core/rtw_wlan_util.c
@@ -1435,10 +1435,6 @@ void update_wireless_mode(struct rtw_ada
 	}
 }
 
-static void fire_write_MAC_cmd(struct rtw_adapter *padapter, unsigned int addr, unsigned int value)
-{
-}
-
 void update_bmc_sta_support_rate(struct rtw_adapter *padapter, u32 mac_id)
 {
 	struct mlme_ext_priv	*pmlmeext = &(padapter->mlmeextpriv);
