From c02b946bae26134043006d1911c0c8d91df445b5 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Mon, 17 Mar 2014 06:30:04 +0100
Subject: [PATCH 377/390] rtl8192du: drop unused function dm_CheckRfCtrlGPIO()

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 hal/rtl8192d_dm.c | 23 -----------------------
 1 file changed, 23 deletions(-)

--- a/drivers/staging/rtl8192du/hal/rtl8192d_dm.c
+++ b/drivers/staging/rtl8192du/hal/rtl8192d_dm.c
@@ -2040,29 +2040,6 @@ rtl8192d_dm_CheckTXPowerTracking(
 }
 
 /*-----------------------------------------------------------------------------
- * Function:	dm_CheckRfCtrlGPIO()
- *
- * Overview:	Copy 8187B template for 9xseries.
- *
- * Input:		NONE
- *
- * Output:		NONE
- *
- * Return:		NONE
- *
- * Revised History:
- *	When		Who		Remark
- *	01/10/2008	MHC		Create Version 0.
- *
- *---------------------------------------------------------------------------*/
-static void
-dm_CheckRfCtrlGPIO(
-	struct rtw_adapter *	adapter
-	)
-{
-}	/* dm_CheckRfCtrlGPIO */
-
-/*-----------------------------------------------------------------------------
  * Function:	dm_CheckPbcGPIO()
  *
  * Overview:	Check if PBC button is pressed.
