From b6b9cbdb60101028dec593e36a15c1a3e10e0df5 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Mon, 17 Mar 2014 06:30:05 +0100
Subject: [PATCH 378/390] rtl8192du: drop unused function
 dm_RefreshRateAdaptiveMask()

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 hal/rtl8192d_dm.c | 25 -------------------------
 1 file changed, 25 deletions(-)

--- a/drivers/staging/rtl8192du/hal/rtl8192d_dm.c
+++ b/drivers/staging/rtl8192du/hal/rtl8192d_dm.c
@@ -2125,26 +2125,6 @@ static void dm_InitRateAdaptiveMask(stru
 		pdmpriv->bUseRAMask = false;
 }
 
-/*-----------------------------------------------------------------------------
- * Function:	dm_RefreshRateAdaptiveMask()
- *
- * Overview:	Update rate table mask according to rssi
- *
- * Input:		NONE
- *
- * Output:		NONE
- *
- * Return:		NONE
- *
- * Revised History:
- *	When		Who		Remark
- *	05/27/2009	hpfan	Create Version 0.
- *
- *---------------------------------------------------------------------------*/
-static void
-dm_RefreshRateAdaptiveMask(struct rtw_adapter *	adapter)
-{
-}
 static void
 dm_CheckProtection(
 	struct rtw_adapter *	adapter
@@ -2415,11 +2395,6 @@ rtl8192d_HalDmWatchDog(
 #endif
 			rtl8192d_dm_CheckTXPowerTracking(adapter);
 
-		/*  */
-		/*  Rate Adaptive by Rx Signal Strength mechanism. */
-		/*  */
-		dm_RefreshRateAdaptiveMask(adapter);
-
 		/*  EDCA turbo */
 		/* update the EDCA paramter according to the Tx/RX mode */
 		dm_CheckEdcaTurbo(adapter);
