From b6d9e43fb2a511ac8db4f38c6a25e588cad8c64a Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Mon, 17 Mar 2014 06:30:06 +0100
Subject: [PATCH 379/390] rtl8192du: drop unused function dm_CheckProtection()

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 hal/rtl8192d_dm.c | 12 ------------
 1 file changed, 12 deletions(-)

--- a/drivers/staging/rtl8192du/hal/rtl8192d_dm.c
+++ b/drivers/staging/rtl8192du/hal/rtl8192d_dm.c
@@ -2126,13 +2126,6 @@ static void dm_InitRateAdaptiveMask(stru
 }
 
 static void
-dm_CheckProtection(
-	struct rtw_adapter *	adapter
-	)
-{
-}
-
-static void
 dm_CheckStatistics(
 	struct rtw_adapter *	adapter
 	)
@@ -2400,11 +2393,6 @@ rtl8192d_HalDmWatchDog(
 		dm_CheckEdcaTurbo(adapter);
 
 		/*  */
-		/*  Dynamically switch RTS/CTS protection. */
-		/*  */
-		/* dm_CheckProtection(adapter); */
-
-		/*  */
 		/* Dynamic BB Power Saving Mechanism */
 		/* vivi, 20101014, to pass DTM item: softap_excludeunencrypted_ext.htm */
 		/* temporarily disable it advised for performance test by yn,2010-11-03. */
