From 92d2dbbfd581d8615955686a7f5827be326d458c Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Mon, 17 Mar 2014 06:30:07 +0100
Subject: [PATCH 380/390] rtl8192du: drop unused function dm_CheckStatistics()

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 hal/rtl8192d_dm.c | 12 ------------
 1 file changed, 12 deletions(-)

--- a/drivers/staging/rtl8192du/hal/rtl8192d_dm.c
+++ b/drivers/staging/rtl8192du/hal/rtl8192d_dm.c
@@ -2125,13 +2125,6 @@ static void dm_InitRateAdaptiveMask(stru
 		pdmpriv->bUseRAMask = false;
 }
 
-static void
-dm_CheckStatistics(
-	struct rtw_adapter *	adapter
-	)
-{
-}
-
 /*  */
 /*  Initialize GPIO setting registers */
 /*  */
@@ -2332,11 +2325,6 @@ rtl8192d_HalDmWatchDog(
 	if ((hw_init_completed == true) && ((!bFwCurrentInPSMode) &&
 	    bFwPSAwake)) {
 		/*  */
-		/*  Calculate Tx/Rx statistics. */
-		/*  */
-		dm_CheckStatistics(adapter);
-
-		/*  */
 		/*  For PWDB monitor and record some value for later use. */
 		/*  */
 		PWDB_Monitor(adapter);
