From 0f26bdbd02918db218d1f4af129bab6988c731b8 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Mon, 17 Mar 2014 06:30:08 +0100
Subject: [PATCH 381/390] rtl8192du: drop unused functions
 phy_FwRFSerialWrite()

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 hal/rtl8192d_phycfg.c | 25 -------------------------
 1 file changed, 25 deletions(-)

--- a/drivers/staging/rtl8192du/hal/rtl8192d_phycfg.c
+++ b/drivers/staging/rtl8192du/hal/rtl8192d_phycfg.c
@@ -266,31 +266,6 @@ phy_FwRFSerialRead(
 	return	(retValue);
 }	/* phy_FwRFSerialRead */
 
-/*-----------------------------------------------------------------------------
- * Function:	phy_FwRFSerialWrite()
- *
- * Overview:	We support firmware to execute RF-R/W.
- *
- * Input:		NONE
- *
- * Output:		NONE
- *
- * Return:		NONE
- *
- * Revised History:
- *	When		Who		Remark
- *	01/21/2008	MHC		Create Version 0.
- *
- *---------------------------------------------------------------------------*/
-static	void
-phy_FwRFSerialWrite(
-	struct rtw_adapter *			adapter,
-	enum RF_RADIO_PATH_E	eRFPath,
-	u32				Offset,
-	u32				Data	)
-{
-}
-
 /**
 * Function:	phy_RFSerialRead
 *
