From 85102a76f235af3961a44aa0706e5d67a74531b2 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Mon, 17 Mar 2014 06:30:12 +0100
Subject: [PATCH 385/390] rtl8192du: drop unused function
 phy_SwChnlStepByStep()

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 hal/rtl8192d_phycfg.c | 12 ------------
 1 file changed, 12 deletions(-)

--- a/drivers/staging/rtl8192du/hal/rtl8192d_phycfg.c
+++ b/drivers/staging/rtl8192du/hal/rtl8192d_phycfg.c
@@ -2608,18 +2608,6 @@ PHY_SwChnl8192D(	/*  Call after initiali
 }
 
 static	bool
-phy_SwChnlStepByStep(
-	struct rtw_adapter *	adapter,
-	u8		channel,
-	u8		*stage,
-	u8		*step,
-	u32		*delay
-	)
-{
-	return true;
-}
-
-static	bool
 phy_SetSwChnlCmdArray(
 	struct sw_chnl_cmd *cmdtable,
 	u32			cmdtableidx,
