From 909d0caf52579aedd50bdfe95af722a33f108ad7 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Mon, 17 Mar 2014 06:30:13 +0100
Subject: [PATCH 386/390] rtl8192du: drop unused function phy_FinishSwChnlNow()

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 hal/rtl8192d_phycfg.c | 11 -----------
 1 file changed, 11 deletions(-)

--- a/drivers/staging/rtl8192du/hal/rtl8192d_phycfg.c
+++ b/drivers/staging/rtl8192du/hal/rtl8192d_phycfg.c
@@ -2634,14 +2634,6 @@ phy_SetSwChnlCmdArray(
 	return true;
 }
 
-static	void
-phy_FinishSwChnlNow(	/*  We should not call this function directly */
-		struct rtw_adapter *	adapter,
-		u8		channel
-		)
-{
-}
-
 /*  */
 /*  Description: */
 /*	Switch channel synchronously. Called by SwChnlByDelayHandler. */
@@ -2673,9 +2665,6 @@ PHY_SwChnlPhy8192D(	/*  Only called duri
 		channel = 1;
 
 	pHalData->CurrentChannel = channel;
-
-	phy_FinishSwChnlNow(adapter, channel);
-
 }
 
 /*  */
