From 52f52200de2eeb123d8556cfe74c9f9431c6e089 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Mon, 17 Mar 2014 06:30:15 +0100
Subject: [PATCH 388/390] rtl8192du: drop unused function
 PHY_CheckIsLegalRfPath8192D()

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 hal/rtl8192d_phycfg.c | 28 ----------------------------
 1 file changed, 28 deletions(-)

--- a/drivers/staging/rtl8192du/hal/rtl8192d_phycfg.c
+++ b/drivers/staging/rtl8192du/hal/rtl8192d_phycfg.c
@@ -2667,34 +2667,6 @@ PHY_SwChnlPhy8192D(	/*  Only called duri
 	pHalData->CurrentChannel = channel;
 }
 
-/*-----------------------------------------------------------------------------
- * Function:	PHYCheckIsLegalRfPath8190Pci()
- *
- * Overview:	Check different RF type to execute legal judgement. If RF Path is illegal
- *			We will return false.
- *
- * Input:		NONE
- *
- * Output:		NONE
- *
- * Return:		NONE
- *
- * Revised History:
- *	When		Who		Remark
- *	11/15/2007	MHC		Create Version 0.
- *
- *---------------------------------------------------------------------------*/
-bool
-PHY_CheckIsLegalRfPath8192D(
-	struct rtw_adapter *	adapter,
-	u32	eRFPath)
-{
-	bool				rtValue = true;
-
-	/*  NOt check RF Path now.! */
-	return	rtValue;
-}	/* PHY_CheckIsLegalRfPath8192D */
-
 /*  */
 /*  */
 /*	IQK */
