From 14011e13f821ab18725d0f3c0bef26f5bd0cfd39 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Sat, 22 Mar 2014 03:02:59 +0100
Subject: [PATCH 401/456] rtl8192du: remove unused function hex2byte_i()

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 os_dep/ioctl_linux.c | 12 ------------
 1 file changed, 12 deletions(-)

--- a/drivers/staging/rtl8192du/os_dep/ioctl_linux.c
+++ b/drivers/staging/rtl8192du/os_dep/ioctl_linux.c
@@ -71,18 +71,6 @@ static int hex2num_i(char c)
 	return -1;
 }
 
-static int hex2byte_i(const char *hex)
-{
-	int a, b;
-	a = hex2num_i(*hex++);
-	if (a < 0)
-		return -1;
-	b = hex2num_i(*hex++);
-	if (b < 0)
-		return -1;
-	return (a << 4) | b;
-}
-
 /**
  * hwaddr_aton - Convert ASCII string to MAC address
  * @txt: MAC address as a string (e.g., "00:11:22:33:44:55")
