From eb5c61fcecc42a1a7047864efa31485636838282 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Sat, 22 Mar 2014 03:03:19 +0100
Subject: [PATCH 421/456] rtl8192du: drop unused function _rtw_vmalloc()

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 os_dep/osdep_service.c | 8 --------
 1 file changed, 8 deletions(-)

--- a/drivers/staging/rtl8192du/os_dep/osdep_service.c
+++ b/drivers/staging/rtl8192du/os_dep/osdep_service.c
@@ -36,14 +36,6 @@ inline int RTW_STATUS_CODE(int error_cod
 	}
 }
 
-inline u8* _rtw_vmalloc(u32 sz)
-{
-	u8	*pbuf;
-
-	pbuf = vmalloc(sz);
-	return pbuf;
-}
-
 int	_rtw_memcmp(void *dst, void *src, u32 sz)
 {
 /* under Linux/GNU/GLibc, the return value of memcmp for two same mem. chunk is 0 */
