From b841be4e2efd297eb977a736078411fcf1468bc0 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Sat, 22 Mar 2014 03:03:40 +0100
Subject: [PATCH 442/456] rtl8192du: simplify _RND4()

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 include/osdep_service.h | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

--- a/drivers/staging/rtl8192du/include/osdep_service.h
+++ b/drivers/staging/rtl8192du/include/osdep_service.h
@@ -163,13 +163,7 @@ void	rtw_usleep_os(int us);
 
 static inline u32 _RND4(u32 sz)
 {
-
-	u32	val;
-
-	val = ((sz >> 2) + ((sz & 3) ? 1: 0)) << 2;
-
-	return val;
-
+	return ((sz >> 2) + ((sz & 3) ? 1: 0)) << 2;
 }
 
 static inline u32 _RND8(u32 sz)
