From 00a04d5402de82feec4485f30d936acba145a569 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Sat, 22 Mar 2014 03:03:41 +0100
Subject: [PATCH 443/456] rtl8192du: simplify _RND8()

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 include/osdep_service.h | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

--- a/drivers/staging/rtl8192du/include/osdep_service.h
+++ b/drivers/staging/rtl8192du/include/osdep_service.h
@@ -168,13 +168,7 @@ static inline u32 _RND4(u32 sz)
 
 static inline u32 _RND8(u32 sz)
 {
-
-	u32	val;
-
-	val = ((sz >> 3) + ((sz & 7) ? 1: 0)) << 3;
-
-	return val;
-
+	return ((sz >> 3) + ((sz & 7) ? 1: 0)) << 3;
 }
 
 static inline u32 _RND128(u32 sz)
