From 36c587e05c8e60e361f5f5e536326dcfd617e9c0 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Sat, 22 Mar 2014 03:03:42 +0100
Subject: [PATCH 444/456] rtl8192du: simplify _RND128()

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 include/osdep_service.h | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

--- a/drivers/staging/rtl8192du/include/osdep_service.h
+++ b/drivers/staging/rtl8192du/include/osdep_service.h
@@ -173,13 +173,7 @@ static inline u32 _RND8(u32 sz)
 
 static inline u32 _RND128(u32 sz)
 {
-
-	u32	val;
-
-	val = ((sz >> 7) + ((sz & 127) ? 1: 0)) << 7;
-
-	return val;
-
+	return ((sz >> 7) + ((sz & 127) ? 1: 0)) << 7;
 }
 
 static inline u32 _RND256(u32 sz)
