From 8bcaa81a79945538bfb48e65906c79c7c57d23c0 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Sat, 22 Mar 2014 03:03:43 +0100
Subject: [PATCH 445/456] rtl8192du: remove unused function _RND256()

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 include/osdep_service.h | 11 -----------
 1 file changed, 11 deletions(-)

--- a/drivers/staging/rtl8192du/include/osdep_service.h
+++ b/drivers/staging/rtl8192du/include/osdep_service.h
@@ -176,17 +176,6 @@ static inline u32 _RND128(u32 sz)
 	return ((sz >> 7) + ((sz & 127) ? 1: 0)) << 7;
 }
 
-static inline u32 _RND256(u32 sz)
-{
-
-	u32	val;
-
-	val = ((sz >> 8) + ((sz & 255) ? 1: 0)) << 8;
-
-	return val;
-
-}
-
 static inline u32 _RND512(u32 sz)
 {
 
