From 63d882b416d0056250905ee28f88a527b8f188e7 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Sat, 22 Mar 2014 03:03:44 +0100
Subject: [PATCH 446/456] rtl8192du: remove unused function _RND512()

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 include/osdep_service.h | 11 -----------
 1 file changed, 11 deletions(-)

--- a/drivers/staging/rtl8192du/include/osdep_service.h
+++ b/drivers/staging/rtl8192du/include/osdep_service.h
@@ -176,17 +176,6 @@ static inline u32 _RND128(u32 sz)
 	return ((sz >> 7) + ((sz & 127) ? 1: 0)) << 7;
 }
 
-static inline u32 _RND512(u32 sz)
-{
-
-	u32	val;
-
-	val = ((sz >> 9) + ((sz & 511) ? 1: 0)) << 9;
-
-	return val;
-
-}
-
 static inline u32 bitshift(u32 bitmask)
 {
 	u32 i;
