From 26ee2d92b6ddaa86976d6a8bec75edb4344390f6 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Sat, 22 Mar 2014 03:03:45 +0100
Subject: [PATCH 447/456] rtl8192du: remove unused function bitshift()

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 include/osdep_service.h | 10 ----------
 1 file changed, 10 deletions(-)

--- a/drivers/staging/rtl8192du/include/osdep_service.h
+++ b/drivers/staging/rtl8192du/include/osdep_service.h
@@ -176,16 +176,6 @@ static inline u32 _RND128(u32 sz)
 	return ((sz >> 7) + ((sz & 127) ? 1: 0)) << 7;
 }
 
-static inline u32 bitshift(u32 bitmask)
-{
-	u32 i;
-
-	for (i = 0; i <= 31; i++)
-		if (((bitmask>>i) &  0x1) == 1) break;
-
-	return i;
-}
-
 /* ifdef __GNUC__ */
 #define STRUCT_PACKED __attribute__ ((packed))
 
