From 7f39eebd4787c7c08e3d619e181cf62e522acb19 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Sat, 22 Mar 2014 03:03:46 +0100
Subject: [PATCH 448/456] rtl8192du: drop unused definition for PATH_LENGTH_MAX

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 include/osdep_service.h | 3 ---
 1 file changed, 3 deletions(-)

--- a/drivers/staging/rtl8192du/include/osdep_service.h
+++ b/drivers/staging/rtl8192du/include/osdep_service.h
@@ -179,9 +179,6 @@ static inline u32 _RND128(u32 sz)
 /* ifdef __GNUC__ */
 #define STRUCT_PACKED __attribute__ ((packed))
 
-/*  limitation of path length */
-	#define PATH_LENGTH_MAX PATH_MAX
-
 struct rtw_netdev_priv_indicator {
 	void *priv;
 	u32 sizeof_priv;
