From 7d7f3010b38250bff0eebc311aebdf1980ce9902 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Sat, 22 Mar 2014 03:03:47 +0100
Subject: [PATCH 449/456] rtl8192du: remove empty function _rtw_free_evt_priv()

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 core/rtw_cmd.c       | 8 --------
 include/mlme_osdep.h | 1 -
 2 files changed, 9 deletions(-)

--- a/drivers/staging/rtl8192du/core/rtw_cmd.c
+++ b/drivers/staging/rtl8192du/core/rtw_cmd.c
@@ -78,13 +78,6 @@ int _rtw_init_evt_priv(struct evt_priv *
 	return res;
 }
 
-void _rtw_free_evt_priv(struct evt_priv *pevtpriv)
-{
-	RT_TRACE(_module_rtl871x_cmd_c_, _drv_info_, ("+_rtw_free_evt_priv\n"));
-
-	RT_TRACE(_module_rtl871x_cmd_c_, _drv_info_, ("-_rtw_free_evt_priv\n"));
-}
-
 void _rtw_free_cmd_priv(struct cmd_priv *pcmdpriv)
 {
 	if (pcmdpriv) {
@@ -159,7 +152,6 @@ u32	rtw_init_evt_priv(struct evt_priv *p
 void rtw_free_evt_priv(struct evt_priv *pevtpriv)
 {
 	RT_TRACE(_module_rtl871x_cmd_c_, _drv_info_, ("rtw_free_evt_priv\n"));
-	_rtw_free_evt_priv(pevtpriv);
 }
 
 void rtw_free_cmd_priv(struct cmd_priv *pcmdpriv)
--- a/drivers/staging/rtl8192du/include/mlme_osdep.h
+++ b/drivers/staging/rtl8192du/include/mlme_osdep.h
@@ -30,7 +30,6 @@ void indicate_wx_scan_complete_event(str
 u8 rtw_do_join(struct rtw_adapter *padapter);
 int _rtw_init_cmd_priv (struct	cmd_priv *pcmdpriv);
 int _rtw_init_evt_priv(struct evt_priv *pevtpriv);
-void _rtw_free_evt_priv (struct	evt_priv *pevtpriv);
 void _rtw_free_cmd_priv (struct	cmd_priv *pcmdpriv);
 int _rtw_enqueue_cmd(struct __queue *queue, struct cmd_obj *obj);
 struct cmd_obj	*_rtw_dequeue_cmd(struct __queue *queue);
