From ed780f7d854b05d1fbf41f3b4b87510e5894b9a6 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Sat, 22 Mar 2014 03:03:50 +0100
Subject: [PATCH 452/456] rtl8192du: remove unused function rtw_cmd_clr_isr()

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 core/rtw_cmd.c    | 5 -----
 include/rtw_cmd.h | 1 -
 2 files changed, 6 deletions(-)

--- a/drivers/staging/rtl8192du/core/rtw_cmd.c
+++ b/drivers/staging/rtl8192du/core/rtw_cmd.c
@@ -210,11 +210,6 @@ struct cmd_obj *rtw_dequeue_cmd(struct c
 	return cmd_obj;
 }
 
-void rtw_cmd_clr_isr(struct cmd_priv *pcmdpriv)
-{
-	pcmdpriv->cmd_done_cnt++;
-}
-
 void rtw_free_cmd_obj(struct cmd_obj *pcmd)
 {
 	if ((pcmd->cmdcode != _JOINBSS_CMD_) &&
--- a/drivers/staging/rtl8192du/include/rtw_cmd.h
+++ b/drivers/staging/rtl8192du/include/rtw_cmd.h
@@ -98,7 +98,6 @@ u32 rtw_init_cmd_priv (struct cmd_priv *
 void rtw_free_cmd_priv (struct cmd_priv *pcmdpriv);
 
 u32 rtw_init_evt_priv (struct evt_priv *pevtpriv);
-void rtw_cmd_clr_isr(struct cmd_priv *pcmdpriv);
 void rtw_evt_notify_isr(struct evt_priv *pevtpriv);
 #ifdef CONFIG_P2P
 u8 p2p_protocol_wk_cmd(struct rtw_adapter*padapter, int intCmdType);
