From 2661c08a378362aface4184cf90336dbe0d92a49 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Sat, 22 Mar 2014 03:03:51 +0100
Subject: [PATCH 453/456] rtl8192du: simplify Efuse_PgPacketWrite()

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 core/rtw_efuse.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

--- a/drivers/staging/rtl8192du/core/rtw_efuse.c
+++ b/drivers/staging/rtl8192du/core/rtw_efuse.c
@@ -329,11 +329,7 @@ int Efuse_PgPacketRead(struct rtw_adapte
 int Efuse_PgPacketWrite(struct rtw_adapter *adapter, u8 offset, u8 word_en,
 			u8 *data, bool test)
 {
-	int ret;
-
-	ret =  adapter->HalFunc.Efuse_PgPacketWrite(adapter, offset, word_en, data, test);
-
-	return ret;
+	return adapter->HalFunc.Efuse_PgPacketWrite(adapter, offset, word_en, data, test);
 }
 
 /*-----------------------------------------------------------------------------
