From 007a970a8f53db3654e990b7bf9939156efc4787 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Sat, 22 Mar 2014 03:03:52 +0100
Subject: [PATCH 454/456] rtl8192du: remove unused function convert_ip_addr()

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 core/rtw_ieee80211.c | 5 -----
 include/usb_osintf.h | 1 -
 2 files changed, 6 deletions(-)

--- a/drivers/staging/rtl8192du/core/rtw_ieee80211.c
+++ b/drivers/staging/rtl8192du/core/rtw_ieee80211.c
@@ -1069,11 +1069,6 @@ u8 key_2char2num(u8 hch, u8 lch)
 	return (key_char2num(hch) << 4) | key_char2num(lch);
 }
 
-u8 convert_ip_addr(u8 hch, u8 mch, u8 lch)
-{
-	return (key_char2num(hch) * 100) + (key_char2num(mch) * 10) + key_char2num(lch);
-}
-
 void rtw_macaddr_cfg(u8 *mac_addr)
 {
 	u8 mac[ETH_ALEN];
--- a/drivers/staging/rtl8192du/include/usb_osintf.h
+++ b/drivers/staging/rtl8192du/include/usb_osintf.h
@@ -45,7 +45,6 @@ void nat25_db_expire(struct rtw_adapter
 u8 str_2char2num(u8 hch, u8 lch);
 u8 str_2char2num(u8 hch, u8 lch);
 u8 key_2char2num(u8 hch, u8 lch);
-u8 convert_ip_addr(u8 hch, u8 mch, u8 lch);
 void process_wmmps_data(struct rtw_adapter *padapter,
 			struct recv_frame_hdr *precv_frame);
 s32 c2h_evt_hdl(struct rtw_adapter *adapter, struct c2h_evt_hdr *c2h_evt,
