From 4ac260606c393f65723a2d1f431d33e1c1e24b7e Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Fri, 25 Apr 2014 14:37:24 -0500
Subject: [PATCH] rtl8192du: Fix some unitialized variable warnings

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 hal/rtl8192d_dm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/staging/rtl8192du/hal/rtl8192d_dm.c
+++ b/drivers/staging/rtl8192du/hal/rtl8192d_dm.c
@@ -1482,7 +1482,7 @@ dm_TXPowerTrackingCallback_ThermalMeter_
 	u32		ThermalValue_AVG = 0;
 	s32		ele_A = 0, ele_D, TempCCk, X, value32;
 	s32		Y, ele_C;
-	s8		OFDM_index[2], CCK_index=0, OFDM_index_old[2], CCK_index_old=0;
+	s8		OFDM_index[2], CCK_index=0, OFDM_index_old[2] = {0}, CCK_index_old=0;
 	s32		i = 0;
 	bool		is2T = IS_92D_SINGLEPHY(pHalData->VersionID);
 	bool		bInteralPA = false;
