From d10219fcae91199284e990cbb1e23b5ec3a81db9 Mon Sep 17 00:00:00 2001
From: Joe Perches <joe@perches.com>
Date: Wed, 24 Mar 2010 22:17:04 -0700
Subject: [PATCH 04/16] Staging: rtl8192u: Hoist assign from if

Signed-off-by: Joe Perches <joe@perches.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 .../staging/rtl8192u/ieee80211/ieee80211_softmac.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

--- a/drivers/staging/rtl8192u/ieee80211/ieee80211_softmac.c
+++ b/drivers/staging/rtl8192u/ieee80211/ieee80211_softmac.c
@@ -1713,7 +1713,8 @@ ieee80211_rx_auth_rq(struct ieee80211_de
 	//IEEE80211DMESG("Rx probe");
 	ieee->softmac_stats.rx_auth_rq++;
 
-	if ((status = auth_rq_parse(skb, dest))!= -1){
+	status = auth_rq_parse(skb, dest);
+	if (status != -1) {
 		ieee80211_resp_to_auth(ieee, status, dest);
 	}
 	//DMESG("Dest is "MACSTR, MAC2STR(dest));
