From 4a533365b477497125204c9e29fa204105093105 Mon Sep 17 00:00:00 2001
From: Mike McCormack <mikem@ring3k.org>
Date: Mon, 26 Jul 2010 22:57:35 +0900
Subject: [PATCH 09/17] staging: rtl8192e: Make static variables const

Signed-off-by: Mike McCormack <mikem@ring3k.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/rtl8192e/r8192E_core.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/staging/rtl8192e/r8192E_core.c
+++ b/drivers/staging/rtl8192e/r8192E_core.c
@@ -349,8 +349,8 @@ u8 rtl8192e_ap_sec_type(struct ieee80211
 	//struct r8192_priv* priv = ieee80211_priv(dev);
 	//struct ieee80211_device *ieee = priv->ieee80211;
 
-	static u8 ccmp_ie[4] = {0x00,0x50,0xf2,0x04};
-	static u8 ccmp_rsn_ie[4] = {0x00, 0x0f, 0xac, 0x04};
+	static const u8 ccmp_ie[4] = {0x00,0x50,0xf2,0x04};
+	static const u8 ccmp_rsn_ie[4] = {0x00, 0x0f, 0xac, 0x04};
 	int wpa_ie_len= ieee->wpa_ie_len;
 	struct ieee80211_crypt_data* crypt;
 	int encrypt;
