From f0161757733f924cefe5ae45d012aef1daccebeb Mon Sep 17 00:00:00 2001
From: Mike McCormack <mikem@ring3k.org>
Date: Mon, 26 Jul 2010 22:58:36 +0900
Subject: [PATCH 14/17] staging: rtl8192e: Remove useless debug code

Signed-off-by: Mike McCormack <mikem@ring3k.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/rtl8192e/r8192E_core.c |    5 -----
 1 files changed, 0 insertions(+), 5 deletions(-)

--- a/drivers/staging/rtl8192e/r8192E_core.c
+++ b/drivers/staging/rtl8192e/r8192E_core.c
@@ -1735,11 +1735,6 @@ short rtl8192_tx(struct net_device *dev,
     pdesc->NoEnc = 1;
     pdesc->SecType = 0x0;
     if (tcb_desc->bHwSec) {
-        static u8 tmp =0;
-        if (!tmp) {
-            printk("==>================hw sec\n");
-            tmp = 1;
-        }
         switch (priv->ieee80211->pairwise_key_type) {
             case KEY_TYPE_WEP40:
             case KEY_TYPE_WEP104:
