From 125ed824d3d03f5c470cf98aca757fd9423c49ab Mon Sep 17 00:00:00 2001
From: Mike McCormack <mikem@ring3k.org>
Date: Mon, 26 Jul 2010 22:58:54 +0900
Subject: [PATCH 16/17] staging: rtl8192e: Delete some redundant code

Signed-off-by: Mike McCormack <mikem@ring3k.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/rtl8192e/r8192E_dm.c |    4 ----
 1 files changed, 0 insertions(+), 4 deletions(-)

--- a/drivers/staging/rtl8192e/r8192E_dm.c
+++ b/drivers/staging/rtl8192e/r8192E_dm.c
@@ -2094,8 +2094,6 @@ static void dm_ctrl_initgain_byrssi(stru
 		dm_ctrl_initgain_byrssi_by_fwfalse_alarm(dev);
 	else if(dm_digtable.dig_algorithm == DIG_ALGO_BY_RSSI)
 		dm_ctrl_initgain_byrssi_by_driverrssi(dev);
-	else
-		return;
 }
 
 
@@ -2938,8 +2936,6 @@ void dm_gpio_change_rf_callback(struct w
 	RT_RF_POWER_STATE	eRfPowerStateToSet;
 	bool bActuallySet = false;
 
-		bActuallySet=false;
-
 		if(!priv->up)
 		{
 		RT_TRACE((COMP_INIT | COMP_POWER | COMP_RF),"dm_gpio_change_rf_callback(): Callback function breaks out!!\n");
