diff -Nrup a/drivers/input/Kconfig b/drivers/input/Kconfig
--- a/drivers/input/Kconfig	2009-11-13 00:46:07.000000000 +0000
+++ b/drivers/input/Kconfig	2009-11-13 13:01:06.725587607 +0000
@@ -170,6 +170,8 @@ source "drivers/input/tablet/Kconfig"
 
 source "drivers/input/touchscreen/Kconfig"
 
+source "drivers/input/lirc/Kconfig"
+
 source "drivers/input/misc/Kconfig"
 
 endif
diff -Nrup a/drivers/input/Makefile b/drivers/input/Makefile
--- a/drivers/input/Makefile	2009-11-13 00:46:07.000000000 +0000
+++ b/drivers/input/Makefile	2009-11-13 13:01:06.726563798 +0000
@@ -25,3 +25,5 @@ obj-$(CONFIG_INPUT_MISC)	+= misc/
 obj-$(CONFIG_INPUT_APMPOWER)	+= apm-power.o
 
 obj-$(CONFIG_XEN_KBDDEV_FRONTEND)	+= xen-kbdfront.o
+
+obj-$(CONFIG_INPUT_LIRC)       += lirc/
diff -Nrup a/MAINTAINERS b/MAINTAINERS
--- a/MAINTAINERS	2009-11-13 00:46:07.000000000 +0000
+++ b/MAINTAINERS	2009-11-13 13:01:06.725587607 +0000
@@ -3219,6 +3219,15 @@ W:	http://www.pasemi.com/
 L:	linuxppc-dev@ozlabs.org
 S:	Supported
 
+LINUX INFRARED REMOTE CONTROL DRIVERS (LIRC)
+P:	Jarod Wilson
+M:	jarod@redhat.com
+P:	Christoph Bartelmus
+M:	lirc@bartelmus.de
+W:	http://www.lirc.org/
+L:	lirc-list@lists.sourceforge.net
+S:	Maintained
+
 LINUX SECURITY MODULE (LSM) FRAMEWORK
 M:	Chris Wright <chrisw@sous-sol.org>
 L:	linux-security-module@vger.kernel.org
