
https://bugzilla.redhat.com/bugzilla/show_bug.cgi?id=181963

--- linux-2.6.21.noarch/fs/fat/inode.c~	2007-05-27 23:22:31.000000000 -0400
+++ linux-2.6.21.noarch/fs/fat/inode.c	2007-05-27 23:22:50.000000000 -0400
@@ -950,7 +950,8 @@ static int parse_options(char *options, 
 		opts->shortname = 0;
 	opts->name_check = 'n';
 	opts->quiet = opts->showexec = opts->sys_immutable = opts->dotsOK =  0;
-	opts->utf8 = opts->unicode_xlate = 0;
+	opts->utf8 = 1;
+	opts->unicode_xlate = 0;
 	opts->numtail = 1;
 	opts->usefree = opts->nocase = 0;
 	*debug = 0;
