
https://bugzilla.redhat.com/bugzilla/show_bug.cgi?id=181963

--- a/fs/fat/inode.c
+++ b/fs/fat/inode.c
@@ -978,7 +978,8 @@ static int parse_options(char *options,
 	}
 	opts->name_check = 'n';
 	opts->quiet = opts->showexec = opts->sys_immutable = opts->dotsOK =  0;
-	opts->utf8 = opts->unicode_xlate = 0;
+	opts->utf8 = 1;
+	opts->unicode_xlate = 0;
 	opts->numtail = 1;
 	opts->usefree = opts->nocase = 0;
 	opts->tz_utc = 0;
