diff -Nrup a/drivers/net/wireless/rtl8180_grf5101.c b/drivers/net/wireless/rtl8180_grf5101.c
--- a/drivers/net/wireless/rtl8180_grf5101.c	2008-04-17 04:49:44.000000000 +0200
+++ b/drivers/net/wireless/rtl8180_grf5101.c	2008-05-10 23:25:33.000000000 +0200
@@ -87,7 +87,7 @@ static void grf5101_rf_set_channel(struc
 	write_grf5101(dev, 0x0B, chan);
 	write_grf5101(dev, 0x07, 0x1000);
 
-	grf5101_write_phy_antenna(dev, chan);
+	grf5101_write_phy_antenna(dev, conf->channel);
 }
 
 static void grf5101_rf_stop(struct ieee80211_hw *dev)
diff -Nrup a/drivers/net/wireless/rtl8180_max2820.c b/drivers/net/wireless/rtl8180_max2820.c
--- a/drivers/net/wireless/rtl8180_max2820.c	2008-04-17 04:49:44.000000000 +0200
+++ b/drivers/net/wireless/rtl8180_max2820.c	2008-05-10 23:24:54.000000000 +0200
@@ -86,7 +86,7 @@ static void max2820_rf_set_channel(struc
 	 * sa2400, for MAXIM we do this directly from BB */
 	rtl8180_write_phy(dev, 3, txpw);
 
-	max2820_write_phy_antenna(dev, chan);
+	max2820_write_phy_antenna(dev, chan_idx + 1);
 	write_max2820(dev, 3, chan);
 }
 
diff -Nrup a/drivers/net/wireless/rtl8180_sa2400.c b/drivers/net/wireless/rtl8180_sa2400.c
--- a/drivers/net/wireless/rtl8180_sa2400.c	2008-04-17 04:49:44.000000000 +0200
+++ b/drivers/net/wireless/rtl8180_sa2400.c	2008-05-10 23:24:00.000000000 +0200
@@ -85,7 +85,7 @@ static void sa2400_rf_set_channel(struct
 
 	write_sa2400(dev, 7, txpw);
 
-	sa2400_write_phy_antenna(dev, chan);
+	sa2400_write_phy_antenna(dev, conf->channel);
 
 	write_sa2400(dev, 0, chan);
 	write_sa2400(dev, 1, 0xbb50);
