diff -Nrup a/drivers/net/wireless/rtl8180_dev.c b/drivers/net/wireless/rtl8180_dev.c
--- a/drivers/net/wireless/rtl8180_dev.c	2008-01-10 21:03:08.000000000 +0000
+++ b/drivers/net/wireless/rtl8180_dev.c	2008-01-10 21:37:43.000000000 +0000
@@ -103,7 +103,6 @@ static void rtl8180_handle_rx(struct iee
 			rx_status.channel = dev->conf.channel;
 			rx_status.phymode = dev->conf.phymode;
 			rx_status.mactime = le64_to_cpu(entry->tsft);
-			rx_status.flag |= RX_FLAG_TSFT;
 			if (flags & RTL8180_RX_DESC_FLAG_CRC32_ERR)
 				rx_status.flag |= RX_FLAG_FAILED_FCS_CRC;
 
@@ -236,8 +235,7 @@ static int rtl8180_tx(struct ieee80211_h
 		kmemdup(control, sizeof(*control), GFP_ATOMIC);
 
 	if (control->flags & IEEE80211_TXCTL_USE_RTS_CTS)
-		rts_duration = ieee80211_rts_duration(dev, priv->vif, skb->len,
-						      control);
+		rts_duration = ieee80211_rts_duration(dev, priv->if_id, skb->len, control);
 
 	if (!priv->r8185) {
 		unsigned int remainder;
@@ -639,8 +637,6 @@ static int rtl8180_add_interface(struct 
 		return -EOPNOTSUPP;
 	}
 
-	priv->vif = conf->vif;
-
 	rtl818x_iowrite8(priv, &priv->map->EEPROM_CMD, RTL818X_EEPROM_CMD_CONFIG);
 	rtl818x_iowrite32(priv, (__le32 __iomem *)&priv->map->MAC[0],
 			  cpu_to_le32(*(u32 *)conf->mac_addr));
@@ -656,7 +652,6 @@ static void rtl8180_remove_interface(str
 {
 	struct rtl8180_priv *priv = dev->priv;
 	priv->mode = IEEE80211_IF_TYPE_MNTR;
-	priv->vif = NULL;
 }
 
 static int rtl8180_config(struct ieee80211_hw *dev, struct ieee80211_conf *conf)
@@ -668,8 +663,7 @@ static int rtl8180_config(struct ieee802
 	return 0;
 }
 
-static int rtl8180_config_interface(struct ieee80211_hw *dev,
-				    struct ieee80211_vif *vif,
+static int rtl8180_config_interface(struct ieee80211_hw *dev, int if_id,
 				    struct ieee80211_if_conf *conf)
 {
 	struct rtl8180_priv *priv = dev->priv;
diff -Nrup a/drivers/net/wireless/rtl8180.h b/drivers/net/wireless/rtl8180.h
--- a/drivers/net/wireless/rtl8180.h	2008-01-10 21:03:08.000000000 +0000
+++ b/drivers/net/wireless/rtl8180.h	2008-01-10 21:39:31.000000000 +0000
@@ -90,8 +90,8 @@ struct rtl8180_priv {
 	/* common between rtl818x drivers */
 	struct rtl818x_csr __iomem *map;
 	const struct rtl818x_rf_ops *rf;
-	struct ieee80211_vif *vif;
 	int mode;
+	int if_id;
 
 	/* rtl8180 driver specific */
 	spinlock_t lock;
diff -Nrup a/drivers/net/wireless/rtl8187_dev.c b/drivers/net/wireless/rtl8187_dev.c
--- a/drivers/net/wireless/rtl8187_dev.c	2008-01-10 21:03:08.000000000 +0000
+++ b/drivers/net/wireless/rtl8187_dev.c	2008-01-10 21:34:52.000000000 +0000
@@ -150,8 +150,7 @@ static int rtl8187_tx(struct ieee80211_h
 		flags |= RTL8187_TX_FLAG_MORE_FRAG;
 	if (control->flags & IEEE80211_TXCTL_USE_RTS_CTS) {
 		flags |= RTL8187_TX_FLAG_RTS;
-		rts_dur = ieee80211_rts_duration(dev, priv->vif,
-						 skb->len, control);
+		rts_dur = ieee80211_rts_duration(dev, priv->if_id, skb->len, control);
 	}
 	if (control->flags & IEEE80211_TXCTL_USE_CTS_PROTECT)
 		flags |= RTL8187_TX_FLAG_CTS;
@@ -228,7 +227,6 @@ static void rtl8187_rx_cb(struct urb *ur
 	rx_status.channel = dev->conf.channel;
 	rx_status.phymode = dev->conf.phymode;
 	rx_status.mactime = le64_to_cpu(hdr->mac_time);
-	rx_status.flag |= RX_FLAG_TSFT;
 	if (flags & (1 << 13))
 		rx_status.flag |= RX_FLAG_FAILED_FCS_CRC;
 	ieee80211_rx_irqsafe(dev, skb, &rx_status);
@@ -561,13 +559,14 @@ static int rtl8187_config(struct ieee802
 	return 0;
 }
 
-static int rtl8187_config_interface(struct ieee80211_hw *dev,
-				    struct ieee80211_vif *vif,
+static int rtl8187_config_interface(struct ieee80211_hw *dev, int if_id,
 				    struct ieee80211_if_conf *conf)
 {
 	struct rtl8187_priv *priv = dev->priv;
 	int i;
 
+	priv->if_id = if_id;
+
 	for (i = 0; i < ETH_ALEN; i++)
 		rtl818x_iowrite8(priv, &priv->map->BSSID[i], conf->bssid[i]);
 
diff -Nrup a/drivers/net/wireless/rtl8187.h b/drivers/net/wireless/rtl8187.h
--- a/drivers/net/wireless/rtl8187.h	2008-01-10 21:03:08.000000000 +0000
+++ b/drivers/net/wireless/rtl8187.h	2008-01-10 21:35:00.000000000 +0000
@@ -65,8 +65,8 @@ struct rtl8187_priv {
 	/* common between rtl818x drivers */
 	struct rtl818x_csr *map;
 	const struct rtl818x_rf_ops *rf;
-	struct ieee80211_vif *vif;
 	int mode;
+	int if_id;
 
 	/* rtl8187 specific */
 	struct ieee80211_channel channels[14];
