--- linux-2.6.21.noarch/drivers/input/keyboard/atkbd.c~	2007-07-06 10:51:04.000000000 -0400
+++ linux-2.6.21.noarch/drivers/input/keyboard/atkbd.c	2007-07-06 10:51:33.000000000 -0400
@@ -409,10 +409,14 @@ static irqreturn_t atkbd_interrupt(struc
 			goto out;
 		case ATKBD_RET_ACK:
 		case ATKBD_RET_NAK:
+#if 0
+			/* Quite a few key switchers and other tools trigger this and it confuses
+			   people who can do nothing about it */
 			if (printk_ratelimit())
 				printk(KERN_WARNING "atkbd.c: Spurious %s on %s. "
 				       "Some program might be trying access hardware directly.\n",
 				       data == ATKBD_RET_ACK ? "ACK" : "NAK", serio->phys);
+#endif
 			goto out;
 		case ATKBD_RET_ERR:
 			atkbd->err_count++;
