diff -Nrup a/security/security.c b/security/security.c
--- a/security/security.c	2008-10-09 22:13:53.000000000 +0000
+++ b/security/security.c	2008-10-14 11:21:53.000000000 +0000
@@ -439,6 +439,7 @@ int security_inode_permission(struct ino
 		return 0;
 	return security_ops->inode_permission(inode, mask);
 }
+EXPORT_SYMBOL_GPL(security_inode_permission);
 
 int security_inode_setattr(struct dentry *dentry, struct iattr *attr)
 {
