--- a/drivers/net/wireless/ath5k/base.c
+++ b/drivers/net/wireless/ath5k/base.c
@@ -2636,6 +2636,12 @@ ath5k_init_leds(struct ath5k_softc *sc)
 		sc->led_pin = 3;
 		sc->led_on = 0;  /* active low */
 	}
+	/* Pin 3 on AzureWave chips used in Fukato Datacask Jupiter */
+	if (pdev->subsystem_vendor == PCI_VENDOR_ID_AZWAVE) {
+		__set_bit(ATH_STAT_LEDSOFT, sc->status);
+		sc->led_pin = 3;
+		sc->led_on = 0;  /* active low */
+	}
 
 	if (!test_bit(ATH_STAT_LEDSOFT, sc->status))
 		goto out;
