diff -up linux-2.6.29.noarch/drivers/gpu/drm/drm_edid.c.jx linux-2.6.29.noarch/drivers/gpu/drm/drm_edid.c
--- linux-2.6.29.noarch/drivers/gpu/drm/drm_edid.c.jx	2009-05-15 15:16:07.000000000 -0400
+++ linux-2.6.29.noarch/drivers/gpu/drm/drm_edid.c	2009-05-15 16:05:16.000000000 -0400
@@ -289,6 +289,11 @@ static struct drm_display_mode *drm_mode
 	struct drm_display_mode *mode;
 	struct detailed_pixel_timing *pt = &timing->data.pixel_data;
 
+	/* ignore tiny modes */
+	if (((pt->hactive_hi << 8) | pt->hactive_lo) < 64 ||
+	    ((pt->vactive_hi << 8) | pt->hactive_lo) < 64)
+		return NULL;
+
 	if (pt->stereo) {
 		printk(KERN_WARNING "stereo mode not supported\n");
 		return NULL;
