kvm_intel: enabled nested on default

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>

--- a/arch/x86/kvm/vmx.c
+++ b/arch/x86/kvm/vmx.c
@@ -94,7 +94,7 @@ module_param_named(enable_shadow_vmcs, e
  * VMX and be a hypervisor for its own guests. If nested=0, guests may not
  * use VMX instructions.
  */
-static bool __read_mostly nested = 0;
+static bool __read_mostly nested = 1;
 module_param(nested, bool, S_IRUGO);
 
 #define KVM_GUEST_CR0_MASK (X86_CR0_NW | X86_CR0_CD)
