diff --git a/kernel/module.c b/kernel/module.c
index 5f80478..ef3b6d9 100644
--- a/kernel/module.c
+++ b/kernel/module.c
@@ -1998,6 +1998,13 @@ static struct module *load_module(void __user *umod,
 	if (strcmp(mod->name, "driverloader") == 0)
 		add_taint_module(mod, TAINT_PROPRIETARY_MODULE);
 
+	/* 
+	 * no games, call it what it is
+	 * even if "certain" tools claim otherwise
+	 * */
+	if (strcmp(mod->name, "fglrx") == 0)
+		add_taint_module(mod, TAINT_PROPRIETARY_MODULE);
+
 	/* Set up MODINFO_ATTR fields */
 	setup_modinfo(mod, sechdrs, infoindex);
 
