--- a/kernel/module.c
+++ b/kernel/module.c
@@ -2329,6 +2329,13 @@ static noinline struct module *load_modu
 	if (strcmp(mod->name, "driverloader") == 0)
 		add_taint_module(mod, TAINT_PROPRIETARY_MODULE);
 
+	/* 
+	 * no games, call it what it is
+	 * even if "certain" tools claim otherwise
+	 */
+	if (strcmp(mod->name, "fglrx") == 0)
+		add_taint_module(mod, TAINT_PROPRIETARY_MODULE);
+
 	/* Set up MODINFO_ATTR fields */
 	setup_modinfo(mod, sechdrs, infoindex);
 
