--- a/drivers/staging/rtl8192su/r8192U.h
+++ b/drivers/staging/rtl8192su/r8192U.h
@@ -90,12 +90,18 @@
 #define DMESG(x,a...)
 #define DMESGW(x,a...)
 #define DMESGE(x,a...)
+
 extern u32 rt_global_debug_component;
+#ifdef RTL819x_DEBUG
 #define RT_TRACE(component, x, args...) \
 do { if(rt_global_debug_component & component) \
 	printk(KERN_DEBUG RTL819xU_MODULE_NAME ":" x "\n" , \
 	       ##args);\
 }while(0);
+#else
+#define RT_TRACE(component, x, args...) do {} while (0)
+#endif /* RTL8169_DEBUG */
+
 //----------------------------------------------------------------------
 //// Get 8192SU  Rx descriptor. Added by Roger, 2008.04.15.
 ////----------------------------------------------------------------------
@@ -145,7 +151,6 @@ do { if(rt_global_debug_component & comp
 #define COMP_RESET				BIT30  //for silent reset
 #define COMP_ERR				BIT31 //for error out, always on
 
-#define RTL819x_DEBUG
 #ifdef RTL819x_DEBUG
 #define assert(expr) \
         if (!(expr)) {                                  \
--- a/drivers/staging/rtl8192su/ieee80211/ieee80211.h
+++ b/drivers/staging/rtl8192su/ieee80211/ieee80211.h
@@ -168,7 +168,6 @@ typedef struct ieee_param {
 #define IEEE80211_QCTL_TID              0x000F
 
 /* debug macros */
-#define CONFIG_IEEE80211_DEBUG
 #ifdef CONFIG_IEEE80211_DEBUG
 extern u32 ieee80211_debug_level;
 #define IEEE80211_DEBUG(level, fmt, args...) \
