--- a/drivers/staging/rtl8192su/r8192U_core.c
+++ b/drivers/staging/rtl8192su/r8192U_core.c
@@ -141,19 +141,17 @@ MODULE_DESCRIPTION("Linux driver for Rea
 
 static char* ifname = "wlan%d";
 static int hwwep = 1;  //default use hw. set 0 to use software security
-static int channels = 0x3fff;
-
-
+static int modparam_override_countrycode = -1;
 
 module_param(ifname, charp, S_IRUGO|S_IWUSR );
 //module_param(hwseqnum,int, S_IRUGO|S_IWUSR);
 module_param(hwwep,int, S_IRUGO|S_IWUSR);
-module_param(channels,int, S_IRUGO|S_IWUSR);
+module_param_named(override_countrycode, modparam_override_countrycode, int, S_IRUGO);
 
 MODULE_PARM_DESC(ifname," Net interface name, wlan%d=default");
 //MODULE_PARM_DESC(hwseqnum," Try to use hardware 802.11 header sequence numbers. Zero=default");
 MODULE_PARM_DESC(hwwep," Try to use hardware security support. ");
-MODULE_PARM_DESC(channels," Channel bitmask for specific locales. NYI");
+MODULE_PARM_DESC(override_countrycode, "Override countrycode hardcoded in EEPROM with this value (DANGEROUS).");
 
 static int __devinit rtl8192_usb_probe(struct usb_interface *intf,
 			 const struct usb_device_id *id);
@@ -218,6 +216,10 @@ static void rtl819x_set_channel_map(u8 c
 {
 	int i, max_chan=-1, min_chan=-1;
 	struct ieee80211_device* ieee = priv->ieee80211;
+
+	if (modparam_override_countrycode != -1)
+		channel_plan = modparam_override_countrycode;
+		
 	switch (channel_plan)
 	{
 		case COUNTRY_CODE_FCC:
