diff -Nrup a/security/security.c b/security/security.c
--- a/security/security.c	2008-04-17 04:49:44.000000000 +0200
+++ b/security/security.c	2008-05-02 13:15:44.000000000 +0200
@@ -433,6 +433,7 @@ int security_inode_permission(struct ino
 		return 0;
 	return security_ops->inode_permission(inode, mask, nd);
 }
+EXPORT_SYMBOL_GPL(security_inode_permission);
 
 int security_inode_setattr(struct dentry *dentry, struct iattr *attr)
 {
