--- a/drivers/md/dm.c
+++ b/drivers/md/dm.c
@@ -271,7 +271,7 @@ static int __init dm_init(void)
 	const int count = ARRAY_SIZE(_inits);
 	int r, i;
 
-#if (IS_ENABLED(CONFIG_IMA) && !IS_ENABLED(CONFIG_IMA_DISABLE_HTABLE))
+#if (IS_ENABLED(CONFIG_IMA) && !IS_ENABLED(CONFIG_IMA_DISABLE_HTABLE)) && 0
 	DMWARN("CONFIG_IMA_DISABLE_HTABLE is disabled."
 	       " Duplicate IMA measurements will not be recorded in the IMA log.");
 #endif
