From 9d67de9a0ec1c9eed5be2c733210e9759c063b12 Mon Sep 17 00:00:00 2001
From: Ben Hutchings <ben@decadent.org.uk>
Date: Sat, 18 Oct 2008 02:46:55 +0100
Subject: [PATCH 16/24] mga: Mark as broken

---
 drivers/gpu/drm/Kconfig |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

--- a/drivers/gpu/drm/Kconfig
+++ b/drivers/gpu/drm/Kconfig
@@ -136,6 +136,7 @@ config DRM_SIS
 config DRM_VIA
 	tristate "Via unichrome video cards"
 	depends on DRM
+	depends on BROKEN
 	help
 	  Choose this option if you have a Via unichrome or compatible video
 	  chipset. If M is selected the module will be called via.
