From a5772e725404a5b7be93b51da6ddbebf69c824f0 Mon Sep 17 00:00:00 2001
From: Pontus Fuchs <pontus.fuchs@gmail.com>
Date: Fri, 7 Sep 2012 14:01:48 +0200
Subject: [PATCH 17/58] Probe EEPROM for max RX buf size and use it.

---
 ar5523.c |   38 ++++++++++++++++++++++++++++++++++----
 ar5523.h |    3 ---
 2 files changed, 34 insertions(+), 7 deletions(-)

--- a/drivers/net/wireless/ar5523.c
+++ b/drivers/net/wireless/ar5523.c
@@ -1409,8 +1409,7 @@ static int ar5523_host_available(struct
 		&setup, sizeof setup, NULL, 0, 0);
 }
 
-static int
-ar5523_get_devstatus(struct ar5523 *ar)
+static int ar5523_get_devstatus(struct ar5523 *ar)
 {
 	u8 macaddr[ETH_ALEN];
 	int error;
@@ -1434,6 +1433,33 @@ ar5523_get_devstatus(struct ar5523 *ar)
 	return 0;
 }
 
+#define AR5523_SANE_RXBUFSZ 2000
+
+static int ar5523_get_max_rxsz(struct ar5523 *ar)
+{
+	int error;
+	__be32 rxsize;
+
+	/* Get max rx size */
+	error = ar5523_get_status(ar, ST_WDC_TRANSPORT_CHUNK_SIZE, &rxsize,
+				  sizeof(rxsize));
+	if (error != 0) {
+		ar5523_err(ar, "could not read max RX size\n");
+		return error;
+	}
+
+	ar->rxbufsz = be32_to_cpu(rxsize);
+
+	if (!ar->rxbufsz || ar->rxbufsz > AR5523_SANE_RXBUFSZ) {
+		ar5523_err(ar, "Bad rxbufsz from device. Using %d instead\n",
+			   AR5523_SANE_RXBUFSZ);
+		ar->rxbufsz = AR5523_SANE_RXBUFSZ;
+	}
+
+	ar5523_dbg(ar, "Mac RX buf size: %d\n", ar->rxbufsz);
+	return 0;
+}
+
 
 /*
  * This is copied from rtl818x, but we should probably move this
@@ -1620,8 +1646,6 @@ static int ar5523_probe(struct usb_inter
 	INIT_LIST_HEAD(&ar->tx_cmd_used);
 	spin_lock_init(&ar->tx_cmd_list_lock);
 
-	ar->rxbufsz = 2000;	//TODO
-
 	error = ar5523_alloc_tx_cmds(ar);
 	if (error) {
 		ar5523_err(ar, "Could not allocate tx command buffers\n");
@@ -1637,6 +1661,12 @@ static int ar5523_probe(struct usb_inter
 		goto out_free_tx_cmds;
 	}
 
+	error = ar5523_get_max_rxsz(ar);
+	if (error) {
+		ar5523_err(ar, "could not get caps from adapter\n");
+		goto out_free_tx_cmds;
+	}
+
 	error = ar5523_get_devcap(ar);
 	if (error) {
 		ar5523_err(ar, "could not get caps from adapter\n");
--- a/drivers/net/wireless/ar5523.h
+++ b/drivers/net/wireless/ar5523.h
@@ -436,9 +436,6 @@ enum {
 	TARGET_DEVICE_RESUME,
 };
 
-#define AR5523_EEPROM_MACADDR	0x0b
-#define AR5523_EEPROM_RXBUFSZ	0x0f
-
 #define AR5523_MAX_TXBUFSZ	\
 	(sizeof(__be32) + sizeof(struct ar5523_tx_desc) + IEEE80211_MAX_LEN)
 
