From 4a5072bb3e7df0a354ac4feb2ed888734a4c9601 Mon Sep 17 00:00:00 2001
From: Pontus Fuchs <pontus.fuchs@gmail.com>
Date: Tue, 11 Sep 2012 13:14:05 +0200
Subject: [PATCH 33/68] Remove comment about HZ / msec.

It's msec and looks fine.
---
 ar5523.c |    1 -
 1 file changed, 1 deletion(-)

--- a/drivers/net/wireless/ar5523.c
+++ b/drivers/net/wireless/ar5523.c
@@ -88,7 +88,6 @@ enum {
 	usb_rcvbulkpipe((dev), AR5523_DATA_RX_PIPE)
 
 enum {
-	/* XXX: msec or HZ? */
 	AR5523_DATA_TIMEOUT		= 10000,
 	AR5523_CMD_TIMEOUT		= 1000,
 };
