From 86cf2acda2d20900b6d03e51e12cead781a4769f Mon Sep 17 00:00:00 2001
From: Pontus Fuchs <pontus.fuchs@gmail.com>
Date: Tue, 11 Sep 2012 13:31:00 +0200
Subject: [PATCH 34/68] Remove some unused defintions

---
 ar5523.c |    4 ++--
 ar5523.h |   18 ++----------------
 2 files changed, 4 insertions(+), 18 deletions(-)

--- a/drivers/net/wireless/ar5523.c
+++ b/drivers/net/wireless/ar5523.c
@@ -592,7 +592,7 @@ static int ar5523_reset_tx_queues(struct
 
 		ar5523_dbg(ar, "resetting Tx queue %d\n", AR5523_AC_TO_QID(ac));
 
-		error = ar5523_cmd_write(ar, AR5523_CMD_RESET_QUEUE,
+		error = ar5523_cmd_write(ar, WDCMSG_RELEASE_TX_QUEUE,
 					 &qid, sizeof(qid), 0);
 		if (error)
 			break;
@@ -650,7 +650,7 @@ static int ar5523_wme_init(struct ar5523
 		ar5523_dbg(ar, "setting up Tx queue %d\n",
 			       AR5523_AC_TO_QID(ac));
 
-		error = ar5523_cmd_write(ar, AR5523_CMD_SET_QUEUE,
+		error = ar5523_cmd_write(ar, WDCMSG_SETUP_TX_QUEUE,
 					 &qinfo, sizeof(qinfo), 0);
 		if (error)
 			break;
--- a/drivers/net/wireless/ar5523.h
+++ b/drivers/net/wireless/ar5523.h
@@ -34,9 +34,6 @@
  * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
 
-#define AR5523_CONFIG_NO		1
-#define AR5523_IFACE_INDEX	0
-
 /* all fields are big endian */
 struct ar5523_fwblock {
 	__be32		flags;
@@ -49,7 +46,7 @@ struct ar5523_fwblock {
 	__be32		remain;
 	__be32		rxtotal;
 	__be32		pad[123];
-};
+} __packed;
 
 #define AR5523_MAX_RXCMDSZ	1024
 #define AR5523_MAX_TXCMDSZ	1024
@@ -119,11 +116,6 @@ struct ar5523_cmd_hdr {
 #define	WDCMSG_RELEASE_TX_QUEUE		0x3b
 #define	WDCMSG_SET_DEFAULT_KEY		0x43
 
-
-/* TODO: OLD commands: REMOVE! */
-#define AR5523_CMD_SET_QUEUE	0x3a
-#define AR5523_CMD_RESET_QUEUE	0x3b
-
 	__u32		priv;	/* driver private data,
 				   don't care about endianess */
 	__be32		magic;
@@ -142,7 +134,6 @@ struct uath_cmd_host_available {
 #define	ATH_SW_VER_PATCH	0
 #define	ATH_SW_VER_BUILD	9999
 
-
 struct ar5523_chunk {
 	u8		seqnum;		/* sequence number for ordering */
 	u8		flags;
@@ -153,7 +144,6 @@ struct ar5523_chunk {
 	/* chunk data follows */
 } __packed;
 
-
 /*
  * Message format for a WDCMSG_DATA_AVAIL message from Target to Host.
  */
@@ -313,7 +303,7 @@ struct ar5523_qinfo {
 	__be32		acm;
 	__be32		magic1;
 	__be32		magic2;
-};
+} __packed;
 
 struct ar5523_cmd_rx_filter {		/* WDCMSG_RX_FILTER */
 	__be32	bits;
@@ -424,7 +414,6 @@ enum {
 	CAP_TWICE_ANTENNAGAIN_2G,
 };
 
-
 enum {
 	ST_NONE,                    /* Sentinal to indicate "no status" */
 	ST_ALL,
@@ -453,9 +442,6 @@ enum {
 	TARGET_DEVICE_RESUME,
 };
 
-#define AR5523_MAX_TXBUFSZ	\
-	(sizeof(__be32) + sizeof(struct ar5523_tx_desc) + IEEE80211_MAX_LEN)
-
 /* this is in net/ieee80211.h, but that conflicts with the mac80211 headers */
 #define IEEE80211_2ADDR_LEN	16
 
