From 94a2515c39642e2c4bb3f41405b7edc4b467bf1b Mon Sep 17 00:00:00 2001
From: Pontus Fuchs <pontus.fuchs@gmail.com>
Date: Tue, 25 Sep 2012 21:50:02 +0200
Subject: [PATCH 49/58] Free TX packet on USB error

---
 ar5523.c |    9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

--- a/drivers/net/wireless/ar5523.c
+++ b/drivers/net/wireless/ar5523.c
@@ -865,17 +865,14 @@ static void ar5523_data_tx_cb(struct urb
 	struct ar5523 *ar = data->ar;
 	unsigned long flags;
 
-	ar5523_dbg(ar, "data tx urb completed %p\n", data->urb);
+	ar5523_dbg(ar, "data tx urb completed: %d\n", urb->status);
 
 	spin_lock_irqsave(&ar->tx_data_list_lock, flags);
 	list_del(&data->list);
 	spin_unlock_irqrestore(&ar->tx_data_list_lock, flags);
 
-	/* sync/async unlink faults aren't errors */
-	if (urb->status && (urb->status != -ENOENT &&
-	    urb->status != -ECONNRESET && urb->status != -ESHUTDOWN)) {
-		ar5523_dbg(ar, "%s: nonzero write bulk status received: %d\n",
-			   __func__, urb->status);
+	if (urb->status) {
+		ar5523_dbg(ar, "%s: urb status: %d\n", __func__, urb->status);
 		ar5523_data_tx_pkt_put(ar);
 		ieee80211_free_txskb(ar->hw, skb);
 	}
