From 4de941dae9cb555a584191f5e70974a2125a2081 Mon Sep 17 00:00:00 2001
From: Pontus Fuchs <pontus.fuchs@gmail.com>
Date: Thu, 27 Sep 2012 17:06:23 +0200
Subject: [PATCH 51/68] Do build time check of the size of struct
 ar5523_tx_data

To avoid future problems if that struct should grow.
---
 ar5523.c |    7 +++++++
 1 file changed, 7 insertions(+)

--- a/drivers/net/wireless/ar5523.c
+++ b/drivers/net/wireless/ar5523.c
@@ -116,6 +116,9 @@ struct ar5523_tx_cmd {
 	struct completion	done;
 };
 
+/* This struct is placed in tx_info->driver_data. It must not be larger
+ *  than IEEE80211_TX_INFO_DRIVER_DATA_SIZE.
+ */
 struct ar5523_tx_data {
 	struct list_head	list;
 	struct ar5523		*ar;
@@ -919,6 +922,7 @@ static void ar5523_tx(struct ieee80211_h
 	list_add_tail(&data->list, &ar->tx_queue_pending);
 	spin_unlock_irqrestore(&ar->tx_data_list_lock, flags);
 
+
 	ieee80211_queue_work(ar->hw, &ar->tx_work);
 }
 
@@ -934,6 +938,9 @@ static void ar5523_tx_work_locked(struct
 	u32 txqid;
 	unsigned long flags;
 
+	BUILD_BUG_ON(sizeof(struct ar5523_tx_data) >
+		     IEEE80211_TX_INFO_DRIVER_DATA_SIZE);
+
 	ar5523_dbg(ar, "%s\n", __func__);
 	do {
 		spin_lock_irqsave(&ar->tx_data_list_lock, flags);
