From 1edb6460e72497722e7150dc01a38cbe7400f9af Mon Sep 17 00:00:00 2001
From: Pontus Fuchs <pontus.fuchs@gmail.com>
Date: Thu, 27 Sep 2012 20:38:13 +0200
Subject: [PATCH 53/82] Switch firmware name to ar5523.bin

Same name as FreeBSD.
---
 ar5523.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/wireless/ath/ar5523/ar5523.c
+++ b/drivers/net/wireless/ath/ar5523/ar5523.c
@@ -68,7 +68,7 @@ enum {
 	AR5523_FLAG_ABG			= (1 << 1),
 };
 
-#define AR5523_FIRMWARE_FILE	"uath-ar5523.bin"
+#define AR5523_FIRMWARE_FILE	"ar5523.bin"
 
 enum {
 	AR5523_CMD_TX_PIPE	= 0x01,
