From 38f45f33c29f62d99d602191f73877c0a41c4857 Mon Sep 17 00:00:00 2001
From: Pontus Fuchs <pontus.fuchs@gmail.com>
Date: Fri, 28 Sep 2012 20:49:14 +0200
Subject: [PATCH 58/68] Add locking to ar5523_cancel_rx_bufs

Locking was missing.
---
 ar5523.c |   18 +++++++++++++-----
 1 file changed, 13 insertions(+), 5 deletions(-)

--- a/drivers/net/wireless/ar5523.c
+++ b/drivers/net/wireless/ar5523.c
@@ -774,15 +774,23 @@ done:
 static void ar5523_cancel_rx_bufs(struct ar5523 *ar)
 {
 	struct ar5523_rx_data *data;
+	unsigned long flags;
+
+	do {
+		spin_lock_irqsave(&ar->rx_data_list_lock, flags);
+		if (!list_empty(&ar->rx_data_used))
+			data = (struct ar5523_rx_data *) ar->rx_data_used.next;
+		else
+			data = NULL;
+		spin_unlock_irqrestore(&ar->rx_data_list_lock, flags);
+
+		if (!data)
+			break;
 
-	while (!list_empty(&ar->rx_data_used)) {
-		data = (struct ar5523_rx_data *) ar->rx_data_used.next;
 		usb_kill_urb(data->urb);
-		kfree_skb(data->skb);
-		data->skb = NULL;
 		list_move(&data->list, &ar->rx_data_free);
 		atomic_inc(&ar->rx_data_free_cnt);
-	}
+	} while (data);
 }
 
 static void ar5523_free_rx_bufs(struct ar5523 *ar)
