From 1eb81b72a618ed141313756bbcd0b005c457a4e0 Mon Sep 17 00:00:00 2001
From: Pontus Fuchs <pontus.fuchs@gmail.com>
Date: Tue, 2 Oct 2012 20:31:44 +0200
Subject: [PATCH 62/77] Print more info if final flag is missing

---
 ar5523.c |    9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

--- a/drivers/net/wireless/ar5523.c
+++ b/drivers/net/wireless/ar5523.c
@@ -690,13 +690,12 @@ static void ar5523_data_rx_cb(struct urb
 
 	chunk = (struct ar5523_chunk *) data->skb->data;
 
-	if (chunk->flags != UATH_CFLAGS_FINAL)
-		ar5523_dbg(ar, "chunk seq: %d flags: %02x len: %d\n",
+	if (((chunk->flags & UATH_CFLAGS_FINAL) == 0 ) ||
+		chunk->seqnum != 0) {
+		ar5523_err(ar, "RX: No final flag "
+			   "seq: %d flags: %02x len: %d\n",
 			   chunk->seqnum, chunk->flags,
 			   be16_to_cpu(chunk->length));
-
-	if ((chunk->flags & UATH_CFLAGS_FINAL) == 0) {
-		ar5523_err(ar, "RX: No final flag\n");
 		goto skip;
 	}
 
