From 96c99bf57245249ffe325b4f6c795223daeb3687 Mon Sep 17 00:00:00 2001
From: Pontus Fuchs <pontus.fuchs@gmail.com>
Date: Sat, 6 Oct 2012 20:29:36 +0200
Subject: [PATCH 69/73] Update TX op to work with latest kernel

---
 drivers/net/wireless/ar5523.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/drivers/net/wireless/ar5523.c
+++ b/drivers/net/wireless/ar5523.c
@@ -907,7 +907,9 @@ static void ar5523_data_tx_cb(struct urb
 	usb_free_urb(urb);
 }
 
-static void ar5523_tx(struct ieee80211_hw *hw, struct sk_buff *skb)
+static void ar5523_tx(struct ieee80211_hw *hw,
+		       struct ieee80211_tx_control *control,
+		       struct sk_buff *skb)
 {
 	struct ieee80211_tx_info *txi = IEEE80211_SKB_CB(skb);
 	struct ar5523_tx_data *data = (struct ar5523_tx_data *)
