From c4a7197e9aaedb2b29afcf154bfc69c06c9a2681 Mon Sep 17 00:00:00 2001
From: Pontus Fuchs <pontus.fuchs@gmail.com>
Date: Sat, 13 Oct 2012 20:28:30 +0200
Subject: [PATCH 76/77] Add comment about error surpression on USB disconnect.

As suggested on review.
---
 ar5523.h |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

--- a/drivers/net/wireless/ar5523.h
+++ b/drivers/net/wireless/ar5523.h
@@ -153,6 +153,11 @@ struct ar5523 {
 
 #define ar5523_dbg(ar, format, arg...) \
 	dev_dbg(&(ar)->dev->dev, format, ## arg)
+
+/* On USB hot-unplug there can be a lot of URBs in flight and they'll all
+ * fail. Instead of dealing with them in every possible place just surpress
+ * any messages on USB disconnect.
+ */
 #define ar5523_err(ar, format, arg...) \
 do { \
 	if (!test_bit(AR5523_USB_DISCONNECTED, &ar->flags)) { \
@@ -161,4 +166,3 @@ do { \
 } while (0)
 #define ar5523_info(ar, format, arg...)	\
 	dev_info(&(ar)->dev->dev, format, ## arg)
-
