--- /dev/null
+++ b/drivers/staging/ar5523/Kconfig
@@ -0,0 +1,6 @@
+config AR5523
+	tristate "Atheros 5523 wireless support"
+	depends on MAC80211 && USB && EXPERIMENTAL
+	---help---
+	  This is an experimental driver for the Atheros 5523 USB
+	  wireless chips.
--- /dev/null
+++ b/drivers/staging/ar5523/Makefile
@@ -0,0 +1 @@
+obj-$(CONFIG_AR5523)	+= ar5523.o
--- a/drivers/staging/Kconfig
+++ b/drivers/staging/Kconfig
@@ -147,5 +147,7 @@ source "drivers/staging/mrst-touchscreen
 
 source "drivers/staging/msm/Kconfig"
 
+source "drivers/staging/ar5523/Kconfig"
+
 endif # !STAGING_EXCLUDE_BUILD
 endif # STAGING
--- a/drivers/staging/Makefile
+++ b/drivers/staging/Makefile
@@ -54,3 +54,4 @@ obj-$(CONFIG_ADIS16255)		+= adis16255/
 obj-$(CONFIG_FB_XGI)		+= xgifb/
 obj-$(CONFIG_TOUCHSCREEN_MRSTOUCH)	+= mrst-touchscreen/
 obj-$(CONFIG_MSM_STAGING)	+= msm/
+obj-$(CONFIG_AR5523)		+= ar5523/
