--- a/drivers/net/wireless/Kconfig
+++ b/drivers/net/wireless/Kconfig
@@ -71,6 +71,14 @@ config AIRO
 
 	  The driver can be compiled as a module and will be named "airo".
 
+config AR5523
+	tristate "Atheros 5523 wireless support"
+	depends on MAC80211 && USB
+        select FW_LOADER
+	---help---
+	  This is an experimental driver for the Atheros 5523 USB
+	  wireless chips.
+
 config ATMEL
       tristate "Atmel at76c50x chipset  802.11b support"
       depends on (PCI || PCMCIA)
--- a/drivers/net/wireless/Makefile
+++ b/drivers/net/wireless/Makefile
@@ -10,6 +10,8 @@ obj-$(CONFIG_HERMES)		+= orinoco/
 obj-$(CONFIG_AIRO)		+= airo.o
 obj-$(CONFIG_AIRO_CS)		+= airo_cs.o airo.o
 
+obj-$(CONFIG_AR5523)		+= ar5523.o
+
 obj-$(CONFIG_ATMEL)             += atmel.o
 obj-$(CONFIG_PCI_ATMEL)         += atmel_pci.o 
 obj-$(CONFIG_PCMCIA_ATMEL)      += atmel_cs.o
