--- /dev/null
+++ b/drivers/staging/ar5523/Kconfig
@@ -0,0 +1,6 @@
+config AR5523
+	tristate "Atheros 5523 wireless support"
+	depends on MAC80211 && USB && EXPERIMENTAL
+	---help---
+	  This is an experimental driver for the Atheros 5523 USB
+	  wireless chips.
--- /dev/null
+++ b/drivers/staging/ar5523/Makefile
@@ -0,0 +1 @@
+obj-$(CONFIG_AR5523)	+= ar5523.o
--- a/drivers/staging/Kconfig
+++ b/drivers/staging/Kconfig
@@ -141,5 +141,7 @@ source "drivers/staging/netwave/Kconfig"
 
 source "drivers/staging/sm7xx/Kconfig"
 
+source "drivers/staging/ar5523/Kconfig"
+
 endif # !STAGING_EXCLUDE_BUILD
 endif # STAGING
--- a/drivers/staging/Makefile
+++ b/drivers/staging/Makefile
@@ -52,3 +52,4 @@ obj-$(CONFIG_WAVELAN)		+= wavelan/
 obj-$(CONFIG_PCMCIA_WAVELAN)	+= wavelan/
 obj-$(CONFIG_PCMCIA_NETWAVE)	+= netwave/
 obj-$(CONFIG_FB_SM7XX)		+= sm7xx/
+obj-$(CONFIG_AR5523)		+= ar5523/
