add kernel 3.3 compatibility.

Changes-licensed-under: ISC
Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>

--- a/drivers/net/wireless/ar5523.c
+++ b/drivers/net/wireless/ar5523.c
@@ -798,7 +798,7 @@ static void ar5523_data_tx_cb(struct urb
 	usb_free_urb(urb);
 }
 
-static int ar5523_tx(struct ieee80211_hw *hw, struct sk_buff *skb)
+static void ar5523_tx(struct ieee80211_hw *hw, struct sk_buff *skb)
 {
 	struct ieee80211_hdr *wh = (struct ieee80211_hdr *)skb->data;
 	struct ar5523 *ar = hw->priv;
@@ -813,7 +813,7 @@ static int ar5523_tx(struct ieee80211_hw
 
 	if (atomic_read(&ar->tx_data_queued) >= AR5523_TX_DATA_COUNT) {
 		ar5523_dbg(ar, "tx queue full\n");
-		return NETDEV_TX_BUSY;
+		return;
 	}
 
 	urb = usb_alloc_urb(0, GFP_ATOMIC);
@@ -860,13 +860,13 @@ static int ar5523_tx(struct ieee80211_hw
 
 	atomic_inc(&ar->tx_data_queued);
 
-	return NETDEV_TX_OK;
+	return;
 
  out_free_urb:
 	usb_free_urb(urb);
  out_free_skb:
 	kfree_skb(skb);
-	return NETDEV_TX_BUSY;
+	return;
 }
 
 static int ar5523_add_interface(struct ieee80211_hw *hw,
