add kernel 2.6.34 compatibility.

Changes-licensed-under: ISC
Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>

--- a/drivers/staging/ar5523/ar5523.c
+++ b/drivers/staging/ar5523/ar5523.c
@@ -870,7 +870,7 @@ static int ar5523_tx(struct ieee80211_hw
 }
 
 static int ar5523_add_interface(struct ieee80211_hw *hw,
-		struct ieee80211_if_init_conf *conf)
+		struct ieee80211_vif *vif)
 {
 	struct ar5523 *ar = hw->priv;
 
@@ -882,10 +882,10 @@ static int ar5523_add_interface(struct i
 		return -EOPNOTSUPP;
 	}
 
-	switch (conf->type) {
+	switch (vif->type) {
 	case NL80211_IFTYPE_STATION:
 	case NL80211_IFTYPE_MONITOR:
-		ar->mode = conf->type;
+		ar->mode = vif->type;
 		break;
 	default:
 		return -EOPNOTSUPP;
@@ -895,7 +895,7 @@ static int ar5523_add_interface(struct i
 }
 
 static void ar5523_remove_interface(struct ieee80211_hw *hw,
-		struct ieee80211_if_init_conf *conf)
+		struct ieee80211_vif *vif)
 {
 	struct ar5523 *ar = hw->priv;
 
