ar5523: remove dead code.

Changes-licensed-under: ISC
Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>

--- a/drivers/net/wireless/ath/ar5523/ar5523.c
+++ b/drivers/net/wireless/ath/ar5523/ar5523.c
@@ -506,14 +506,7 @@ static int ar5523_set_rates(struct ar552
 
 	memset(&rates, 0, sizeof(rates));
 	rates.magic1 = cpu_to_be32(0x02);
-#if 0
-	rates.size   = cpu_to_be32(1 + sizeof(rates.rates));
-	rates.nrates = rs->rs_nrates;
-	memcpy(rates.rates, rs->rs_rates, rs->rs_nrates);
-#endif
-	
-//	ar5523_dbg(ar, "setting supported rates nrates=%d\n", rs->rs_nrates);
-	
+
 	return ar5523_cmd_write(ar, AR5523_CMD_SET_RATES,
 				&rates, sizeof(rates), 0);
 }
@@ -546,7 +539,6 @@ static void ar5523_data_rx_cb(struct urb
 
 	if (len < AR5523_MIN_RXBUFSZ) {
 		ar5523_err(ar, "wrong xfer size (len=%d)\n", len);
-//		ifp->if_ierrors++;
 		goto skip;
 	}
 
@@ -559,7 +551,6 @@ static void ar5523_data_rx_cb(struct urb
 	if (be32_to_cpu(desc->len) > ar->rxbufsz) {
 		ar5523_err(ar, "bad descriptor (len=%d)\n",
 			       be32_to_cpu(desc->len));
-//		ifp->if_ierrors++;
 		goto skip;
 	}
 
@@ -712,16 +703,6 @@ static int ar5523_start(struct ieee80211
 	
 	ar5523_dbg(ar, "command 07h return code: %x\n", be32_to_cpu(val));
 
-#if 0
-	/* set default channel */
-	ic->ic_bss->ni_chan = ic->ic_ibss_chan;
-	error = ar5523_set_chan(ar, ic->ic_bss->ni_chan);
-	if (error) {
-		ar5523_err(ar, "could not set channel\n");
-		goto fail;
-	}
-#endif
-
 	error = ar5523_wme_init(ar);
 	if (error) {
 		ar5523_err(ar, "could not setup WME parameters\n");
