From patchwork Sat Nov  7 21:43:59 2009
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Subject: [06/75] isicom: declare MODULE_FIRMWARE
Date: Sat, 07 Nov 2009 21:43:59 -0000
From: Ben Hutchings <ben@decadent.org.uk>
X-Patchwork-Id: 58391

Signed-off-by: Ben Hutchings <ben@decadent.org.uk>
Reviewed-by: Jiri Slaby <jirislaby@gmail.com>

---
drivers/char/isicom.c |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

--- a/drivers/char/isicom.c
+++ b/drivers/char/isicom.c
@@ -1782,3 +1782,8 @@ module_exit(isicom_exit);
 MODULE_AUTHOR("MultiTech");
 MODULE_DESCRIPTION("Driver for the ISI series of cards by MultiTech");
 MODULE_LICENSE("GPL");
+MODULE_FIRMWARE("isi608.bin");
+MODULE_FIRMWARE("isi608em.bin");
+MODULE_FIRMWARE("isi616em.bin");
+MODULE_FIRMWARE("isi4608.bin");
+MODULE_FIRMWARE("isi4616.bin");
