From patchwork Sat Nov  7 21:44:52 2009
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Subject: [07/75] moxa: declare MODULE_FIRMWARE
Date: Sat, 07 Nov 2009 21:44:52 -0000
From: Ben Hutchings <ben@decadent.org.uk>
X-Patchwork-Id: 58392

Signed-off-by: Ben Hutchings <ben@decadent.org.uk>
Reviewed-by: Jiri Slaby <jirislaby@gmail.com>

---
drivers/char/moxa.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/drivers/char/moxa.c b/drivers/char/moxa.c
index dd0083b..26c9259 100644
--- a/drivers/char/moxa.c
+++ b/drivers/char/moxa.c
@@ -175,6 +175,9 @@ static unsigned int numports[MAX_BOARDS];
 MODULE_AUTHOR("William Chen");
 MODULE_DESCRIPTION("MOXA Intellio Family Multiport Board Device Driver");
 MODULE_LICENSE("GPL");
+MODULE_FIRMWARE("c218tunx.cod");
+MODULE_FIRMWARE("cp204unx.cod");
+MODULE_FIRMWARE("c320tunx.cod");
 #ifdef MODULE
 module_param_array(type, uint, NULL, 0);
 MODULE_PARM_DESC(type, "card type: C218=2, C320=4");
