From patchwork Sat Nov  7 22:05:55 2009
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Subject: [61/75] icom: declare MODULE_FIRMWARE
Date: Sat, 07 Nov 2009 22:05:55 -0000
From: Ben Hutchings <ben@decadent.org.uk>
X-Patchwork-Id: 58438

Signed-off-by: Ben Hutchings <ben@decadent.org.uk>

---
drivers/serial/icom.c |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/drivers/serial/icom.c b/drivers/serial/icom.c
index 0028b6f..6e715b7 100644
--- a/drivers/serial/icom.c
+++ b/drivers/serial/icom.c
@@ -1654,4 +1654,6 @@ MODULE_DESCRIPTION("IBM iSeries Serial IOA driver");
 MODULE_SUPPORTED_DEVICE
     ("IBM iSeries 2745, 2771, 2772, 2742, 2793 and 2805 Communications adapters");
 MODULE_LICENSE("GPL");
-
+MODULE_FIRMWARE("icom_call_setup.bin");
+MODULE_FIRMWARE("icom_res_dce.bin");
+MODULE_FIRMWARE("icom_asc.bin");
