Signed-off-by: Ben Hutchings
---
drivers/staging/rtl8192e/r819xE_firmware.c | 4 ++++
1 files changed, 4 insertions(+), 0 deletions(-)

--- a/drivers/staging/rtl8192e/r819xE_firmware.c
+++ b/drivers/staging/rtl8192e/r819xE_firmware.c
@@ -350,3 +350,7 @@ download_firmware_fail:
 	return rt_status;
 
 }
+
+MODULE_FIRMWARE("RTL8192E/boot.img");
+MODULE_FIRMWARE("RTL8192E/main.img");
+MODULE_FIRMWARE("RTL8192E/data.img");
