Signed-off-by: Ben Hutchings
---
drivers/staging/sep/sep_driver.c | 3 +++
1 files changed, 3 insertions(+), 0 deletions(-)

--- a/drivers/staging/sep/sep_driver.c
+++ b/drivers/staging/sep/sep_driver.c
@@ -222,6 +222,9 @@ static int sep_load_firmware(struct sep_
 	return 0;
 }
 
+MODULE_FIRMWARE("sep/cache.image.bin");
+MODULE_FIRMWARE("sep/resident.image.bin");
+
 /**
  *	sep_map_and_alloc_shared_area	-	allocate shared block
  *	@sep: security processor
