Signed-off-by: Ben Hutchings <ben@xxxxxxxxxxxxxxx>
---
 drivers/usb/atm/cxacru.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

--- a/drivers/usb/atm/cxacru.c
+++ b/drivers/usb/atm/cxacru.c
@@ -1041,6 +1041,9 @@ static int cxacru_heavy_init(struct usba
 	return ret;
 }
 
+MODULE_FIRMWARE("cxacru-fw.bin");
+MODULE_FIRMWARE("cxacru-bp.bin");
+
 static int cxacru_bind(struct usbatm_data *usbatm_instance,
 		       struct usb_interface *intf, const struct usb_device_id *id)
 {
