Signed-off-by: Ben Hutchings <ben@xxxxxxxxxxxxxxx>
---
 drivers/usb/atm/ueagle-atm.c |   23 +++++++++++++++++++++++
 1 files changed, 23 insertions(+), 0 deletions(-)

diff --git a/drivers/usb/atm/ueagle-atm.c b/drivers/usb/atm/ueagle-atm.c
index 25af0bc..2062da1 100644
--- a/drivers/usb/atm/ueagle-atm.c
+++ b/drivers/usb/atm/ueagle-atm.c
@@ -716,6 +716,13 @@ static int uea_load_firmware(struct usb_device *usb, unsigned int ver)
 	return ret;
 }
 
+MODULE_FIRMWARE(FW_DIR "eagle.fw");
+MODULE_FIRMWARE(FW_DIR "adi930.fw");
+MODULE_FIRMWARE(FW_DIR "eagleI.fw");
+MODULE_FIRMWARE(FW_DIR "eagleII.fw");
+MODULE_FIRMWARE(FW_DIR "eagleIII.fw");
+MODULE_FIRMWARE(FW_DIR "eagleIV.fw");
+
 /* modem management : dsp firmware, send/read CMV, monitoring statistic
  */
 
@@ -895,6 +902,13 @@ static int request_dsp(struct uea_softc *sc)
 	return 0;
 }
 
+MODULE_FIRMWARE(FW_DIR "DSP4i.bin");
+MODULE_FIRMWARE(FW_DIR "DSP4p.bin");
+MODULE_FIRMWARE(FW_DIR "DSP9i.bin");
+MODULE_FIRMWARE(FW_DIR "DSP9p.bin");
+MODULE_FIRMWARE(FW_DIR "DSPei.bin");
+MODULE_FIRMWARE(FW_DIR "DSPep.bin");
+
 /*
  * The uea_load_page() function must be called within a process context
  */
@@ -1571,6 +1585,13 @@ static void cmvs_file_name(struct uea_softc *sc, char *const cmv_name, int ver)
 		strlcat(cmv_name, ".v2", UEA_FW_NAME_MAX);
 }
 
+MODULE_FIRMWARE(FW_DIR "CMV4i.bin.v2");
+MODULE_FIRMWARE(FW_DIR "CMV4p.bin.v2");
+MODULE_FIRMWARE(FW_DIR "CMV9i.bin.v2");
+MODULE_FIRMWARE(FW_DIR "CMV9p.bin.v2");
+MODULE_FIRMWARE(FW_DIR "CMVei.bin.v2");
+MODULE_FIRMWARE(FW_DIR "CMVep.bin.v2");
+
 static int request_cmvs_old(struct uea_softc *sc,
 		 void **cmvs, const struct firmware **fw)
 {
@@ -1945,6 +1966,8 @@ err0:
 	return ret;
 }
 
+MODULE_FIRMWARE(FW_DIR "930-fpga.bin");
+
 /* The modem send us an ack. First with check if it right */
 static void uea_dispatch_cmv_e1(struct uea_softc *sc, struct intr_pkt *intr)
 {
-- 
1.6.5.2
