From patchwork Sat Nov  7 22:12:50 2009
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Subject: [74/75] metronomefb: declare MODULE_FIRMWARE
Date: Sat, 07 Nov 2009 22:12:50 -0000
From: Ben Hutchings <ben@decadent.org.uk>
X-Patchwork-Id: 58440

Signed-off-by: Ben Hutchings <ben@decadent.org.uk>

---
drivers/video/metronomefb.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

--- a/drivers/video/metronomefb.c
+++ b/drivers/video/metronomefb.c
@@ -792,3 +792,4 @@ module_exit(metronomefb_exit);
 MODULE_DESCRIPTION("fbdev driver for Metronome controller");
 MODULE_AUTHOR("Jaya Kumar");
 MODULE_LICENSE("GPL");
+MODULE_FIRMWARE("metronome.wbf");
