From patchwork Sat Nov  7 22:13:39 2009
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Subject: [75/75] snd-aica: declare MODULE_FIRMWARE
Date: Sat, 07 Nov 2009 22:13:39 -0000
From: Ben Hutchings <ben@decadent.org.uk>
X-Patchwork-Id: 58441

Signed-off-by: Ben Hutchings <ben@decadent.org.uk>

---
sound/sh/aica.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/sound/sh/aica.c b/sound/sh/aica.c
index 583a369..a0df401 100644
--- a/sound/sh/aica.c
+++ b/sound/sh/aica.c
@@ -49,6 +49,7 @@ MODULE_AUTHOR("Adrian McMenamin <adrian@mcmen.demon.co.uk>");
 MODULE_DESCRIPTION("Dreamcast AICA sound (pcm) driver");
 MODULE_LICENSE("GPL");
 MODULE_SUPPORTED_DEVICE("{{Yamaha/SEGA, AICA}}");
+MODULE_FIRMWARE("aica_firmware.bin");
 
 /* module parameters */
 #define CARD_NAME "AICA"
