From 1ce67353a77a260fa917086405e326693c35abf2 Mon Sep 17 00:00:00 2001
From: Boris Barbulovski <bbarbulovski@gmail.com>
Date: Tue, 22 Sep 2015 11:36:08 -0700
Subject: [PATCH 08/40] Port xconfig to Qt5 - Remove unused #include
 <q3dragobject.h>

Signed-off-by: Boris Barbulovski <bbarbulovski@gmail.com>
Signed-off-by: Thiago Macieira <thiago.macieira@intel.com>
Signed-off-by: Michal Marek <mmarek@suse.com>
---
 scripts/kconfig/qconf.cc | 1 -
 1 file changed, 1 deletion(-)

--- a/scripts/kconfig/qconf.cc
+++ b/scripts/kconfig/qconf.cc
@@ -12,7 +12,6 @@
 #include <QAction>
 #include <q3header.h>
 #include <QFileDialog>
-#include <q3dragobject.h>
 #include <QMenu>
 
 #include <qapplication.h>
