From a52cb321de597542eeab63e1033156e05d9f075b Mon Sep 17 00:00:00 2001
From: Boris Barbulovski <bbarbulovski@gmail.com>
Date: Tue, 22 Sep 2015 11:36:24 -0700
Subject: [PATCH 29/43] Port xconfig to Qt5 - Tree widget set column titles.

Signed-off-by: Boris Barbulovski <bbarbulovski@gmail.com>
Signed-off-by: Thiago Macieira <thiago.macieira@intel.com>
Signed-off-by: Michal Marek <mmarek@suse.com>
---
 scripts/kconfig/qconf.cc | 2 ++
 1 file changed, 2 insertions(+)

--- a/scripts/kconfig/qconf.cc
+++ b/scripts/kconfig/qconf.cc
@@ -318,6 +318,8 @@ ConfigList::ConfigList(ConfigView* p, co
 	setSortingEnabled(-1);
 	setRootIsDecorated(true);
 
+	setHeaderLabels(QStringList() << _("Option") << _("Name") << "N" << "M" << "Y" << _("Value"));
+
 	connect(this, SIGNAL(itemSelectionChanged(void)),
 		SLOT(updateSelection(void)));
 
