From 86c052827b68db49a70597980d0f85c8546a09be Mon Sep 17 00:00:00 2001
From: Boris Barbulovski <bbarbulovski@gmail.com>
Date: Tue, 22 Sep 2015 11:36:25 -0700
Subject: [PATCH 30/43] Port xconfig to Qt5 - Add ConfigItem::nextItem to
 initializer list.

Signed-off-by: Boris Barbulovski <bbarbulovski@gmail.com>
Signed-off-by: Thiago Macieira <thiago.macieira@intel.com>
Signed-off-by: Michal Marek <mmarek@suse.com>
---
 scripts/kconfig/qconf.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- a/scripts/kconfig/qconf.h
+++ b/scripts/kconfig/qconf.h
@@ -124,17 +124,17 @@ class ConfigItem : public QTreeWidgetIte
 	typedef class QTreeWidgetItem Parent;
 public:
 	ConfigItem(QTreeWidget *parent, ConfigItem *after, struct menu *m, bool v)
-	: Parent(parent, after), menu(m), visible(v), goParent(false)
+	: Parent(parent, after), nextItem(0), menu(m), visible(v), goParent(false)
 	{
 		init();
 	}
 	ConfigItem(ConfigItem *parent, ConfigItem *after, struct menu *m, bool v)
-	: Parent(parent, after), menu(m), visible(v), goParent(false)
+	: Parent(parent, after), nextItem(0), menu(m), visible(v), goParent(false)
 	{
 		init();
 	}
 	ConfigItem(QTreeWidget *parent, ConfigItem *after, bool v)
-	: Parent(parent, after), menu(0), visible(v), goParent(true)
+	: Parent(parent, after), nextItem(0), menu(0), visible(v), goParent(true)
 	{
 		init();
 	}
