From b3c48f964cda9311030416d1ee17bd5bdc4729f2 Mon Sep 17 00:00:00 2001
From: Boris Barbulovski <bbarbulovski@gmail.com>
Date: Tue, 22 Sep 2015 11:36:32 -0700
Subject: [PATCH 31/40] Port xconfig to Qt5 - Source format.

Signed-off-by: Boris Barbulovski <bbarbulovski@gmail.com>
Signed-off-by: Thiago Macieira <thiago.macieira@intel.com>
Signed-off-by: Michal Marek <mmarek@suse.com>
---
 scripts/kconfig/qconf.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- a/scripts/kconfig/qconf.h
+++ b/scripts/kconfig/qconf.h
@@ -157,10 +157,10 @@ public:
 		ConfigItem *_parent = (ConfigItem *)parent();
 
 		if(_parent) {
-		  ret = (ConfigItem *)_parent->child(_parent->indexOfChild(this)+1);
+			ret = (ConfigItem *)_parent->child(_parent->indexOfChild(this)+1);
 		} else {
-		  QTreeWidget *_treeWidget = treeWidget();
-		  ret = (ConfigItem *)_treeWidget->topLevelItem(_treeWidget->indexOfTopLevelItem(this)+1);
+			QTreeWidget *_treeWidget = treeWidget();
+			ret = (ConfigItem *)_treeWidget->topLevelItem(_treeWidget->indexOfTopLevelItem(this)+1);
 		}
 
 		return ret;
