From e336b9f1325acb65121c6911f6f0bd057073db6a Mon Sep 17 00:00:00 2001
From: Boris Barbulovski <bbarbulovski@gmail.com>
Date: Tue, 22 Sep 2015 11:36:34 -0700
Subject: [PATCH 38/43] Port xconfig to Qt5 - Add(back) lineedit editing.

Signed-off-by: Boris Barbulovski <bbarbulovski@gmail.com>
Signed-off-by: Thiago Macieira <thiago.macieira@intel.com>
Signed-off-by: Michal Marek <mmarek@suse.com>
---
 scripts/kconfig/qconf.cc | 1 +
 1 file changed, 1 insertion(+)

--- a/scripts/kconfig/qconf.cc
+++ b/scripts/kconfig/qconf.cc
@@ -518,6 +518,7 @@ void ConfigList::changeValue(ConfigItem*
 	case S_INT:
 	case S_HEX:
 	case S_STRING:
+		parent()->lineEdit->show(item);
 		break;
 	}
 }
