From 621ce73313e9d0cb855534ca7315c4c4005d9f2d Mon Sep 17 00:00:00 2001
From: Peter De Schrijver <p2@debian.org>
Date: Mon, 1 Jul 2013 02:18:10 +0200
Subject: [PATCH 03/65] acerhk: add support for Medion MD97600

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 drivers/platform/x86/acerhk.c | 25 ++++++++++++++++++++++++-
 1 file changed, 24 insertions(+), 1 deletion(-)

--- a/drivers/platform/x86/acerhk.c
+++ b/drivers/platform/x86/acerhk.c
@@ -1238,6 +1238,17 @@ static void __init setup_keymap_model(un
     acerhk_key2name[5]  = k_display;    /* FN+F3 (Display switch) */
     acerhk_key2name[6]  = k_res;        /* FN+F4 (Display ein/ausschalten) */
     break;
+  case 97600:
+    /* Medion MD97600, 7 keys, no setup */
+    acerhk_key2name[1]  = k_help;       /* FN+F1 (Help) */
+    acerhk_key2name[2]	= k_none;
+    acerhk_key2name[5]  = k_display;    /* FN+F3 (Display switch) */
+    acerhk_key2name[6]  = k_res;        /* FN+F4 (Display ein/ausschalten) */
+    acerhk_key2name[17] = k_p1;
+    acerhk_key2name[18] = k_p2;
+    acerhk_key2name[19] = k_p3;
+    acerhk_key2name[48] = k_wireless;
+    break;
   case 42200:
     /* Medion MD42200, 7 keys, no setup */
     acerhk_key2name[2] = k_none;
@@ -1336,6 +1347,14 @@ static void __init setup_model_features(
     acerhk_model_features = 0x00f00000;
     acerhk_type = TM_new;
     break;
+  case 97600:
+    /* has WLAN button */
+    /* The MD97600 seems to require TM_F_CONNECT at least
+       once after cold boot, otherwise enabling the WLAN
+       radio does not work */
+    acerhk_model_features = TM_F_WBUTTON | TM_F_CONNECT;
+    acerhk_type = TM_new;
+    break;
   case 42200: /* Medion MD42200 */
     /* has WLAN button, should call connect() */
     acerhk_model_features = TM_F_WBUTTON | TM_F_CONNECT;
@@ -1907,7 +1926,11 @@ static unsigned int __init determine_lap
   }
   else if (strncmp(str, "MEDIONPC", 8) == 0) {
     uint medionmodel;
-    if ((medionmodel = COLUSSI("WIM 2040", 4, reg1, AREA_SIZE)) >= 0) {
+    if ((medionmodel = COLUSSI("WIM 2090", 8, reg1, AREA_SIZE)) >= 0) {
+      printk(KERN_INFO"acerhk: found Medion model string:'%s'\n", (char*)reg1+medionmodel);
+      series = 97600;
+    }
+    else if ((medionmodel = COLUSSI("WIM 2040", 4, reg1, AREA_SIZE)) >= 0) {
       printk(KERN_INFO"acerhk: found Medion model string:'%s'\n", (char*)reg1+medionmodel);
       series = 96500;
     } else {
