From c48624d2f7dfbd3d85635d91d389bcd6aa65176e Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Mon, 1 Jul 2013 02:20:13 +0200
Subject: [PATCH 05/51] acerhk: add kbuild hooks

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 drivers/platform/x86/Kconfig  | 34 ++++++++++++++++++++++++++++++++++
 drivers/platform/x86/Makefile |  1 +
 2 files changed, 35 insertions(+)

--- a/drivers/platform/x86/Kconfig
+++ b/drivers/platform/x86/Kconfig
@@ -55,6 +55,40 @@ config ACERHDF
 	  If you have an Acer Aspire One netbook, say Y or M
 	  here.
 
+config ACERHK
+	tristate "Acer Travelmate laptop special key support"
+	depends on X86
+	---help---
+	  This is an experimental keyboard driver for acer laptops. If you
+	  have a notebook with a ipw2X00 wireless card, it allows you to
+	  turn off rf_kill.
+
+	  Supported Acer Aspire series:
+
+	    * 1300/ 1310/ 1350/ 1360
+	    * 1400/ 1450
+	    * 1800
+	    * 2000/ 2010/ 2020
+	    * 4150
+	    * 4650
+
+	  Supported Acer Travelmate series:
+	    * 290
+	    * 420/ 430
+	    * 530/ 540
+	    * 650/ 660
+	    * 800
+	    * 1700
+	    * 2300/ 2350
+	    * 3200
+	    * 4000/ 4050
+	    * 4500
+
+	  This module adds support for various special keys and LEDs on these
+	  notebooks, most importantly rfkill for wlan and bluetooth.
+
+	  If you have one of these laptops, say Y or M here.
+
 config ASUS_LAPTOP
 	tristate "Asus Laptop Extras"
 	depends on ACPI
--- a/drivers/platform/x86/Makefile
+++ b/drivers/platform/x86/Makefile
@@ -15,6 +15,7 @@ obj-$(CONFIG_DELL_WMI)		+= dell-wmi.o
 obj-$(CONFIG_DELL_WMI_AIO)	+= dell-wmi-aio.o
 obj-$(CONFIG_ACER_WMI)		+= acer-wmi.o
 obj-$(CONFIG_ACERHDF)		+= acerhdf.o
+obj-$(CONFIG_ACERHK)		+= acerhk.o
 obj-$(CONFIG_HP_ACCEL)		+= hp_accel.o
 obj-$(CONFIG_HP_WMI)		+= hp-wmi.o
 obj-$(CONFIG_AMILO_RFKILL)	+= amilo-rfkill.o
