From 706af0cda1ff5cd84a6fce17ae974b215b977d22 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Mon, 1 Jul 2013 02:22:38 +0200
Subject: [PATCH 08/65] acerhk: add compatibility for kernel >= 2.6.36

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 drivers/platform/x86/acerhk.c | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

--- a/drivers/platform/x86/acerhk.c
+++ b/drivers/platform/x86/acerhk.c
@@ -2755,8 +2755,8 @@ static void acerhk_proc_cleanup(void)
 
 /* {{{ file operations */
 
-static int acerhk_ioctl(struct inode *inode, struct file *file,
-			unsigned int cmd, unsigned long arg)
+static long acerhk_unlocked_ioctl(struct file *file, unsigned int cmd,
+				  unsigned long arg)
 {
 	int retval;
 	switch (cmd) {
@@ -2862,11 +2862,11 @@ static int acerhk_resume(struct platform
 }
 #endif
 
-static struct file_operations acerhk_fops = {
-owner:	THIS_MODULE,
-ioctl:	acerhk_ioctl,
-open:	acerhk_open,
-release:acerhk_release,
+static const struct file_operations acerhk_fops = {
+owner:		THIS_MODULE,
+unlocked_ioctl:	acerhk_unlocked_ioctl,
+open:		acerhk_open,
+release:	acerhk_release,
 };
 
 static struct miscdevice acerhk_misc_dev = {
