From 8605bba8060c3cf43385320f67e6f77262e2b6d9 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Mon, 1 Jul 2013 02:23:07 +0200
Subject: [PATCH 09/65] acerhk: add compatibility for kernel >= 2.6.38

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 drivers/platform/x86/acerhk.c | 4 ----
 1 file changed, 4 deletions(-)

--- a/drivers/platform/x86/acerhk.c
+++ b/drivers/platform/x86/acerhk.c
@@ -35,10 +35,6 @@
  *
  */
 
-#ifndef AUTOCONF_INCLUDED
-#include <linux/config.h>
-#endif
-
 /* This driver is heavily dependent on the architecture, don't let
  * anyone without an X86 machine use it. On laptops with AMD64
  * architecture this driver is only useable in 32 bit mode.
