From fdf1474c7596f1707e66323b2d46199b913635db Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Mon, 1 Jul 2013 02:23:52 +0200
Subject: [PATCH 11/65] acerhk: remove undefined function calls on x86_64

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 drivers/platform/x86/acerhk.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

--- a/drivers/platform/x86/acerhk.c
+++ b/drivers/platform/x86/acerhk.c
@@ -635,6 +635,7 @@ static int wbutton_fct_2(int val)
 	return regs.eax & 0xffff;
 }
 
+#if !(defined(DUMMYHW) || defined(__x86_64__))
 /* get_cmos_index
  *
  * gets index of CMOS port from ROM. The number of events is monitored
@@ -656,6 +657,7 @@ static int get_cmos_index(void)
 		       cmos_index);
 	return cmos_index;
 }
+#endif
 
 /* get_nr_events
  *
@@ -758,6 +760,7 @@ static int launch_connect(int val)
 
 static struct proc_dir_entry *proc_acer_dir;
 
+#if !(defined(DUMMYHW) || defined(__x86_64__))
 static unsigned long __init find_hk_area(void)
 {
 	long offset, sig;
@@ -787,6 +790,7 @@ static unsigned long __init find_hk_area
 	}
 	return fkt;
 }
+#endif
 
 static void print_features(void)
 {
@@ -1172,6 +1176,7 @@ static void __init setup_model_features(
 		print_features();
 }
 
+#if !(defined(DUMMYHW) || defined(__x86_64__))
 static unsigned int __init determine_laptop_series(char *str)
 {
 	/* 0 means unknown series, handled like TM 200 */
@@ -1618,7 +1623,9 @@ static unsigned int __init determine_lap
 	}
 	return (series);
 }
+#endif
 
+#if !(defined(DUMMYHW) || defined(__x86_64__))
 static void __init probe_model(void)
 {
 	int offset;		/* offset from beginning of reg1 to Model string */
@@ -1671,6 +1678,7 @@ static void __init probe_model(void)
 		acerhk_series = 200;
 	}
 }
+#endif
 
 /* }}} */
 
@@ -1752,6 +1760,7 @@ static int filter_idle_value(int keycode
 	return validkey;
 }
 
+#if !(defined(DUMMYHW) || defined(__x86_64__))
 static void send_key_event(t_key_names key)
 {
 	unsigned int input_key;
@@ -1767,7 +1776,9 @@ static void send_key_event(t_key_names k
 		input_report_key(acerhk_input_dev_ptr, input_key, 0);
 	}
 }
+#endif
 
+#if !(defined(DUMMYHW) || defined(__x86_64__))
 static t_key_names transl8_key_code(int keycode)
 {
 	t_key_names keyname = k_none;
@@ -1815,6 +1826,7 @@ static t_key_names transl8_key_code(int
 	}
 	return keyname;
 }
+#endif
 
 /* polling timer handler */
 static void acerhk_poll_event(unsigned long save_size)
@@ -2422,6 +2434,7 @@ static struct miscdevice acerhk_misc_dev
 
 /* }}} */
 
+#if !(defined(DUMMYHW) || defined(__x86_64__))
 static void __devinit model_init(void)
 {
 	/* set callroutine, features and keymap for model */
@@ -2448,6 +2461,7 @@ static void __devinit model_init(void)
 	/* added by Antonio Cuni */
 	init_timer(&acerhk_timer_blinking);
 }
+#endif
 
 static int __devexit acerhk_remove(struct platform_device *dev);
 
