From 9bce2da66e64208806d1979baa08a04592f72024 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Mon, 1 Jul 2013 02:24:14 +0200
Subject: [PATCH 12/65] acerhk: kill TM_old

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 drivers/platform/x86/acerhk.c | 81 -------------------------------------------
 1 file changed, 81 deletions(-)

--- a/drivers/platform/x86/acerhk.c
+++ b/drivers/platform/x86/acerhk.c
@@ -110,8 +110,6 @@ static struct input_dev acerhk_input_dev
 static void *reg1;
 /* mapped IO area from 0xe0000 */
 static void *reg2;
-/* Pointer to mapped area at 0x400 on 520 series */
-static void *preg400;
 /* location of IO routine in mapped area */
 static unsigned int bios_routine;
 /* index of CMOS port to get key event */
@@ -514,7 +512,6 @@ static asmlinkage void call_bios_52x(str
 				     "movl %%edi, 16(%%ebp)\n\t"
 				     "movl %%esi, 20(%%ebp)\n\t"
 				     "popa\n\t"::"m"(bios_routine),
-				     "m"(preg400), "m"(buf)
 				     :"%eax", "%ebx", "%ecx", "%edx", "%edi",
 				     "%esi", "%ebp");
 		local_irq_enable();
@@ -910,18 +907,6 @@ static void __init setup_keymap_model(un
 	default:
 		/* only the two common keys are supported */
 		break;
-	case 210:
-		acerhk_key2name[19] = k_p1;
-		acerhk_key2name[20] = k_p2;
-		acerhk_key2name[17] = k_www;
-		acerhk_key2name[18] = k_mail;
-		break;
-	case 520:
-		acerhk_key2name[19] = k_p1;
-		acerhk_key2name[20] = k_p2;
-		acerhk_key2name[17] = k_www;
-		acerhk_key2name[18] = k_mail;
-		break;
 	case 290:
 	case 420:
 	case 430:
@@ -1008,27 +993,12 @@ static void __init setup_keymap_model(un
 		acerhk_key2name[110] = k_setup;	//This is the Fancy Fan (cool-n'-quiet) key on A1650g
 		acerhk_key2name[48] = k_wireless;
 		break;
-	case 98200:
-		/* Medion MD98200, 4 keys, no setup */
-		acerhk_key2name[2] = k_none;
-		acerhk_key2name[48] = k_wireless;
-		acerhk_key2name[0x79] = k_play;
-		acerhk_key2name[17] = k_p1;
-		acerhk_key2name[18] = k_p2;
-		break;
 	}
 }
 
 static void __init setup_model_features(unsigned int series)
 {
 	switch (series) {
-	case 200:
-	case 210:
-	case 520:
-		/* nothing special */
-		acerhk_model_features = 0;
-		acerhk_type = TM_old;
-		break;
 	case 97600:
 		/* has WLAN button */
 		/* The MD97600 seems to require TM_F_CONNECT at least
@@ -1037,11 +1007,6 @@ static void __init setup_model_features(
 		acerhk_model_features = TM_F_WBUTTON | TM_F_CONNECT;
 		acerhk_type = TM_new;
 		break;
-	case 98200:		/* Medion MD98200 */
-		/* has WLAN button, should call connect() */
-		acerhk_model_features = TM_F_WBUTTON;
-		acerhk_type = TM_old;
-		break;
 	case 6800:
 		/* mail led and all special functions for FS Amilo D */
 		acerhk_model_features = TM_F_MAIL_LED | 0x00f00000;
@@ -1207,21 +1172,6 @@ static unsigned int __init determine_lap
 						       "acerhk: model string indicates unknown TM 2xxx series\n");
 					break;
 				}
-			} else {
-				/* older Travelmate 2xx series */
-				switch (str[12]) {
-				case '0':
-					series = 200;
-					break;
-				case '1':
-					series = 210;
-					break;
-				default:
-					if (verbose > 1)
-						printk(KERN_INFO
-						       "acerhk: model string indicates unknown TM 2xx series\n");
-					break;
-				}
 			}
 			break;
 		case '3':
@@ -1270,13 +1220,6 @@ static unsigned int __init determine_lap
 				}
 			}
 			break;
-		case '5':	/* Travelmate 5xx series */
-			if (str[12] == '2')
-				series = 520;
-			else if (verbose > 1)
-				printk(KERN_INFO
-				       "acerhk: model string indicates unknown TM 5xx series\n");
-			break;
 		case '6':	/* older Travelmate 6xx series */
 			switch (str[12]) {
 			default:
@@ -1587,13 +1530,6 @@ static unsigned int __init determine_lap
 		if (verbose > 1)
 			printk(KERN_INFO
 			       "acerhk: model string indicates no supported Medion hardware\n");
-	} else if (strncmp(str, "MEDION", 6) == 0) {
-		if (COLUSSI("WIM2120", 7, reg1, AREA_SIZE) >= 0) {
-			if (verbose > 1)
-				printk(KERN_INFO
-				       "acerhk: model string indicates a Medion MD 98200\n");
-			series = 98200;
-		}
 	} else if (strncmp(str, "AOpen", 5) == 0) {
 		/* Unless I know of other models no further differentiation,
 		   although there is a second part of the model string */
@@ -1672,10 +1608,6 @@ static void __init probe_model(void)
 			       "acerhk: offset from model string to function address: 0x%lx\n",
 			       bios_routine - (unsigned long)acerhk_model_addr);
 		acerhk_series = determine_laptop_series(acerhk_model_string);
-	} else {
-		printk(KERN_WARNING
-		       "acerhk: Could not find model string, will assume type 200 series\n");
-		acerhk_series = 200;
 	}
 }
 #endif
@@ -1999,9 +1931,6 @@ static int pc_proc_infos(char *buffer, i
 	default:
 		PRINT_PROC("unknown)\n");
 		break;
-	case TM_old:
-		PRINT_PROC("old)\n");
-		break;
 	case TM_new:
 		PRINT_PROC("new)\n");
 		break;
@@ -2038,8 +1967,6 @@ static int pc_proc_infos(char *buffer, i
 		else
 			PRINT_PROC("disabled\n");
 	}
-	if (acerhk_type == TM_old)
-		PRINT_PROC("preg400\t\t: 0x%p\n", preg400);
 	return (1);
 }
 
@@ -2489,12 +2416,6 @@ static int __devinit acerhk_probe(struct
 			printk(KERN_INFO
 			       "acerhk: area from 0xe000 to 0xffff mapped to %p\n",
 			       reg2);
-		/* the area 0x400 is used as data area by earlier (520) series  */
-		preg400 = ioremap(0x400, 0xfff);
-		if (verbose > 1)
-			printk(KERN_INFO
-			       "acerhk: area from 0x400 to 0x13ff mapped to %p\n",
-			       preg400);
 		/* attach to input system */
 		init_input();
 		memset(acerhk_model_string, 0x00, ACERHK_MODEL_STRLEN);
@@ -2569,8 +2490,6 @@ static int __devexit acerhk_remove(struc
 		iounmap(reg1);
 	if (reg2)
 		iounmap(reg2);
-	if (preg400)
-		iounmap(preg400);
 	release_input();
 	misc_deregister(&acerhk_misc_dev);
 	if (acerhk_type == TM_dritek) {
