From 56a677dfe82afe1bd7b374a4eaf7fd14b6aed54b Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Mon, 1 Jul 2013 02:24:39 +0200
Subject: [PATCH 13/65] acerhk: kill TM_new

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 drivers/platform/x86/acerhk.c | 83 +------------------------------------------
 1 file changed, 1 insertion(+), 82 deletions(-)

--- a/drivers/platform/x86/acerhk.c
+++ b/drivers/platform/x86/acerhk.c
@@ -964,59 +964,12 @@ static void __init setup_keymap_model(un
 		acerhk_key2name[1] = k_none;
 		acerhk_key2name[2] = k_none;
 		break;
-	case 97600:
-		/* Medion MD97600, 7 keys, no setup */
-		acerhk_key2name[1] = k_help;	/* FN+F1 (Help) */
-		acerhk_key2name[2] = k_none;
-		acerhk_key2name[5] = k_display;	/* FN+F3 (Display switch) */
-		acerhk_key2name[6] = k_res;	/* FN+F4 (Display ein/ausschalten) */
-		acerhk_key2name[17] = k_p1;
-		acerhk_key2name[18] = k_p2;
-		acerhk_key2name[19] = k_p3;
-		acerhk_key2name[48] = k_wireless;
-		break;
-	case 6800:
-		/* Amilo D does not have Setup key */
-		acerhk_key2name[2] = k_none;
-		acerhk_key2name[49] = k_mail;
-		acerhk_key2name[54] = k_www;
-		acerhk_key2name[17] = k_p1;
-		acerhk_key2name[18] = k_p2;
-		acerhk_key2name[19] = k_p3;
-		acerhk_key2name[8] = k_mute;
-		break;
-	case 6805:		/* Added by damagedspline@aim.com */
-		/* Amilo A1xxx does not have Setup key nor a mail key */
-		acerhk_key2name[2] = k_none;
-		acerhk_key2name[54] = k_www;
-		acerhk_key2name[5] = k_display;
-		acerhk_key2name[110] = k_setup;	//This is the Fancy Fan (cool-n'-quiet) key on A1650g
-		acerhk_key2name[48] = k_wireless;
-		break;
 	}
 }
 
 static void __init setup_model_features(unsigned int series)
 {
 	switch (series) {
-	case 97600:
-		/* has WLAN button */
-		/* The MD97600 seems to require TM_F_CONNECT at least
-		   once after cold boot, otherwise enabling the WLAN
-		   radio does not work */
-		acerhk_model_features = TM_F_WBUTTON | TM_F_CONNECT;
-		acerhk_type = TM_new;
-		break;
-	case 6800:
-		/* mail led and all special functions for FS Amilo D */
-		acerhk_model_features = TM_F_MAIL_LED | 0x00f00000;
-		acerhk_type = TM_new;
-		break;
-	case 6805:		/* Added by damagedspline@aim.com */
-		/* Amilo A1xxx does not have a mail led */
-		acerhk_model_features = 0x00f00000;
-		acerhk_type = TM_new;
-		break;
 	case 2350:
 	case 4050:
 		acerhk_wlan_state = 1;	// Default state is on
@@ -1122,7 +1075,7 @@ static void __init setup_model_features(
 		break;
 	}
 	/* set the correct bios call function according to type */
-	if ((acerhk_type == TM_new) || (acerhk_type == TM_dritek)) {
+	if (acerhk_type == TM_dritek) {
 		call_bios = call_bios_6xx;
 		if (verbose > 2)
 			printk(KERN_INFO "acerhk: using call_bios_6xx mode\n");
@@ -1489,39 +1442,8 @@ static unsigned int __init determine_lap
 				printk(KERN_INFO
 				       "acerhk: model string indicates unknown FS Amilo XX series\n");
 		}
-	} else if (strncmp(str, "AMILO ", 6) == 0) {
-		switch (str[6]) {
-		case 'D':	/* AMILO D 6800 P4-2000 */
-			if (verbose > 1)
-				printk(KERN_INFO
-				       "acerhk: model string indicates FS AMILO D series\n");
-			series = 6800;
-			break;
-		case 'A':	/* AMILO Axxxx - added by damagedspline@aim.com */
-			switch (str[7]) {
-			case '1':	/* AMILO A1xxx */
-				if (verbose > 1)
-					printk(KERN_INFO
-					       "acerhk: model string indicates FS AMILO A1xxx series\n");
-				series = 6805;
-				break;
-			};
-			break;
-		default:
-			if (verbose > 1)
-				printk(KERN_INFO
-				       "acerhk: model string indicates unknown FS AMILO XX series\n");
-			series = 6800;
-		}
 	} else if (strncmp(str, "MEDIONPC", 8) == 0) {
 		uint medionmodel;
-		if ((medionmodel =
-		     COLUSSI("WIM 2090", 8, reg1, AREA_SIZE)) >= 0) {
-			printk(KERN_INFO
-			       "acerhk: found Medion model string:'%s'\n",
-			       (char *)reg1 + medionmodel);
-			series = 97600;
-		}
 		if (verbose > 1)
 			printk(KERN_INFO
 			       "acerhk: model string indicates a medion MD %d\n",
@@ -1931,9 +1853,6 @@ static int pc_proc_infos(char *buffer, i
 	default:
 		PRINT_PROC("unknown)\n");
 		break;
-	case TM_new:
-		PRINT_PROC("new)\n");
-		break;
 	case TM_dritek:
 		PRINT_PROC("Dritek)\n");
 		break;
