From 9622b08a2a858e91ea338bbe196cd1b6f90fdb21 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Mon, 1 Jul 2013 02:26:14 +0200
Subject: [PATCH 15/65] acerhk: undefine DUMMYHW

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 drivers/platform/x86/acerhk.c | 30 ++++++++++++------------------
 1 file changed, 12 insertions(+), 18 deletions(-)

--- a/drivers/platform/x86/acerhk.c
+++ b/drivers/platform/x86/acerhk.c
@@ -59,8 +59,6 @@
 
 #include "acerhk.h"
 
-/* #define DUMMYHW */
-
 #define ACERHK_VERSION "0.5.35"
 #define MODULE_NAME "acerhk"
 
@@ -583,7 +581,7 @@ static int wbutton_fct_2(int val)
 	return regs.eax & 0xffff;
 }
 
-#if !(defined(DUMMYHW) || defined(__x86_64__))
+#ifndef __x86_64__
 /* get_cmos_index
  *
  * gets index of CMOS port from ROM. The number of events is monitored
@@ -620,8 +618,7 @@ static int get_nr_events(void)
 	unsigned char c = 0;
 
 	spin_lock_irqsave(&rtc_lock, flags);
-/* #ifndef DUMMYHW */
-#if !(defined(DUMMYHW) || defined(__x86_64__))
+#ifndef __x86_64__
 	if (cmos_index)
 		c = CMOS_READ(cmos_index);
 	else if (verbose > 3)
@@ -708,7 +705,7 @@ static int launch_connect(int val)
 
 static struct proc_dir_entry *proc_acer_dir;
 
-#if !(defined(DUMMYHW) || defined(__x86_64__))
+#ifndef __x86_64__
 static unsigned long __init find_hk_area(void)
 {
 	long offset, sig;
@@ -1011,7 +1008,7 @@ static void __init setup_model_features(
 		print_features();
 }
 
-#if !(defined(DUMMYHW) || defined(__x86_64__))
+#ifndef __x86_64__
 static unsigned int __init determine_laptop_series(char *str)
 {
 	/* 0 means unknown series, handled like TM 200 */
@@ -1382,7 +1379,7 @@ static unsigned int __init determine_lap
 }
 #endif
 
-#if !(defined(DUMMYHW) || defined(__x86_64__))
+#ifndef __x86_64__
 static void __init probe_model(void)
 {
 	int offset;		/* offset from beginning of reg1 to Model string */
@@ -1500,7 +1497,7 @@ static void init_keymap_input(void)
 		print_mapping();
 }
 
-#if !(defined(DUMMYHW) || defined(__x86_64__))
+#ifndef __x86_64__
 static int filter_idle_value(int keycode)
 {
 	int validkey = 0;
@@ -1515,7 +1512,7 @@ static int filter_idle_value(int keycode
 }
 #endif
 
-#if !(defined(DUMMYHW) || defined(__x86_64__))
+#ifndef __x86_64__
 static void send_key_event(t_key_names key)
 {
 	unsigned int input_key;
@@ -1533,7 +1530,7 @@ static void send_key_event(t_key_names k
 }
 #endif
 
-#if !(defined(DUMMYHW) || defined(__x86_64__))
+#ifndef __x86_64__
 static t_key_names transl8_key_code(int keycode)
 {
 	t_key_names keyname = k_none;
@@ -1586,8 +1583,7 @@ static t_key_names transl8_key_code(int
 /* polling timer handler */
 static void acerhk_poll_event(unsigned long save_size)
 {
-/* #ifndef DUMMYHW */
-#if !(defined(DUMMYHW) || defined(__x86_64__))
+#ifndef __x86_64__
 	unsigned int max = MAX_POLLING_LOOPS;
 	/* make sure not to loop more then 32 times */
 	if (!max || max > 32)
@@ -1603,8 +1599,7 @@ static void acerhk_blink_event(unsigned
 {
 	if (acerhk_blueled_blinking != -1) {
 		acerhk_blueled_blinking = !acerhk_blueled_blinking;
-/* #ifndef DUMMYHW */
-#if !(defined(DUMMYHW) || defined(__x86_64__))
+#ifndef __x86_64__
 		wbutton_fct_1(acerhk_blueled_blinking);
 #endif
 		acerhk_timer_blinking.expires =
@@ -2155,7 +2150,7 @@ static struct miscdevice acerhk_misc_dev
 
 /* }}} */
 
-#if !(defined(DUMMYHW) || defined(__x86_64__))
+#ifndef __x86_64__
 static void __devinit model_init(void)
 {
 	/* set callroutine, features and keymap for model */
@@ -2207,8 +2202,7 @@ static int __devinit acerhk_probe(struct
 		/* attach to input system */
 		init_input();
 		memset(acerhk_model_string, 0x00, ACERHK_MODEL_STRLEN);
-/* #ifdef DUMMYHW */
-#if (defined(DUMMYHW) || defined(__x86_64__))
+#ifdef __x86_64__
 		acerhk_model_addr = (void *)0x12345678;
 		/* copy the string, but not more than 15 characters */
 		strncpy(acerhk_model_string, "TravelmateDummy",
