From 929d481283bb5f46bac87234c20511286310c685 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Mon, 1 Jul 2013 02:26:30 +0200
Subject: [PATCH 16/65] acerhk: drop obsolete ACERHK_VERSION

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 drivers/platform/x86/acerhk.c | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

--- a/drivers/platform/x86/acerhk.c
+++ b/drivers/platform/x86/acerhk.c
@@ -59,7 +59,6 @@
 
 #include "acerhk.h"
 
-#define ACERHK_VERSION "0.5.35"
 #define MODULE_NAME "acerhk"
 
 /* maximum number of polling loops, adjust it if needed to values between
@@ -1736,7 +1735,7 @@ static int acerhk_proc_init(void)
 static int pc_proc_infos(char *buffer, int *len,
 			 off_t * begin, off_t offset, int size)
 {
-	PRINT_PROC("Acer hotkeys version %s\n", ACERHK_VERSION);
+	PRINT_PROC("Acer hotkeys\n");
 	PRINT_PROC("Model(Type)\t: %s(", acerhk_model_string);
 	PRINT_PROC("Dritek)\n");
 	if (bios_routine != 0) {
@@ -2213,8 +2212,7 @@ static int __devinit acerhk_probe(struct
 		else
 			acerhk_series = 2000;
 		setup_model_features(acerhk_series);
-		printk(KERN_INFO "Acer Travelmate hotkey driver v"
-		       ACERHK_VERSION " dummy\n");
+		printk(KERN_INFO "Acer Travelmate hotkey driver dummy\n");
 		enable_dritek_keyboard();
 		if (poll)
 			start_polling();
@@ -2239,8 +2237,7 @@ static int __devinit acerhk_probe(struct
 			printk(KERN_INFO
 			       "acerhk: bios routine found at 0x%x\n",
 			       bios_routine);
-		printk(KERN_INFO "Acer Travelmate hotkey driver v"
-		       ACERHK_VERSION "\n");
+		printk(KERN_INFO "Acer Travelmate hotkey driver\n");
 		/* If automatic switching of wlan is wanted but polling is disabled,
 		   automatically enable it */
 		if (!poll && autowlan) {
