From 81046f5e93e1c3f9042238cfabed29ed77cd6e0b Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Mon, 1 Jul 2013 02:27:10 +0200
Subject: [PATCH 17/65] acerhk: use KBUILD_MODNAME

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 drivers/platform/x86/acerhk.c | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

--- a/drivers/platform/x86/acerhk.c
+++ b/drivers/platform/x86/acerhk.c
@@ -59,8 +59,6 @@
 
 #include "acerhk.h"
 
-#define MODULE_NAME "acerhk"
-
 /* maximum number of polling loops, adjust it if needed to values between
  * 1 and 32
  */
@@ -2143,7 +2141,7 @@ release:	acerhk_release,
 
 static struct miscdevice acerhk_misc_dev = {
 	.minor = MISC_DYNAMIC_MINOR,
-	.name = "acerhk",
+	.name = KBUILD_MODNAME,
 	.fops = &acerhk_fops,
 };
 
@@ -2272,9 +2270,9 @@ static int __devexit acerhk_remove(struc
 
 static struct platform_driver acerhk_driver = {
 	.driver = {
-		   .name = "acerhk",
-		   .owner = THIS_MODULE,
-		   },
+		.name = KBUILD_MODNAME,
+		.owner = THIS_MODULE,
+	},
 	.probe = acerhk_probe,
 	.remove = __devexit_p(acerhk_remove),
 #ifdef CONFIG_PM
