From fd2a9bbebdd054440841456802cb8081f3419402 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Mon, 1 Jul 2013 02:27:33 +0200
Subject: [PATCH 18/65] acerhk: add myself

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 drivers/platform/x86/acerhk.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/drivers/platform/x86/acerhk.c
+++ b/drivers/platform/x86/acerhk.c
@@ -2,7 +2,8 @@
  * Filename:      acerhk.c
  * Version:       0.5
  *
- * Copyright (C) 2002-2007, Olaf Tauber (olaf-tauber@versanet.de)
+ * Copyright (C) 2002-2007, Olaf Tauber <olaf-tauber@versanet.de>
+ * Copyright (C) 2009-2013, Stefan Lippers-Hollmann <s.l-h@gmx.de>
  *
  * Description:   kernel driver for Acer Travelmate and similar
  *                laptops special keys
@@ -2320,6 +2321,7 @@ module_init(acerhk_init);
 module_exit(acerhk_exit);
 
 MODULE_AUTHOR("Olaf Tauber");
+MODULE_AUTHOR("Stefan Lippers-Hollmann <s.l-h@gmx.de>");
 MODULE_DESCRIPTION("AcerHotkeys extra buttons keyboard driver");
 MODULE_LICENSE("GPL");
 
