From 255f726108167b94d5e6feb2acf349563b800adb Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Mon, 1 Jul 2013 02:28:20 +0200
Subject: [PATCH 20/65] acerhk: drop removed vendors

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 drivers/platform/x86/acerhk.c | 56 +------------------------------------------
 1 file changed, 1 insertion(+), 55 deletions(-)

--- a/drivers/platform/x86/acerhk.c
+++ b/drivers/platform/x86/acerhk.c
@@ -1331,38 +1331,6 @@ static unsigned int __init determine_lap
 				series);
 		else if (verbose > 1)
 			pr_info("model string indicates unknown Extensa series\n");
-	} else if (strncmp(str, "Amilo ", 6) == 0) {
-		switch (str[6]) {
-		default:
-			if (verbose > 1)
-				pr_info("model string indicates unknown FS Amilo XX series\n");
-		}
-	} else if (strncmp(str, "MEDIONPC", 8) == 0) {
-		if (verbose > 1)
-			pr_info("model string indicates a medion MD %d\n",
-				series);
-	} else if (strncmp(str, "MEDIONNB", 8) == 0) {
-		if (verbose > 1)
-			pr_info("model string indicates no supported Medion hardware\n");
-	} else if (strncmp(str, "AOpen", 5) == 0) {
-		/* Unless I know of other models no further differentiation,
-		   although there is a second part of the model string */
-		if (verbose > 1)
-			pr_info("model string indicates a AOpen\n");
-	} else if (strncmp(str, "CL56", 4) == 0) {
-		/* Unless I know of other models no further differentiation,
-		   although there are strings with more numbers ("CL561" on a Compal
-		   CL56/Zepto 4200, reported by Stian B. Barmen)
-		   It has the same functions as Acer Aspire 2010
-		 */
-		if (verbose > 1)
-			pr_info("model string indicates a Compal CL56 (or similar)\n");
-		series = 2010;
-	} else if (strncmp(str, "Geneva2", 7) == 0) {
-		/* This might be an Aspire 9110 which is very similar to 4650 */
-		if (verbose > 1)
-			pr_info("model string indicates an Aspire 9110\n");
-		series = 4650;
 	} else {
 		if (verbose > 1)
 			pr_info("model string indicates no supported hardware\n");
@@ -1380,7 +1348,7 @@ static void __init probe_model(void)
 		pr_info("start search for model string at %p\n", reg1);
 
 	/* first we look for Travelmate, if it isn't one we try to identify other
-	   laptops, such as Medion or Aspire */
+	   laptops, such as Aspire */
 	offset = COLUSSI("Travel", 6, reg1, AREA_SIZE);
 
 	/* Try to detect Aspire laptops */
@@ -1391,28 +1359,6 @@ static void __init probe_model(void)
 	if (offset < 0)
 		offset = COLUSSI("Extensa", 7, reg1, AREA_SIZE);
 
-	/* Try to detect Medion laptops */
-	if (offset < 0)
-		offset = COLUSSI("MEDION", 6, reg1, AREA_SIZE);
-	/* Try to detect AOpen laptops */
-
-	if (offset < 0)
-		offset = COLUSSI("AOpen", 5, reg1, AREA_SIZE);
-
-	/* Try to detect Fujitsu Siemens Amilo laptops */
-	if (offset < 0)
-		offset = COLUSSI("Amilo", 5, reg1, AREA_SIZE);
-	if (offset < 0)
-		offset = COLUSSI("AMILO", 5, reg1, AREA_SIZE);
-
-	/* Try to detect Compal */
-	if (offset < 0)
-		offset = COLUSSI("CL56", 4, reg1, AREA_SIZE);
-
-	/* That might be an Aspire 9110 */
-	if (offset < 0)
-		offset = COLUSSI("Geneva2", 7, reg1, AREA_SIZE);
-
 	if (offset >= 0) {
 		acerhk_model_addr = reg1 + offset;
 		/* copy the string, but not more than 15 characters */
