From e4977607e0f6b0ed099edf07bcb7ecd3c8131d6a Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Mon, 1 Jul 2013 02:28:46 +0200
Subject: [PATCH 21/65] acerhk: drop unused case

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 drivers/platform/x86/acerhk.c | 85 ++++++++++++-------------------------------
 1 file changed, 23 insertions(+), 62 deletions(-)

--- a/drivers/platform/x86/acerhk.c
+++ b/drivers/platform/x86/acerhk.c
@@ -872,68 +872,29 @@ static void __init setup_keymap_model(un
 {
 	/* clear mapping keycode -> keyname, */
 	memset(&acerhk_key2name[0], k_none, sizeof(acerhk_key2name));
-	/* first set the common keys, namely FnF1 and FnF2, */
-	acerhk_key2name[1] = k_help;
-	acerhk_key2name[2] = k_setup;
-	/* then set known keycodes according to model */
-	switch (series) {
-	default:
-		/* only the two common keys are supported */
-		break;
-	case 290:
-	case 420:
-	case 430:
-	case 530:
-	case 540:
-	case 650:
-	case 660:
-	case 800:
-	case 1450:
-	case 2300:
-	case 2350:
-	case 4000:
-	case 4050:
-	case 6000:
-	case 8000:
-	case 4150:
-	case 4500:
-	case 4650:
-		/* keys are handled by dritek EC */
-		acerhk_key2name[1] = k_none;
-		acerhk_key2name[2] = k_none;
-		break;
-	case 1300:
-	case 1310:
-	case 1350:
-	case 1360:
-	case 1400:
-	case 1700:
-	case 1800:
-	case 2000:
-	case 2010:
-	case 2020:
-	case 5100:
-		/* Aspire 13xx series laptops use dritek hardware, no
-		   acerhk-mapping needed
-		   VolUp and VolDown are managed as normal keys
-		   1300/1310 series should have P1, P2, Mail, WWW, Mute buttons
-		   1353 has bluetooth, wifi, p1, p2, www, mail, help, setup, power
-		   and mute
-		   Aspire 1400/1450/Ferrari use dritek EC, too
-		   1450 should have bluetooth, wifi, p1, p2, www, mail, help,
-		   setup, power and mute
-		   Aspire 1700 uses dritek EC, too
-		   1700 should have bluetooth, wifi, p1, p2, www, mail, help,
-		   setup, power and mute
-		   need the MM-buttons Activation? (forward, shuffle, ...)
-		   2000 hast lots of MM buttons
-		   2010 should have bluetooth, wifi, p1, p2, www, mail, help,
-		   setup, power and mute
-		 */
-		acerhk_key2name[1] = k_none;
-		acerhk_key2name[2] = k_none;
-		break;
-	}
+
+	/*
+	 * keys are handled by dritek EC
+	 *
+	 * Aspire 13xx series laptops use dritek hardware, no
+	 * acerhk-mapping needed
+	 * VolUp and VolDown are managed as normal keys
+	 * 1300/1310 series should have P1, P2, Mail, WWW, Mute buttons
+	 * 1353 has bluetooth, wifi, p1, p2, www, mail, help, setup, power
+	 * and mute
+	 * Aspire 1400/1450/Ferrari use dritek EC, too
+	 * 1450 should have bluetooth, wifi, p1, p2, www, mail, help,
+	 * setup, power and mute
+	 * Aspire 1700 uses dritek EC, too
+	 * 1700 should have bluetooth, wifi, p1, p2, www, mail, help,
+	 * setup, power and mute
+	 * need the MM-buttons Activation? (forward, shuffle, ...)
+	 * 2000 hast lots of MM buttons
+	 * 2010 should have bluetooth, wifi, p1, p2, www, mail, help,
+	 * setup, power and mute
+	 */
+	acerhk_key2name[1] = k_none;
+	acerhk_key2name[2] = k_none;
 }
 
 static void __init setup_model_features(unsigned int series)
