From 069aaccf353448a84db9b2f3674f8369bcae7109 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Mon, 1 Jul 2013 02:31:44 +0200
Subject: [PATCH 26/65] acerhk: invert arch ifdef, to follow the normal code
 flow

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 drivers/platform/x86/acerhk.c | 32 ++++++++++++++++----------------
 1 file changed, 16 insertions(+), 16 deletions(-)

--- a/drivers/platform/x86/acerhk.c
+++ b/drivers/platform/x86/acerhk.c
@@ -1722,22 +1722,7 @@ static int __devinit acerhk_probe(struct
 
 		/* attach to input system */
 		init_input();
-#ifdef __x86_64__
-		/* set callroutine for model */
-		if (force_series)
-			acerhk_series = force_series;
-		else
-			acerhk_series = 2000;
-
-		setup_model_features(acerhk_series);
-		pr_info("Acer Travelmate hotkey driver dummy\n");
-		enable_dritek_keyboard();
-
-		if (poll)
-			start_polling();
-
-		init_timer(&acerhk_timer_blinking);
-#else
+#ifndef __x86_64__
 		bios_routine = find_hk_area();
 		if (!force_series)
 			probe_model();
@@ -1765,6 +1750,21 @@ static int __devinit acerhk_probe(struct
 		 * routine found */
 		if (poll && bios_routine)
 			start_polling();
+#else
+		/* set callroutine for model */
+		if (force_series)
+			acerhk_series = force_series;
+		else
+			acerhk_series = 2000;
+
+		setup_model_features(acerhk_series);
+		pr_info("Acer Travelmate hotkey driver dummy\n");
+		enable_dritek_keyboard();
+
+		if (poll)
+			start_polling();
+
+		init_timer(&acerhk_timer_blinking);
 #endif
 	}
 
