From 5a8fb7f279b11227bc78977cd9d718dc40671c3c Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Tue, 2 Jul 2013 02:09:34 +0200
Subject: [PATCH 32/65] acerhk: cosmetic fixes for remove_proc_entry()
 formatting

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 drivers/platform/x86/acerhk.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

--- a/drivers/platform/x86/acerhk.c
+++ b/drivers/platform/x86/acerhk.c
@@ -1418,6 +1418,7 @@ static int acerhk_proc_init(void)
 		pr_info("cannot create LED file\n");
 		remove_proc_entry("key", proc_acer_dir);
 		remove_proc_entry("driver/acerhk", NULL);
+
 		return 0;
 	}
 
@@ -1426,10 +1427,8 @@ static int acerhk_proc_init(void)
 	entry = create_proc_entry("wirelessled", 0220, proc_acer_dir);
 	if (entry == NULL) {
 		pr_info("cannot create wirelessled file\n");
-		remove_proc_entry("key",
-				 proc_acer_dir);
-		remove_proc_entry("led",
-				 proc_acer_dir);
+		remove_proc_entry("key", proc_acer_dir);
+		remove_proc_entry("led", proc_acer_dir);
 		remove_proc_entry("driver/acerhk", NULL);
 
 		return 0;
