From 1103a4f1b030a18da3401c03e2eb8be77ec9e601 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Tue, 2 Jul 2013 02:31:33 +0200
Subject: [PATCH 33/65] acerhk: cosmetic fixes for Aspire 1601 LC keys

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 drivers/platform/x86/acerhk.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

--- a/drivers/platform/x86/acerhk.c
+++ b/drivers/platform/x86/acerhk.c
@@ -998,9 +998,13 @@ static int filter_idle_value(int keycode
 	int validkey = 0;
 
 	/* Francois Valenduc, Aspire 1601 LC */
-	if (keycode != 0x0 && keycode != 0x9610 && keycode != 0xc100 &&
+	if (keycode != 0x0000 &&
+	    keycode != 0x9610 &&
+	    keycode != 0xc100 &&
 	    keycode != 0x8610 &&
-	    keycode != 0x861 && keycode != 0x8650 && keycode != 0x865)
+	    keycode != 0x0861 &&
+	    keycode != 0x8650 &&
+	    keycode != 0x0865)
 		validkey = keycode;
 
 	if (verbose > 4 && !validkey)
