From 261efd94e41bcfcdfdf333140043671c8e1ef5cf Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Tue, 13 Aug 2013 22:24:37 +0200
Subject: [PATCH 42/65] acerhk: fix cosmetic checkpatch complaints

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 drivers/platform/x86/acerhk.c | 22 ++++++++++++----------
 1 file changed, 12 insertions(+), 10 deletions(-)

--- a/drivers/platform/x86/acerhk.c
+++ b/drivers/platform/x86/acerhk.c
@@ -694,9 +694,9 @@ static unsigned long __init find_hk_area
 		}
 	}
 
-	if (sig < 0)
+	if (sig < 0) {
 		pr_warn("could not find request handler, possibly not all functions available\n");
-	else {
+	} else {
 		/* compute location of bios routine */
 		fkt = readl(reg1 + sig + 5);
 
@@ -1109,12 +1109,12 @@ static t_key_names transl8_key_code(int
 			pr_info("received key code 0x%x\n", keycode);
 
 		/* translate keycode to key name */
-		if (keycode >= 0 && keycode <= 255)
+		if (keycode >= 0 && keycode <= 255) {
 			keyname = acerhk_key2name[keycode];
-		else {
+		} else {
 			if (verbose >= 4)
 				pr_info("keycode 0x%x too big, will use only 8 bits\n",
-				        keycode);
+					keycode);
 
 			/* use only lower 8 bits of value to distinguish keys */
 			keyname = acerhk_key2name[keycode & 0xff];
@@ -1195,8 +1195,9 @@ static void stop_polling(void)
 			pr_info("key polling stopped\n");
 
 		acerhk_polling_state = 0;
-	} else if (verbose)
+	} else if (verbose) {
 		pr_info("key polling not active\n");
+	}
 }
 
 static void start_polling(void)
@@ -1216,8 +1217,9 @@ static void start_polling(void)
 		if (verbose)
 			pr_info("starting key polling, every %d ms\n",
 				acerhk_polling_delay);
-	} else if (verbose)
+	} else if (verbose) {
 		pr_info("key polling already active\n");
+	}
 }
 
 
@@ -1287,7 +1289,7 @@ static long acerhk_unlocked_ioctl(struct
 		break;
 	case ACERHK_GET_KEY_MAP:
 		if (copy_to_user
-		    ((t_map_name2event *) arg, &acerhk_name2event,
+		    ((t_map_name2event *)arg, &acerhk_name2event,
 		     sizeof(acerhk_name2event)))
 			retval = -EFAULT;
 		else
@@ -1295,7 +1297,7 @@ static long acerhk_unlocked_ioctl(struct
 		break;
 	case ACERHK_SET_KEY_MAP:
 		if (copy_from_user
-		    (&acerhk_name2event, (t_map_name2event *) arg,
+		    (&acerhk_name2event, (t_map_name2event *)arg,
 		     sizeof(acerhk_name2event)))
 			retval = -EFAULT;
 		else {
@@ -1327,7 +1329,7 @@ static int acerhk_release(struct inode *
 static int acerhk_resume(struct platform_device *dev)
 {
 	pr_info("Resuming. Setting wlan_state to: %d\n",
-	       acerhk_wlan_state);
+		acerhk_wlan_state);
 
 	if (acerhk_wlan_state)
 		wbutton_fct_2(1);
