From eb9d62440679b6163733e22d11fd695fd4f913bf Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Tue, 13 Aug 2013 22:17:27 +0200
Subject: [PATCH 42/51] acerhk: streamline verbosity parsing

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 drivers/platform/x86/acerhk.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

--- a/drivers/platform/x86/acerhk.c
+++ b/drivers/platform/x86/acerhk.c
@@ -971,7 +971,7 @@ static void __init setup_model_features(
 	/* setup available keys */
 	setup_keymap_model(acerhk_series);
 
-	if (verbose > 1)
+	if (verbose)
 		print_features();
 }
 
@@ -1050,7 +1050,7 @@ static void init_keymap_input(void)
 	set_keymap_name(k_next, KEY_NEXTSONG);
 	set_keymap_name(k_display, KEY_MEDIA);	/* also not happy with this */
 
-	if (verbose > 1)
+	if (verbose)
 		print_mapping();
 }
 
@@ -1390,12 +1390,12 @@ static int acerhk_probe(struct platform_
 		ret = -EAGAIN;
 	} else {
 		reg1 = ioremap(0xf0000, 0xffff);
-		if (verbose > 1)
+		if (verbose)
 			pr_info("area from 0xf000 to 0xffff mapped to %p\n",
 				reg1);
 
 		reg2 = ioremap(0xe0000, 0xffff);
-		if (verbose > 1)
+		if (verbose)
 			pr_info("area from 0xe000 to 0xffff mapped to %p\n",
 				reg2);
 
