From 2d29e83c726f85001d7c387babf59719c3b66db8 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Tue, 13 Aug 2013 22:39:51 +0200
Subject: [PATCH 44/65] acerhk: simplify safeguards against building on non-x86

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 drivers/platform/x86/acerhk.c | 12 +++---------
 1 file changed, 3 insertions(+), 9 deletions(-)

--- a/drivers/platform/x86/acerhk.c
+++ b/drivers/platform/x86/acerhk.c
@@ -19,11 +19,9 @@
  *
  */
 
-/* This driver is heavily dependent on the architecture, don't let
- * anyone without an X86 machine use it. On laptops with AMD64
- * architecture this driver is only useable in 32 bit mode.
- */
-#ifdef CONFIG_X86
+#ifndef CONFIG_X86
+#error This driver is only available for the x86 architecture
+#endif
 
 #define pr_fmt(fmt) KBUILD_MODNAME ": " fmt
 
@@ -1543,7 +1541,3 @@ MODULE_AUTHOR("Olaf Tauber");
 MODULE_AUTHOR("Stefan Lippers-Hollmann <s.l-h@gmx.de>");
 MODULE_DESCRIPTION("AcerHotkeys extra buttons keyboard driver");
 MODULE_LICENSE("GPL");
-
-#else
-#error This driver is only available for X86 architecture
-#endif
