From 9a893c6730104e62a7016d0cd83e0ebe9b3338cf Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Wed, 14 Aug 2013 04:12:00 +0200
Subject: [PATCH 50/65] acerhk: drop unused include of version.h

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 drivers/platform/x86/acerhk.c | 1 -
 1 file changed, 1 deletion(-)

--- a/drivers/platform/x86/acerhk.c
+++ b/drivers/platform/x86/acerhk.c
@@ -25,7 +25,6 @@
 
 #define pr_fmt(fmt) KBUILD_MODNAME ": " fmt
 
-#include <linux/version.h>
 #include <linux/moduleparam.h>
 
 #include <linux/module.h>
