From 3ecffe8219550639eb99aa09d7385ed414c0d75e Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Wed, 14 Aug 2013 20:39:05 +0200
Subject: [PATCH 52/65] acerhk: group stray #includes

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 drivers/platform/x86/acerhk.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

--- a/drivers/platform/x86/acerhk.c
+++ b/drivers/platform/x86/acerhk.c
@@ -41,6 +41,7 @@
 #include <linux/delay.h>
 #include <linux/efi.h>
 #include <linux/ioctl.h>
+#include <linux/preempt.h>
 
 #define ACERHK_MINOR MISC_DYNAMIC_MINOR
 
@@ -191,9 +192,6 @@ static void stop_polling(void);
 #define EC_CNTL_REG		0x66 /* Controller command register of EC (W) */
 #define EC_DATA_REG		0x62 /* EC data register (R/W) */
 
-
-#include <linux/preempt.h>
-
 #define KBD_STATUS_REG		0x64	/* Status register (R) */
 #define KBD_CNTL_REG		0x64	/* Controller command register (W) */
 #define KBD_DATA_REG		0x60	/* Keyboard data register (R/W) */
