From baf884cd6980aaeb7d171702c08770cf2fcf5261 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Wed, 14 Aug 2013 20:42:38 +0200
Subject: [PATCH 53/65] acerhk: drop obsolete #defines

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 drivers/platform/x86/acerhk.c | 14 --------------
 1 file changed, 14 deletions(-)

--- a/drivers/platform/x86/acerhk.c
+++ b/drivers/platform/x86/acerhk.c
@@ -119,15 +119,6 @@ typedef asmlinkage void (*bios_call) (st
  */
 #define MAX_POLLING_LOOPS 16U
 
-/* maximum length for model string */
-#define ACERHK_MODEL_STRLEN 16
-
-/* size of mapped areas */
-#define AREA_SIZE 0xffff
-
-/* needed for colussi algorithm */
-#define XSIZE     20
-
 /* Module parameters */
 static int verbose = 1;
 
@@ -187,11 +178,6 @@ static void start_polling(void);
 static void stop_polling(void);
 
 /* {{{ Experimental use of dritek keyboard extension */
-
-#define EC_STATUS_REG		0x66 /* Status register of EC (R) */
-#define EC_CNTL_REG		0x66 /* Controller command register of EC (W) */
-#define EC_DATA_REG		0x62 /* EC data register (R/W) */
-
 #define KBD_STATUS_REG		0x64	/* Status register (R) */
 #define KBD_CNTL_REG		0x64	/* Controller command register (W) */
 #define KBD_DATA_REG		0x60	/* Keyboard data register (R/W) */
