From 9b95e566d62df431bd647b290d23b252cf6a4e84 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Wed, 14 Aug 2013 20:47:01 +0200
Subject: [PATCH 54/65] acerhk: whitespace fixes for module parameter
 definition

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 drivers/platform/x86/acerhk.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

--- a/drivers/platform/x86/acerhk.c
+++ b/drivers/platform/x86/acerhk.c
@@ -119,11 +119,9 @@ typedef asmlinkage void (*bios_call) (st
  */
 #define MAX_POLLING_LOOPS 16U
 
-/* Module parameters */
+/* module parameters */
 static int verbose = 1;
-
 module_param(verbose, int, 0444);
-
 MODULE_PARM_DESC(verbose, "output additional information");
 
 /* input device */
