From e6cad7c330c24bd24d33b56fbded6293932c100b Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Wed, 14 Aug 2013 21:01:08 +0200
Subject: [PATCH 56/65] acerhk: simplify variable definition in find_hk_area()

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 drivers/platform/x86/acerhk.c | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

--- a/drivers/platform/x86/acerhk.c
+++ b/drivers/platform/x86/acerhk.c
@@ -626,12 +626,8 @@ static int launch_connect(int val)
 #ifndef __x86_64__
 static unsigned long __init find_hk_area(void)
 {
-	long offset, sig;
-	unsigned int fkt;
-	fkt = 0;
-
-	/* offset to signature in io area */
-	sig = -1;
+	long offset, sig = -1; /* offset to signature in io area */
+	unsigned int fkt =  0;
 
 	/* Look for signature, start at 0xf0000, search until 0xffff0 */
 	for (offset = 0; offset < 0xfffd; offset += 16) {
