From bb1ba024339260d4a0c49575e3ef1a55c361ba91 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Wed, 14 Aug 2013 21:08:40 +0200
Subject: [PATCH 57/65] acerhk: whitespace fixes

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 drivers/platform/x86/acerhk.c | 24 +++++-------------------
 1 file changed, 5 insertions(+), 19 deletions(-)

--- a/drivers/platform/x86/acerhk.c
+++ b/drivers/platform/x86/acerhk.c
@@ -171,13 +171,11 @@ static int acerhk_wlan_state;
 /* bluetooth hardware toggle */
 static int acerhk_bluetooth_state;
 
-
 /* {{{ Experimental use of dritek keyboard extension */
 #define KBD_STATUS_REG		0x64	/* Status register (R) */
 #define KBD_CNTL_REG		0x64	/* Controller command register (W) */
 #define KBD_DATA_REG		0x60	/* Keyboard data register (R/W) */
 
-
 static inline int my_i8042_read_status(void)
 {
 	return inb(KBD_STATUS_REG);
@@ -353,11 +351,9 @@ static void disable_mail_led_ec_3(void)
 		outw(0xC060, 0xCFC);
 	preempt_enable_no_resched();
 }
-
-/* }}} */
+/* Experimental use of dritek keyboard extension }}} */
 
 /* {{{ hardware access functions */
-
 /* call_bios_<model family>
  *
  * call request handler in mapped system rom
@@ -618,11 +614,9 @@ static int launch_connect(int val)
 
 	return regs.eax & 0xffff;
 }
-
-/* }}} */
+/* hardware access functions }}} */
 
 /* {{{ hardware probing */
-
 #ifndef __x86_64__
 static unsigned long __init find_hk_area(void)
 {
@@ -829,11 +823,9 @@ static void __init setup_model_features(
 	if (verbose >= 2)
 		print_features();
 }
-
-/* }}} */
+/* hardware probing }}} */
 
 /* {{{ key polling and translation */
-
 static void print_mapping(void)
 {
 	pr_info("key mapping help      0x%x\n", acerhk_name2event[k_help]);
@@ -996,7 +988,6 @@ static void init_input(void)
 {
 	int i;
 
-
 #ifndef STATIC_INPUT_DEV
 	/* allocate acerhk input device */
 	acerhk_input_dev_ptr = input_allocate_device();
@@ -1062,18 +1053,15 @@ static void start_polling(void)
 	}
 }
 
-
 static void release_input(void)
 {
 	stop_polling();
 	input_unregister_device(acerhk_input_dev_ptr);
 }
-
-/* }}} */
+/* key polling and translation }}} */
 
 
 /* {{{ file operations */
-
 static long acerhk_unlocked_ioctl(struct file *file, unsigned int cmd,
 				  unsigned long arg)
 {
@@ -1154,7 +1142,6 @@ static long acerhk_unlocked_ioctl(struct
 	return retval;
 }
 
-
 static int acerhk_open(struct inode *inode, struct file *file)
 {
 	return 0;
@@ -1192,8 +1179,7 @@ static struct miscdevice acerhk_misc_dev
 	.name = KBUILD_MODNAME,
 	.fops = &acerhk_fops,
 };
-
-/* }}} */
+/* file operations }}} */
 
 #ifndef __x86_64__
 static void model_init(void)
