From fd7cc5b33608fe6276b02781cb382e8d08918430 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Wed, 14 Aug 2013 21:09:22 +0200
Subject: [PATCH 58/65] acerhk: drop undefined conditionals for ancient kernels
 (STATIC_INPUT_DEV)

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 drivers/platform/x86/acerhk.c | 7 -------
 1 file changed, 7 deletions(-)

--- a/drivers/platform/x86/acerhk.c
+++ b/drivers/platform/x86/acerhk.c
@@ -126,9 +126,6 @@ MODULE_PARM_DESC(verbose, "output additi
 
 /* input device */
 static struct input_dev *acerhk_input_dev_ptr;
-#ifdef STATIC_INPUT_DEV
-static struct input_dev acerhk_input_dev;
-#endif
 
 /* mapped IO area from 0xf0000 */
 static void *reg1;
@@ -988,15 +985,11 @@ static void init_input(void)
 {
 	int i;
 
-#ifndef STATIC_INPUT_DEV
 	/* allocate acerhk input device */
 	acerhk_input_dev_ptr = input_allocate_device();
 
 	/* enter some name */
 	acerhk_input_dev_ptr->name = "Acer hotkey driver";
-#else
-	acerhk_input_dev_ptr = &acerhk_input_dev;
-#endif
 
 	/* some laptops have a mail led, should I announce it here? */
 	acerhk_input_dev_ptr->evbit[0] = BIT(EV_KEY);
