From 2c047bf602b87f9202c87d18c5f3b5f2a513fc27 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Wed, 14 Aug 2013 22:02:21 +0200
Subject: [PATCH 59/59] acerhk: check for failures of input_register_device()

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 drivers/platform/x86/acerhk.c | 15 +++++++++++----
 1 file changed, 11 insertions(+), 4 deletions(-)

--- a/drivers/platform/x86/acerhk.c
+++ b/drivers/platform/x86/acerhk.c
@@ -981,9 +981,10 @@ static void acerhk_poll_event(unsigned l
 	add_timer(&acerhk_timer_poll);
 }
 
-static void init_input(void)
+static int init_input(void)
 {
 	int i;
+	int ret;
 
 	/* allocate acerhk input device */
 	acerhk_input_dev_ptr = input_allocate_device();
@@ -1008,9 +1009,16 @@ static void init_input(void)
 	if (verbose)
 		pr_info("registered input device\n");
 
-	input_register_device(acerhk_input_dev_ptr);
+	ret = input_register_device(acerhk_input_dev_ptr);
+	if (ret) {
+		pr_err("failed to register power button: %d\n", ret);
+		input_free_device(acerhk_input_dev_ptr);
+	}
+
 	init_timer(&acerhk_timer_poll);
 	acerhk_polling_state = 0;
+
+	return ret;
 }
 
 static void stop_polling(void)
@@ -1740,7 +1748,7 @@ static int acerhk_probe(struct platform_
 				reg2);
 
 		/* attach to input system */
-		init_input();
+		ret = init_input();
 #ifndef __x86_64__
 		bios_routine = find_hk_area();
 
@@ -1749,7 +1757,6 @@ static int acerhk_probe(struct platform_
 
 		/* Without a bios routine we cannot do anything except on dritek
 		   type HW, unload on other types */
-		ret = 0;
 		if (verbose && bios_routine)
 			pr_info("bios routine found at 0x%x\n", bios_routine);
 
