From 0cedd514f408604633f464dddb5f40383633780d Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Thu, 15 Aug 2013 21:43:04 +0200
Subject: [PATCH 62/65] acerhk: dissolve setup_keymap_model() into its only
 caller

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 drivers/platform/x86/acerhk.c | 37 ++++++-------------------------------
 1 file changed, 6 insertions(+), 31 deletions(-)

--- a/drivers/platform/x86/acerhk.c
+++ b/drivers/platform/x86/acerhk.c
@@ -601,35 +601,6 @@ static unsigned long __init find_hk_area
 }
 #endif
 
-static void __init setup_keymap_model(void)
-{
-	/* clear mapping keycode -> keyname, */
-	memset(&acerhk_key2name[0], k_none, sizeof(acerhk_key2name));
-
-	/*
-	 * keys are handled by dritek EC
-	 *
-	 * Aspire 13xx series laptops use dritek hardware, no
-	 * acerhk-mapping needed
-	 * VolUp and VolDown are managed as normal keys
-	 * 1300/1310 series should have P1, P2, Mail, WWW, Mute buttons
-	 * 1353 has bluetooth, wifi, p1, p2, www, mail, help, setup, power
-	 * and mute
-	 * Aspire 1400/1450/Ferrari use dritek EC, too
-	 * 1450 should have bluetooth, wifi, p1, p2, www, mail, help,
-	 * setup, power and mute
-	 * Aspire 1700 uses dritek EC, too
-	 * 1700 should have bluetooth, wifi, p1, p2, www, mail, help,
-	 * setup, power and mute
-	 * need the MM-buttons Activation? (forward, shuffle, ...)
-	 * 2000 hast lots of MM buttons
-	 * 2010 should have bluetooth, wifi, p1, p2, www, mail, help,
-	 * setup, power and mute
-	 */
-	acerhk_key2name[1] = k_none;
-	acerhk_key2name[2] = k_none;
-}
-
 static void __init setup_model_features(void)
 {
 	/* set the correct bios call function according to type */
@@ -638,8 +609,12 @@ static void __init setup_model_features(
 	if (verbose >= 3)
 		pr_info("using call_bios_6xx mode\n");
 
-	/* setup available keys */
-	setup_keymap_model();
+	/* setup available keys, clear mapping keycode -> keyname, */
+	memset(&acerhk_key2name[0], k_none, sizeof(acerhk_key2name));
+
+	/* keys are handled by dritek EC */
+	acerhk_key2name[1] = k_none;
+	acerhk_key2name[2] = k_none;
 }
 /* hardware probing }}} */
 
