From 9edbfafcc1f1b0d2d390bd9f6a97e69b07333b0c Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Mon, 19 Aug 2013 03:29:36 +0200
Subject: [PATCH 65/65] acerhk: make functions void, which are never checked
 anyways

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 drivers/platform/x86/acerhk.c | 31 ++++---------------------------
 1 file changed, 4 insertions(+), 27 deletions(-)

--- a/drivers/platform/x86/acerhk.c
+++ b/drivers/platform/x86/acerhk.c
@@ -383,10 +383,8 @@ static int get_thermal_event(void)
  *
  * val: 0       turns off the LED
  *      1       turns the LED to green/blue
- *
- * return value: ?
  */
-static int wbutton_fct_1(int val)
+static void wbutton_fct_1(int val)
 {
 	struct register_buffer regs;
 
@@ -404,11 +402,7 @@ static int wbutton_fct_1(int val)
 	} else {
 		if (verbose >= 4)
 			pr_info("wbutton function 1 not supported\n");
-
-		regs.eax = 0x00;
 	}
-
-	return regs.eax & 0xffff;
 }
 
 /* wbutton_fct_2
@@ -417,10 +411,8 @@ static int wbutton_fct_1(int val)
  *
  * val: 0       turns off the LED
  *      1       turns the LED to orange
- *
- * return value: ?
  */
-static int wbutton_fct_2(int val)
+static void wbutton_fct_2(int val)
 {
 	struct register_buffer regs;
 
@@ -438,11 +430,7 @@ static int wbutton_fct_2(int val)
 	} else {
 		if (verbose >= 4)
 			pr_info("wbutton function 2 not supported\n");
-
-		regs.eax = 0x00;
 	}
-
-	return regs.eax & 0xffff;
 }
 
 /* get_nr_events
@@ -475,11 +463,8 @@ static int get_nr_events(void)
  *
  * val: 0 - switch led off
  *		1 - switch led on
- *
- * return value: 1 - action succesfull (val valid)
- *				 0 - no action taken (val invalid)
  */
-static int set_mail_led(int val)
+static void set_mail_led(int val)
 {
 	struct register_buffer regs;
 
@@ -511,11 +496,7 @@ static int set_mail_led(int val)
 	} else {
 		if (verbose >= 4)
 			pr_info("mail led not supported\n");
-
-		regs.eax = 0x00;
 	}
-
-	return regs.eax & 0xffff;
 }
 
 /* launch_connect
@@ -523,7 +504,7 @@ static int set_mail_led(int val)
  * does what?
  * val: 1 - only known value from windows driver
  */
-static int launch_connect(int val)
+static void launch_connect(int val)
 {
 	struct register_buffer regs;
 
@@ -540,11 +521,7 @@ static int launch_connect(int val)
 	} else {
 		if (verbose >= 4)
 			pr_info("connect not supported\n");
-
-		regs.eax = 0x00;
 	}
-
-	return regs.eax & 0xffff;
 }
 /* hardware access functions }}} */
 
