--- a/drivers/platform/x86/acerhk.c
+++ b/drivers/platform/x86/acerhk.c
@@ -2863,10 +2863,10 @@ static int acerhk_resume(struct platform
 #endif
 
 static struct file_operations acerhk_fops = {
-owner:	THIS_MODULE,
-ioctl:	acerhk_ioctl,
-open:	acerhk_open,
-release:acerhk_release,
+owner:		THIS_MODULE,
+unlocked_ioctl:	acerhk_ioctl,
+open:		acerhk_open,
+release:	acerhk_release,
 };
 
 static struct miscdevice acerhk_misc_dev = {
