acerhk: don't depend on EXPERIMENTAL

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>

--- a/drivers/platform/x86/Kconfig
+++ b/drivers/platform/x86/Kconfig
@@ -57,7 +57,7 @@ config ACERHDF
 
 config ACERHK
 	tristate "Acer Travelmate laptop special key support"
-	depends on EXPERIMENTAL && X86
+	depends on X86
 	---help---
 	  This is an experimental keyboard driver for acer laptops. If you
 	  have a notebook with a ipw2X00 wireless card, it allows you to
