From aa89be2e984f6d8174ccad80d059d81e7b1d3696 Mon Sep 17 00:00:00 2001
From: Christian Lamparter <chunkeey@googlemail.com>
Date: Fri, 29 Mar 2013 22:50:51 +0100
Subject: [PATCH 06/65] r92su: firmware signature is of __le16 type

Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
---
 r92su/fw.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/net/wireless/r92su/fw.c
+++ b/drivers/net/wireless/r92su/fw.c
@@ -51,8 +51,8 @@ static int r92su_parse_firmware(struct r
 
 	r92su->fw_header = hdr = (const void *) r92su->fw->data;
 
-	if ((hdr->signature != cpu_to_le32(R8192SU_FW_SIGNATURE)) &&
-	    (hdr->signature != cpu_to_le32(R8712SU_FW_SIGNATURE))) {
+	if ((hdr->signature != cpu_to_le16(R8192SU_FW_SIGNATURE)) &&
+	    (hdr->signature != cpu_to_le16(R8712SU_FW_SIGNATURE))) {
 		wiphy_err(r92su->wdev.wiphy, "firmware signature check has failed.\n");
 		return -EINVAL;
 	}
