From 50ebe03ff28c2e9cdf3a814a65b913acba49da8c Mon Sep 17 00:00:00 2001
From: Christian Lamparter <chunkeey@googlemail.com>
Date: Fri, 5 Apr 2013 20:56:41 +0200
Subject: [PATCH 12/65] r92su: print error when firmware wasn't found

Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
---
 r92su/fw.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

--- a/drivers/net/wireless/r92su/fw.c
+++ b/drivers/net/wireless/r92su/fw.c
@@ -302,8 +302,11 @@ int r92su_load_firmware(struct r92su *r9
 
 	err = request_firmware(&r92su->fw, RTL8192SU_FIRMWARE,
 			       &r92su->udev->dev);
-	if (err)
+	if (err) {
+		wiphy_err(r92su->wdev.wiphy, "firmware '%s' not found.\n",
+			  RTL8192SU_FIRMWARE);
 		return err;
+	}
 
 	err = r92su_parse_firmware(r92su);
 	if (err)
