From df8455c25dd9d54d96c7bdd31f8418c03c0454e3 Mon Sep 17 00:00:00 2001
From: Christian Lamparter <chunkeey@googlemail.com>
Date: Sat, 6 Apr 2013 02:17:35 +0200
Subject: [PATCH 14/65] r92su: fix fwdbg length printout on BE architectures

Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
---
 r92su/event.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/wireless/r92su/event.c
+++ b/drivers/net/wireless/r92su/event.c
@@ -182,7 +182,7 @@ void r92su_c2h_event(struct r92su *r92su
 	}
 
 	wiphy_notice(r92su->wdev.wiphy, "c2h event:%x len:%d\n",
-		     c2h->event, c2h->len);
+		     c2h->event, le16_to_cpu(c2h->len));
 
 	switch (c2h->event) {
 	ADD_HANDLER(C2H_FWDBG_EVENT, c2h_fwdbg_event);
