From 3c3c673464c459feda498b3856776bf9f38ca5a8 Mon Sep 17 00:00:00 2001
From: Christian Lamparter <chunkeey@googlemail.com>
Date: Sat, 6 Apr 2013 12:23:38 +0200
Subject: [PATCH 16/65] r92su: port wildcard scan feature from vendor driver

Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
---
 r92su/fw.c | 5 +++++
 1 file changed, 5 insertions(+)

--- a/drivers/net/wireless/r92su/fw.c
+++ b/drivers/net/wireless/r92su/fw.c
@@ -108,6 +108,11 @@ static int r92su_prepare_firmware(struct
 	dmem->vcs_type = 2; /* 0: off, 1: on, 2: auto */
 	dmem->vcs_mode = 1; /* 0: off(presumably), 1:RTS/CTS, 2:CTS-Self */
 
+	/* F/W will issue two probe request. One is with ssid ( if exists ),
+	 * another is with the wildcard ssid.
+	 */
+	dmem->rsvd024 = 1;
+
 	dmem->turbo_mode = 0;
 	dmem->low_power_mode = 0;
 	dmem->chip_version = r92su->chip_rev; /* not necessarily correct ?! */
