From 9b74dc4638447fb991f35955f3c562b8cc8bc10e Mon Sep 17 00:00:00 2001
From: Christian Lamparter <chunkeey@googlemail.com>
Date: Sat, 6 Apr 2013 16:57:54 +0200
Subject: [PATCH 18/65] r92su: fix type of length in ssid field

Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
---
 r92su/h2cc2h.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/wireless/r92su/h2cc2h.h
+++ b/drivers/net/wireless/r92su/h2cc2h.h
@@ -246,7 +246,7 @@ struct h2c_op_mode {
 } __packed;
 
 struct h2c_ssid {		/* ndis_802_11_ssid */
-	u32 length;
+	__le32 length;
 	u8 ssid[IEEE80211_MAX_SSID_LEN];
 } __packed;
 
