From 641a9032e53ed0a63e5116edfa206ee858ac4eaa Mon Sep 17 00:00:00 2001
From: Christian Lamparter <chunkeey@googlemail.com>
Date: Sun, 7 Apr 2013 03:20:26 +0200
Subject: [PATCH 26/65] r92su: add note about tcp rx offload

Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
---
 r92su/rx.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

--- a/drivers/net/wireless/r92su/rx.c
+++ b/drivers/net/wireless/r92su/rx.c
@@ -233,7 +233,6 @@ static void r92su_rx_deliver(struct r92s
 {
 	skb_reset_mac_header(skb);
 	skb->dev = r92su->wdev.netdev;
-	/* TODO: we can check the TCP checksum in the header */
 	skb->ip_summed = CHECKSUM_NONE;
 	skb->protocol = eth_type_trans(skb, skb->dev);
 	__r92su_rx_deliver(r92su, skb);
@@ -618,6 +617,15 @@ r92su_rx_hw_header_check(struct r92su *r
 		return RX_DROP;
 	}
 
+	/* TCP/IP checksum offloading needs to be tested and verified first.
+	 * If you enable this code, don't forget to edit r92su_rx_deliver!
+	 *
+	 * if (GET_RX_DESC_TCP_CHK_VALID(&rx->hdr) &&
+	 *     GET_RX_DESC_TCP_CHK_RPT(&rx->hdr) &&
+	 *     GET_RX_DESC_IP_CHK_RPT(&rx->hdr))
+	 *	skb->ip_summed = CHECKSUM_UNNECESSARY;
+	 */
+
 	/* report icv error
 	 * The vendor driver ignores the flag, probably because someone
 	 * "knew" that the hardware/firmware doesn't calculate the right
