From 0f1aba21e0f929b966537e99f5675307eb88f609 Mon Sep 17 00:00:00 2001
From: Christian Lamparter <chunkeey@googlemail.com>
Date: Sun, 7 Apr 2013 21:05:11 +0200
Subject: [PATCH 27/65] r92su: replace fixed value with proper symbol

Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
---
 r92su/sta.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/drivers/net/wireless/r92su/sta.h
+++ b/drivers/net/wireless/r92su/sta.h
@@ -112,7 +112,8 @@ struct r92su_sta {
 
 	u8 mac_addr[ETH_ALEN];
 	unsigned int mac_id;
-	unsigned int tx_seq[16];
+	unsigned int tx_seq;
+	unsigned int tx_seq_tid[IEEE80211_NUM_TIDS];
 	unsigned int aid;
 
 	bool qos_sta;
