From 80bd5a734cf5d22ac437e6f176e32260d4672dcb Mon Sep 17 00:00:00 2001
From: Christian Lamparter <chunkeey@googlemail.com>
Date: Mon, 8 Apr 2013 19:30:36 +0200
Subject: [PATCH 28/65] r92su: fix crash due to unsafe rcu operation

Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
---
 r92su/sta.c | 2 ++
 r92su/sta.h | 3 ++-
 2 files changed, 4 insertions(+), 1 deletion(-)

--- a/drivers/net/wireless/r92su/sta.c
+++ b/drivers/net/wireless/r92su/sta.c
@@ -189,7 +189,9 @@ void r92su_sta_del(struct r92su *r92su,
 
 void r92su_sta_add(struct r92su *r92su, struct r92su_sta *new_sta)
 {
+	rcu_read_lock();
 	r92su_free_sta(r92su_sta_xchg(r92su, new_sta));
+	rcu_read_unlock();
 }
 
 static u32 get_random_wep_seq(void)
--- a/drivers/net/wireless/r92su/sta.h
+++ b/drivers/net/wireless/r92su/sta.h
@@ -147,9 +147,10 @@ struct r92su_key *r92su_key_alloc(const
 
 void r92su_key_free(struct r92su_key *key);
 
+void r92su_sta_add(struct r92su *r92su, struct r92su_sta *new_sta);
+
 /* the following functions need rcu_read_lock! */
 struct r92su_sta *r92su_sta_get(struct r92su *r92su, const u8 *mac_addr);
 void r92su_sta_del(struct r92su *r92su, int mac_id);
-void r92su_sta_add(struct r92su *r92su, struct r92su_sta *new_sta);
 
 #endif /* __R92SU_STA_H__ */
