From 939b1ce1605a37b0474e583953f236ee346d4dfa Mon Sep 17 00:00:00 2001
From: Christian Lamparter <chunkeey@googlemail.com>
Date: Mon, 8 Apr 2013 22:17:53 +0200
Subject: [PATCH 30/65] r92su: don't schedule tasklet when device is not
 initialized

Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
---
 r92su/rx.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/drivers/net/wireless/r92su/rx.c
+++ b/drivers/net/wireless/r92su/rx.c
@@ -1173,7 +1173,8 @@ void r92su_rx(struct r92su *r92su, void
 		}
 		buf += ALIGN(hdr_len + pkt_len, r92su->rx_alignment);
 	}
-	tasklet_schedule(&r92su->rx_tasklet);
+	if (r92su_is_connected(r92su))
+		tasklet_schedule(&r92su->rx_tasklet);
 	return;
 
 err_garbage:
