From 78c56df12d92b3b8f8b170302ba227c39040fdbb Mon Sep 17 00:00:00 2001
From: Christian Lamparter <chunkeey@googlemail.com>
Date: Tue, 9 Apr 2013 18:02:06 +0200
Subject: [PATCH 33/65] r92su: tear down link when device dies

Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
---
 r92su/r92su.h | 4 ++++
 1 file changed, 4 insertions(+)

--- a/drivers/net/wireless/r92su/r92su.h
+++ b/drivers/net/wireless/r92su/r92su.h
@@ -262,7 +262,11 @@ static inline void r92su_set_state(struc
 
 static inline void r92su_mark_dead(struct r92su *r92su)
 {
+	struct net_device *ndev = r92su->wdev.netdev;
 	r92su_set_state(r92su, R92SU_DEAD);
+
+	if (ndev)
+		netif_carrier_off(ndev);
 }
 
 static inline
