From e5024e5997ffe295f651840124ed7348968c8a3c Mon Sep 17 00:00:00 2001
From: Christian Lamparter <chunkeey@googlemail.com>
Date: Wed, 10 Apr 2013 00:06:53 +0200
Subject: [PATCH 35/65] r92su: add switch to disable ht

Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
---
 r92su/usb.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

--- a/drivers/net/wireless/r92su/usb.c
+++ b/drivers/net/wireless/r92su/usb.c
@@ -38,6 +38,10 @@
 #include "rx.h"
 #include "tx.h"
 
+static bool modparam_noht;
+module_param_named(noht, modparam_noht, bool, S_IRUGO);
+MODULE_PARM_DESC(noht, "Disable MPDU aggregation.");
+
 static int r92su_sync_write(struct r92su *r92su, u16 address,
 			    const void *data, u16 size)
 {
@@ -536,7 +540,8 @@ static int r92su_usb_probe(struct usb_in
 		goto err_out;
 	}
 
-	if (id->driver_info & DISABLE_HT)
+	if (id->driver_info & DISABLE_HT ||
+	    modparam_noht)
 		r92su->disable_ht = true;
 
 	r92su_set_state(r92su, R92SU_PROBE);
