From 0bd82bc25ff80acf78ef502c3e2fba9aa7a67af7 Mon Sep 17 00:00:00 2001
From: Christian Lamparter <chunkeey@googlemail.com>
Date: Sun, 14 Apr 2013 15:35:18 +0200
Subject: [PATCH 38/65] r92su: convert memcpy to direct assignments

Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
---
 r92su/cmd.c |  2 +-
 r92su/rx.c  | 15 +++------------
 2 files changed, 4 insertions(+), 13 deletions(-)

--- a/drivers/net/wireless/r92su/cmd.c
+++ b/drivers/net/wireless/r92su/cmd.c
@@ -205,7 +205,7 @@ int r92su_h2c_connect(struct r92su *r92s
 		return -ENOMEM;
 
 	bss = (struct h2cc2h_bss *)skb_put(skb, sizeof(*bss));
-	memcpy(bss, orig_bss, sizeof(*bss));
+	*bss = *orig_bss;
 
 	/* the ie_length also contains the fixed 12-byte ies (tsf/...) */
 	bss->ie_length = cpu_to_le32(12 + ie_len);
--- a/drivers/net/wireless/r92su/rx.c
+++ b/drivers/net/wireless/r92su/rx.c
@@ -537,12 +537,8 @@ r92su_rx_data_to_8023(struct r92su *r92s
 	}
 
 	if (is_amsdu) {
-		struct r92su_rx_info tmp_rx_info;
+		struct r92su_rx_info tmp_rx_info = *r92su_get_rx_info(skb);
 
-		memcpy(&tmp_rx_info, r92su_get_rx_info(skb),
-		       sizeof(tmp_rx_info));
-
-		/* we can use the skb queue without spinlocks */
 		ieee80211_amsdu_to_8023s(skb, queue,
 					 wdev_address(&r92su->wdev),
 					 r92su->wdev.iftype, 0, true);
@@ -553,13 +549,8 @@ r92su_rx_data_to_8023(struct r92su *r92s
 			return RX_DROP;
 		}
 
-		skb_queue_walk(queue, skb) {
-			struct r92su_rx_info *new_info;
-
-			new_info = r92su_get_rx_info(skb);
-			memcpy(new_info, &tmp_rx_info, sizeof(*new_info));
-		}
-
+		skb_queue_walk(queue, skb)
+			*r92su_get_rx_info(skb) = tmp_rx_info;
 	} else {
 		int err;
 
