From 8e508da62ec876930ebda94d68a816ed5b4c7b5c Mon Sep 17 00:00:00 2001
From: Christian Lamparter <chunkeey@googlemail.com>
Date: Sun, 14 Apr 2013 22:25:08 +0200
Subject: [PATCH 40/65] r92su: fix skb leak in error path

Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
---
 r92su/rx.c | 1 +
 1 file changed, 1 insertion(+)

--- a/drivers/net/wireless/r92su/rx.c
+++ b/drivers/net/wireless/r92su/rx.c
@@ -1010,6 +1010,7 @@ rx_drop:
 	rcu_read_unlock();
 	r92su_rx_dropped(r92su, 1);
 	__skb_queue_purge(&frames);
+	dev_kfree_skb_any(skb);
 
 #undef __RX_HANDLER
 #undef RX_HANDLER
