From c96f8e9596675b699912bc20ada90d3e84e312c1 Mon Sep 17 00:00:00 2001
From: Christian Lamparter <chunkeey@googlemail.com>
Date: Tue, 16 Apr 2013 00:20:10 +0200
Subject: [PATCH 43/65] r92su: bss_priv->sta is not a rcu pointer

Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
---
 r92su/hw.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/wireless/r92su/hw.c
+++ b/drivers/net/wireless/r92su/hw.c
@@ -492,7 +492,7 @@ static void r92su_query_fw_rx_phy_status
 			bss->signal = qual;
 
 			bss_priv = r92su_get_bss_priv(bss);
-			sta = rcu_dereference(bss_priv->sta);
+			sta = bss_priv->sta;
 			if (sta)
 				sta->signal = qual;
 		}
