From 2828dbf2ef2014c956b58a416afcdf902917b776 Mon Sep 17 00:00:00 2001
From: Christian Lamparter <chunkeey@googlemail.com>
Date: Sun, 21 Apr 2013 20:56:04 +0200
Subject: [PATCH 50/65] r92su: move noht module parameter into common code

Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
---
 r92su/main.c | 8 ++++++++
 r92su/usb.c  | 7 +------
 2 files changed, 9 insertions(+), 6 deletions(-)

--- a/drivers/net/wireless/r92su/main.c
+++ b/drivers/net/wireless/r92su/main.c
@@ -29,6 +29,7 @@
  *
  *****************************************************************************/
 #include <linux/kernel.h>
+#include <linux/module.h>
 #include <linux/ieee80211.h>
 #include <linux/etherdevice.h>
 #include <linux/if_arp.h>
@@ -47,6 +48,10 @@
 #include "debug.h"
 #include "debugfs.h"
 
+static bool modparam_noht;
+module_param_named(noht, modparam_noht, bool, S_IRUGO);
+MODULE_PARM_DESC(noht, "Disable MPDU aggregation.");
+
 #define CHAN2G(_hw_value, _freq, _flags) {	\
 	.band		= IEEE80211_BAND_2GHZ,	\
 	.center_freq	= (_freq),		\
@@ -1530,6 +1535,9 @@ struct r92su *r92su_alloc(struct device
 	mutex_init(&r92su->lock);
 	spin_lock_init(&r92su->rx_path);
 
+	if (modparam_noht)
+		r92su->disable_ht = true;
+
 	INIT_LIST_HEAD(&r92su->sta_list);
 	/* Note: The sta_lock is only needed, if an entry in the
 	 * station list is updated. The station data itself is
--- a/drivers/net/wireless/r92su/usb.c
+++ b/drivers/net/wireless/r92su/usb.c
@@ -40,10 +40,6 @@
 #include "trace.h"
 #include "debug.h"
 
-static bool modparam_noht;
-module_param_named(noht, modparam_noht, bool, S_IRUGO);
-MODULE_PARM_DESC(noht, "Disable MPDU aggregation.");
-
 static bool r92su_deal_with_usb_errors(struct r92su *r92su, int err)
 {
 	switch (err) {
@@ -569,8 +565,7 @@ static int r92su_usb_probe(struct usb_in
 		goto err_out;
 	}
 
-	if (id->driver_info & DISABLE_HT ||
-	    modparam_noht)
+	if (id->driver_info & DISABLE_HT)
 		r92su->disable_ht = true;
 
 	r92su_set_state(r92su, R92SU_PROBE);
