From 2ec498156bc94630bf24eec3974be7739407842b Mon Sep 17 00:00:00 2001
From: Christian Lamparter <chunkeey@googlemail.com>
Date: Sun, 21 Apr 2013 20:57:43 +0200
Subject: [PATCH 51/65] r92su: make unexpected cmd seq message less obnoxious

Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
---
 r92su/event.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/net/wireless/r92su/event.c
+++ b/drivers/net/wireless/r92su/event.c
@@ -177,8 +177,8 @@ void r92su_c2h_event(struct r92su *r92su
 	trace_r92su_c2h(wiphy_dev(r92su->wdev.wiphy), c2h);
 
 	if (sequence != c2h->cmd_seq) {
-		R92SU_ERR(r92su, "received an c2h event out of sequence.\n");
-		R92SU_ERR(r92su, "expected: %d, got %d\n", sequence,
+		R92SU_DBG(r92su, "received an c2h event out of sequence.\n");
+		R92SU_DBG(r92su, "expected: %d, got %d\n", sequence,
 			  c2h->cmd_seq);
 
 		r92su->c2h_seq = c2h->cmd_seq + 1;
