From 8daea2ad41b97f8804bcb92eec874a71ea244256 Mon Sep 17 00:00:00 2001
From: Christian Lamparter <chunkeey@googlemail.com>
Date: Tue, 23 Apr 2013 00:37:00 +0200
Subject: [PATCH 56/65] r92su: use new channel width types

Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
---
 r92su/main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/wireless/r92su/main.c
+++ b/drivers/net/wireless/r92su/main.c
@@ -817,7 +817,7 @@ static int r92su_set_monitor_channel(str
 	struct r92su *r92su = wiphy_priv(wiphy);
 	int err = -EAGAIN;
 
-	if (cfg80211_get_chandef_type(chandef) != NL80211_CHAN_NO_HT)
+	if (chandef->width != NL80211_CHAN_WIDTH_20_NOHT)
 		return -EOPNOTSUPP;
 
 	mutex_lock(&r92su->lock);
