From 2789cb2811a602b6d2964d770d856cf085e02777 Mon Sep 17 00:00:00 2001
From: Christian Lamparter <chunkeey@googlemail.com>
Date: Wed, 24 Apr 2013 00:08:11 +0200
Subject: [PATCH 58/65] r92su: don't print any error messages when device is
 removed

Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
---
 r92su/r92su.h |  5 +++++
 r92su/usb.c   | 11 +++++++++--
 2 files changed, 14 insertions(+), 2 deletions(-)

--- a/drivers/net/wireless/r92su/r92su.h
+++ b/drivers/net/wireless/r92su/r92su.h
@@ -232,6 +232,11 @@ struct r92su_bss_priv {
 	bool control_port;
 };
 
+static inline bool r92su_is_dead(struct r92su *r92su)
+{
+	return r92su->state == R92SU_DEAD;
+}
+
 static inline bool r92su_is_probing(struct r92su *r92su)
 {
 	return r92su->state >= R92SU_PROBE;
--- a/drivers/net/wireless/r92su/usb.c
+++ b/drivers/net/wireless/r92su/usb.c
@@ -42,10 +42,16 @@
 
 static bool r92su_deal_with_usb_errors(struct r92su *r92su, int err)
 {
-	switch (err) {
-	case 0:
+	if (likely(err == 0))
+		return false;
+
+	if (r92su_is_dead(r92su))
 		return false;
 
+	if (r92su->intf->condition == USB_INTERFACE_UNBINDING)
+		return false;
+
+	switch (err) {
 	case -ENOENT:
 	case -ECONNRESET:
 	case -ENODEV:
@@ -558,6 +564,7 @@ static int r92su_usb_probe(struct usb_in
 
 	r92su_set_state(r92su, R92SU_PROBE);
 
+	r92su->intf = intf;
 	r92su->udev = interface_to_usbdev(intf);
 	usb_set_intfdata(intf, r92su);
 
