r92su: compat backports

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>

--- a/drivers/net/wireless/r92su/main.c
+++ b/drivers/net/wireless/r92su/main.c
@@ -34,6 +34,7 @@
 #include <linux/etherdevice.h>
 #include <linux/if_arp.h>
 #include <net/cfg80211.h>
+#include <linux/version.h>
 
 #include "r92su.h"
 #include "reg.h"
@@ -827,7 +828,11 @@ static int r92su_set_monitor_channel(str
 	struct r92su *r92su = wiphy_priv(wiphy);
 	int err = -EAGAIN;
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 10, 0)
 	if (chandef->width != NL80211_CHAN_WIDTH_20_NOHT)
+#else
+	if (cfg80211_get_chandef_type(chandef) != NL80211_CHAN_NO_HT)
+#endif
 		return -EOPNOTSUPP;
 
 	mutex_lock(&r92su->lock);
--- a/drivers/net/wireless/r92su/sta.c
+++ b/drivers/net/wireless/r92su/sta.c
@@ -31,6 +31,7 @@
 #include <linux/kernel.h>
 #include <linux/workqueue.h>
 #include <linux/random.h>
+#include <linux/version.h>
 #include "r92su.h"
 #include "sta.h"
 #include "rx.h"
@@ -273,12 +274,20 @@ void r92su_sta_set_sinfo(struct r92su *r
 	else
 		sinfo->rxrate.legacy = sta->last_rx_rate;
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 10, 0)
 	sinfo->sta_flags.mask = BIT(NL80211_STA_FLAG_ASSOCIATED) |
 				BIT(NL80211_STA_FLAG_AUTHENTICATED) |
+#else
+	sinfo->sta_flags.mask = BIT(NL80211_STA_FLAG_AUTHENTICATED) |
+#endif
 				BIT(NL80211_STA_FLAG_WME);
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 10, 0)
 	sinfo->sta_flags.set = BIT(NL80211_STA_FLAG_ASSOCIATED) |
 			       BIT(NL80211_STA_FLAG_AUTHENTICATED);
+#else
+	sinfo->sta_flags.set = BIT(NL80211_STA_FLAG_AUTHENTICATED);
+#endif
 
 	if (sta->qos_sta)
 		sinfo->sta_flags.set |= BIT(NL80211_STA_FLAG_WME);
