--- a/drivers/net/wireless/ath/carl9170/main.c
+++ b/drivers/net/wireless/ath/carl9170/main.c
@@ -1030,17 +1030,17 @@ static int carl9170_op_set_key(struct ie
 	    !(key->flags & IEEE80211_KEY_FLAG_PAIRWISE))
 		return -EOPNOTSUPP;
 
-	switch (key->cipher) {
-	case WLAN_CIPHER_SUITE_WEP40:
-		ktype = AR9170_ENC_ALG_WEP64;
+	switch (key->alg) {
+	case ALG_WEP:
+		if (key->keylen == WLAN_KEY_LEN_WEP40)
+			ktype = AR9170_ENC_ALG_WEP64;
+		else
+			ktype = AR9170_ENC_ALG_WEP128;
 		break;
-	case WLAN_CIPHER_SUITE_WEP104:
-		ktype = AR9170_ENC_ALG_WEP128;
-		break;
-	case WLAN_CIPHER_SUITE_TKIP:
+	case ALG_TKIP:
 		ktype = AR9170_ENC_ALG_TKIP;
 		break;
-	case WLAN_CIPHER_SUITE_CCMP:
+	case ALG_CCMP:
 		ktype = AR9170_ENC_ALG_AESCCMP;
 		break;
 	default:
@@ -1074,7 +1074,7 @@ static int carl9170_op_set_key(struct ie
 		if (err)
 			goto out;
 
-		if (key->cipher == WLAN_CIPHER_SUITE_TKIP) {
+		if (key->alg == ALG_TKIP) {
 			err = carl9170_upload_key(ar, i, sta ? sta->addr :
 						  NULL, ktype, 1,
 						  key->key + 16, 16);
@@ -1108,7 +1108,7 @@ static int carl9170_op_set_key(struct ie
 			if (err)
 				goto out;
 
-			if (key->cipher == WLAN_CIPHER_SUITE_TKIP) {
+			if (key->alg == ALG_TKIP) {
 				err = carl9170_upload_key(ar, key->hw_key_idx,
 							  NULL,
 							  AR9170_ENC_ALG_NONE,
--- a/drivers/net/wireless/ath/carl9170/tx.c
+++ b/drivers/net/wireless/ath/carl9170/tx.c
@@ -820,13 +820,12 @@ static int carl9170_tx_prepare(struct ar
 	if (info->control.hw_key) {
 		len += info->control.hw_key->icv_len;
 
-		switch (info->control.hw_key->cipher) {
-		case WLAN_CIPHER_SUITE_WEP40:
-		case WLAN_CIPHER_SUITE_WEP104:
-		case WLAN_CIPHER_SUITE_TKIP:
+		switch (info->control.hw_key->alg) {
+		case ALG_WEP:
+		case ALG_TKIP:
 			mac_tmp |= cpu_to_le16(AR9170_TX_MAC_ENCR_RC4);
 			break;
-		case WLAN_CIPHER_SUITE_CCMP:
+		case ALG_CCMP:
 			mac_tmp |= cpu_to_le16(AR9170_TX_MAC_ENCR_AES);
 			break;
 		default:
