--- a/drivers/net/wireless/ath/carl9170/main.c
+++ b/drivers/net/wireless/ath/carl9170/main.c
@@ -413,7 +413,7 @@ static void ar9170_restart_work(struct w
 }
 
 static int ar9170_op_add_interface(struct ieee80211_hw *hw,
-				   struct ieee80211_vif *vif)
+				   struct ieee80211_if_init_conf *conf)
 {
 	struct ar9170 *ar = hw->priv;
 	struct ath_common *common = &ar->common;
@@ -426,8 +426,8 @@ static int ar9170_op_add_interface(struc
 		goto unlock;
 	}
 
-	ar->vif = vif;
-	memcpy(common->macaddr, vif->addr, ETH_ALEN);
+	ar->vif = conf->vif;
+	memcpy(common->macaddr, conf->mac_addr, ETH_ALEN);
 
 	if (modparam_nohwcrypt || (ar->vif->type != NL80211_IFTYPE_STATION)) {
 		ar->rx_software_decryption = true;
@@ -447,7 +447,7 @@ unlock:
 }
 
 static void ar9170_op_remove_interface(struct ieee80211_hw *hw,
-				       struct ieee80211_vif *vif)
+				       struct ieee80211_if_init_conf *conf)
 {
 	struct ar9170 *ar = hw->priv;
 
