--- a/fs/aufs/cpup.c
+++ b/fs/aufs/cpup.c
@@ -42,7 +42,7 @@ void au_cpup_attr_timesizes(struct inode
 
 	h_inode = au_h_iptr(inode, au_ibstart(inode));
 	fsstack_copy_attr_times(inode, h_inode);
-	vfsub_copy_inode_size(inode, h_inode);
+	fsstack_copy_inode_size(inode, h_inode);
 }
 
 void au_cpup_attr_nlink(struct inode *inode, int force)
--- a/fs/aufs/vfsub.h
+++ b/fs/aufs/vfsub.h
@@ -26,7 +26,6 @@
 #ifdef __KERNEL__
 
 #include <linux/fs.h>
-#include <linux/fs_stack.h>
 
 /* ---------------------------------------------------------------------- */
 
@@ -49,14 +48,6 @@ enum {
 
 /* ---------------------------------------------------------------------- */
 
-static inline void vfsub_copy_inode_size(struct inode *inode,
-					 struct inode *h_inode)
-{
-	spin_lock(&inode->i_lock);
-	fsstack_copy_inode_size(inode, h_inode);
-	spin_unlock(&inode->i_lock);
-}
-
 static inline struct file *vfsub_dentry_open(struct path *path, int flags)
 {
 	path_get(path);
