From e415f0a14c077389fd5b6b8415d16f2ab4fc2f39 Mon Sep 17 00:00:00 2001
From: Florian Schilhabel <florian.c.schilhabel@googlemail.com>
Date: Wed, 14 Jul 2010 14:47:40 +0200
Subject: [PATCH 25/41] staging: rtl8192su: update TODO

Signed-off-by: Florian Schilhabel <florian.c.schilhabel@googlemail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/rtl8192su/TODO |    7 ++++++-
 1 files changed, 6 insertions(+), 1 deletions(-)

--- a/drivers/staging/rtl8192su/TODO
+++ b/drivers/staging/rtl8192su/TODO
@@ -1,4 +1,10 @@
 TODO:
+- merge realteks bugfixes and new features into the driver:
+  - an updated version of this driver can be found here:
+    http://www.getnet.eu/products_GN-621U.html
+  - note:
+    realtek has stripped alomost all comments from the source,
+    so please leave all comments that may help in development in the code.
 - prepare private ieee80211 stack for merge with rtl8187se's version:
   - remove rtl8192su's specific dead code
   - cleanup ieee80211.h
@@ -7,7 +13,6 @@ TODO:
 - switch to use shared "librtl" instead of private ieee80211 stack
 - switch to use LIB80211
 - switch to use MAC80211
-- switch to use EEPROM_93CX6
 - use kernel coding style
 - checkpatch.pl fixes
 - sparse fixes
