From 26cc71540ab80714f7635139526b4c0cb6fe465d Mon Sep 17 00:00:00 2001
From: Florian Schilhabel <florian.c.schilhabel@googlemail.com>
Date: Thu, 15 Jul 2010 19:02:59 +0200
Subject: [PATCH 28/40] staging: rtl819su: ieee80211_associate_procedure_wq() update

Signed-off-by: Florian Schilhabel <florian.c.schilhabel@googlemail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 .../rtl8192su/ieee80211/ieee80211_softmac.c        |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

--- a/drivers/staging/rtl8192su/ieee80211/ieee80211_softmac.c
+++ b/drivers/staging/rtl8192su/ieee80211/ieee80211_softmac.c
@@ -1407,6 +1407,12 @@ void ieee80211_associate_procedure_wq(st
 	//ieee->set_chan(ieee->dev, ieee->current_network.channel);
 	HTSetConnectBwMode(ieee, HT_CHANNEL_WIDTH_20, HT_EXTCHNL_OFFSET_NO_EXT);
 
+	if(ieee->eRFPowerState == eRfOff)
+	{
+            printk("=============>%s():Rf state is eRfOff, schedule ipsleave wq again,return\n",__FUNCTION__);
+		up(&ieee->wx_sem);
+		return;
+	}
 	ieee->associate_seq = 1;
 	ieee80211_associate_step1(ieee);
 
