From 1fc9cb288e144fbf98b04b37c5225d3c2b452012 Mon Sep 17 00:00:00 2001
From: Luis R. Rodriguez <lrodriguez@atheros.com>
Date: Thu, 7 Jan 2010 17:24:54 -0500
Subject: [PATCH 3/8] cfg80211: add debug print when we drop a bogus country IE

commit b74d12e116528fadc35f305eb6e9525a3a2b62b1 upstream.

Signed-off-by: Luis R. Rodriguez <lrodriguez@atheros.com>
Signed-off-by: John W. Linville <linville@tuxdriver.com>
---
 net/wireless/reg.c |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

--- a/net/wireless/reg.c
+++ b/net/wireless/reg.c
@@ -1811,8 +1811,10 @@ void regulatory_hint_11d(struct wiphy *w
 		goto out;
 
 	rd = country_ie_2_rd(country_ie, country_ie_len, &checksum);
-	if (!rd)
+	if (!rd) {
+		REG_DBG_PRINT("cfg80211: Ignoring bogus country IE\n");
 		goto out;
+	}
 
 	/*
 	 * This will not happen right now but we leave it here for the
